{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>status</th>\n",
       "      <th>sample</th>\n",
       "      <th>relatives</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Patient</td>\n",
       "      <td>22334455</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Relative</td>\n",
       "      <td>77092483</td>\n",
       "      <td>77092484:child</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Patient</td>\n",
       "      <td>77092484</td>\n",
       "      <td>77092482:father; 77092483:mother</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Relative</td>\n",
       "      <td>77092482</td>\n",
       "      <td>77092484:child</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Relative</td>\n",
       "      <td>77092481</td>\n",
       "      <td>77092479:child</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     status    sample                         relatives\n",
       "0   Patient  22334455                              None\n",
       "1  Relative  77092483                    77092484:child\n",
       "2   Patient  77092484  77092482:father; 77092483:mother\n",
       "3  Relative  77092482                    77092484:child\n",
       "4  Relative  77092481                    77092479:child"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trio_info=pd.read_excel('/storage/haplotypes/excels/trio_info.xlsx')\n",
    "trio_info['relatives'] = trio_info['relatives'].replace(np.nan, 'None', regex=False)\n",
    "trio_info['relatives']=trio_info['relatives'].astype(str)\n",
    "trio_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "patients=trio_info.loc[trio_info['status']=='Patient', 'sample'].to_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "defaultdict(dict,\n",
       "            {77092483: {'child': 77092484},\n",
       "             77092484: {'father': 77092482, 'mother': 77092483},\n",
       "             77092482: {'child': 77092484},\n",
       "             77092481: {'child': 77092479},\n",
       "             77092480: {'child': 77092479},\n",
       "             77092479: {'father': 77092480, 'mother': 77092481},\n",
       "             77092478: {'child': 77092477},\n",
       "             77092477: {'mother': 77092478},\n",
       "             77092476: {'sibling': 77092477},\n",
       "             77092473: {'child': 77092472, 'spouse': 77092474},\n",
       "             77092472: {'mother': 77092473, 'father': 77092474},\n",
       "             77092474: {'child': 77092472, 'spouse': 77092473},\n",
       "             77092471: {'spouse': 77092470, 'child': 77092469},\n",
       "             77092469: {'mother': 77092471, 'father': 77092470},\n",
       "             77092470: {'spouse': 77092471, 'child': 77092469},\n",
       "             77092468: {'child': 77092467, 'spouse': 77092475},\n",
       "             77092475: {'spouse': 77092468, 'child': 77092467},\n",
       "             77092467: {'father': 77092475, 'mother': 77092468},\n",
       "             77092465: {'child': 77092464},\n",
       "             77092464: {'mother': 77092465, 'sibling': 77092466},\n",
       "             77092466: {'mother': 77092465, 'sibling': 77092464},\n",
       "             77092459: {'spouse': 77092460, 'child': 77092461},\n",
       "             77092460: {'child': 77092461, 'spouse': 77092459},\n",
       "             77092461: {'mother': 77092460, 'father': 77092459},\n",
       "             77092462: {'mother': 77092463},\n",
       "             77092463: {'child': 77092462},\n",
       "             77092457: {'mother': 77092458},\n",
       "             77092458: {'child': 77092457},\n",
       "             77092456: {'father': 77092455, 'mother': 77092454},\n",
       "             77092454: {'spouse': 77092455, 'child': 77092456},\n",
       "             77092455: {'child': 77092456, 'spouse': 77092454},\n",
       "             77092453: {'spouse': 77092451, 'child': 77092452},\n",
       "             77092451: {'spouse': 77092453, 'child': 77092452},\n",
       "             77092452: {'mother': 77092451, 'father': 77092453},\n",
       "             77092500: {'mother': 77092501, 'father': 77092502},\n",
       "             77092502: {'spouse': 77092501, 'child': 77092500},\n",
       "             77092501: {'spouse': 77092502, 'child': 77092500},\n",
       "             77092450: {'mother': 77092449, 'father': 77092448},\n",
       "             77092448: {'child': 77092450, 'spouse': 77092449},\n",
       "             77092449: {'child': 77092450, 'spouse': 77092448},\n",
       "             77092491: {'child': 77092492, 'spouse': 77092552},\n",
       "             77092492: {'mother': 77092491, 'father': 77092552},\n",
       "             77092447: {'child': 77092446, 'spouse': 77092445},\n",
       "             77092445: {'child': 77092446, 'spouse': 77092447},\n",
       "             77092446: {'father': 77092447, 'mother': 77092445},\n",
       "             77092493: {'mother': 77092494, 'father': 77092495},\n",
       "             77092495: {'child': 77092493, 'spouse': 77092494},\n",
       "             77092494: {'spouse': 77092495, 'child': 77092493},\n",
       "             77092499: {'spouse': 77092498, 'child': 77092497},\n",
       "             77092496: {'father': 77092498,\n",
       "              'mother': 77092499,\n",
       "              'sibling': 77092497},\n",
       "             77092497: {'father': 77092498,\n",
       "              'mother': 77092499,\n",
       "              'sibling': 77092496},\n",
       "             77092498: {'spouse': 77092499, 'child': 77092496},\n",
       "             77092444: {'spouse': 77092443, 'child': 77092442},\n",
       "             77092443: {'spouse': 77092444, 'child': 77092442},\n",
       "             77092442: {'mother': 77092444, 'father': 77092443},\n",
       "             77092441: {'spouse': 77092439, 'child': 77092440},\n",
       "             77092439: {'spouse': 77092441, 'child': 77092440},\n",
       "             77092440: {'father': 77092439, 'mother': 77092441},\n",
       "             77092434: {'child': 77092435},\n",
       "             77092435: {'mother': 77092434},\n",
       "             77092437: {'spouse': 77092438, 'child': 77092436},\n",
       "             77092438: {'spouse': 77092437, 'child': 77092436},\n",
       "             77092436: {'father': 77092438, 'mother': 77092437},\n",
       "             77092552: {'child': 77092492, 'spouse': 77092491},\n",
       "             77092551: {'mother': 77092550},\n",
       "             77092550: {'child': 77092551},\n",
       "             77092424: {'sibling': 77092425,\n",
       "              'father': 77092422,\n",
       "              'mother': 77092423},\n",
       "             77092423: {'spouse': 77092422, 'child': 77092425},\n",
       "             77092422: {'child': 77092425, 'spouse': 77092423},\n",
       "             77092425: {'sibling': 77092424,\n",
       "              'mother': 77092423,\n",
       "              'father': 77092422},\n",
       "             77092421: {'child': 77092548, 'spouse': 77092420},\n",
       "             77092420: {'spouse': 77092421, 'child': 77092548},\n",
       "             77092426: {'child': 77092418, 'spouse': 77092428},\n",
       "             77092428: {'spouse': 77092426, 'child': 77092418},\n",
       "             77092418: {'mother': 77092428, 'father': 77092426},\n",
       "             77092429: {'spouse': 77092432, 'child': 77092433},\n",
       "             77092433: {'father': 77092429, 'mother': 77092432},\n",
       "             77092432: {'child': 77092433, 'spouse': 77092429},\n",
       "             77092548: {'mother': 77092421,\n",
       "              'sibling': 77092419,\n",
       "              'father': 77092420},\n",
       "             77092419: {'mother': 77092421,\n",
       "              'sibling': 77092548,\n",
       "              'father': 77092420},\n",
       "             77092431: {'mother': 77092430},\n",
       "             77092430: {'child': 77092431},\n",
       "             77092385: {'child': 77092390},\n",
       "             77092390: {'sibling': 77092387, 'mother': 77092385},\n",
       "             77092387: {'sibling': 77092390, 'mother': 77092385},\n",
       "             77092412: {'child': 77092413, 'spouse': 77092414},\n",
       "             77092413: {'mother': 77092414, 'father': 77092412},\n",
       "             77092414: {'child': 77092413, 'spouse': 77092412},\n",
       "             77092415: {'spouse': 77092416, 'child': 77092417},\n",
       "             77092416: {'spouse': 77092415, 'child': 77092417},\n",
       "             77092417: {'mother': 77092416, 'father': 77092415},\n",
       "             77092368: {'spouse': 77092371, 'child': 77092370},\n",
       "             77092371: {'spouse': 77092368, 'child': 77092370},\n",
       "             77092370: {'father': 77092368, 'mother': 77092371},\n",
       "             77092388: {'mother': 77092391,\n",
       "              'sibling': 77092392,\n",
       "              'father': 77092389},\n",
       "             77092392: {'mother': 77092391,\n",
       "              'sibling': 77092388,\n",
       "              'father': 77092389},\n",
       "             77092391: {'child': 77092388, 'spouse': 77092389},\n",
       "             77092389: {'child': 77092388, 'spouse': 77092391},\n",
       "             77092411: {'sibling': 77092410,\n",
       "              'mother': 77092408,\n",
       "              'father': 77092409},\n",
       "             77092410: {'sibling': 77092411,\n",
       "              'mother': 77092408,\n",
       "              'father': 77092409},\n",
       "             77092386: {'child': 77092383, 'spouse': 77092384},\n",
       "             77092408: {'child': 77092411, 'spouse': 77092409},\n",
       "             77092384: {'child': 77092383, 'spouse': 77092386},\n",
       "             77092383: {'mother': 77092384, 'father': 77092386},\n",
       "             77092407: {'sibling': 77092410,\n",
       "              'mother': 77092408,\n",
       "              'father': 77092409},\n",
       "             77092409: {'child': 77092411, 'spouse': 77092408},\n",
       "             77092372: {'spouse': 77092373, 'child': 77092369},\n",
       "             77092373: {'spouse': 77092372, 'child': 77092369},\n",
       "             77092406: {'spouse': 77092404, 'child': 77092405},\n",
       "             77092369: {'mother': 77092373, 'father': 77092372},\n",
       "             77092404: {'spouse': 77092406, 'child': 77092405},\n",
       "             77092405: {'mother': 77092404, 'father': 77092406},\n",
       "             77092403: {'sibling': 77092401,\n",
       "              'father': 77092400,\n",
       "              'mother': 77092402},\n",
       "             77092401: {'father': 77092400,\n",
       "              'mother': 77092402,\n",
       "              'sibling': 77092403},\n",
       "             77092400: {'spouse': 77092402, 'child': 77092401},\n",
       "             77092394: {'child': 77092393000, 'spouse': 77092395000},\n",
       "             77092393: {'mother': 77092394, 'father': 77092395},\n",
       "             77092402: {'child': 77092401, 'spouse': 77092400},\n",
       "             77092395: {'spouse': 77092394, 'child': 77092393},\n",
       "             77092396: {'child': 77092398, 'spouse': 77092399},\n",
       "             77092397: {'mother': 77092396,\n",
       "              'sibling': 77092398,\n",
       "              'father': 77092399},\n",
       "             77092398: {'mother': 77092396,\n",
       "              'sibling': 77092397,\n",
       "              'father': 77092399},\n",
       "             77092399: {'spouse': 77092396, 'child': 77092398},\n",
       "             77092378: {'spouse': 77092379, 'child': 77092381},\n",
       "             77092379: {'spouse': 77092378, 'child': 77092381},\n",
       "             77092381: {'father': 77092378, 'mother': 77092379},\n",
       "             77092377: {'mother': 77092375, 'sibling': 77092376},\n",
       "             77092376: {'sibling': 77092377,\n",
       "              'mother': 77092375,\n",
       "              'father': 77092374},\n",
       "             77092375: {'child': 77092376, 'spouse': 77092374},\n",
       "             77092374: {'child': 77092376, 'spouse': 77092375},\n",
       "             77092647: {'mother': 77092619},\n",
       "             77092619: {'child': 77092647},\n",
       "             77092650: {'mother': 77092621},\n",
       "             77092621: {'child': 77092650},\n",
       "             77092622: {'spouse': 77092623, 'child': 77092649},\n",
       "             77092623: {'child': 77092649, 'spouse': 77092622},\n",
       "             77092649: {'mother': 77092623, 'father': 77092622},\n",
       "             77092624: {'spouse': 77092625, 'child': 77092652},\n",
       "             77092625: {'spouse': 77092624, 'child': 77092652},\n",
       "             77092652: {'mother': 77092625, 'father': 77092624},\n",
       "             77092626: {'child': 77092651},\n",
       "             77092651: {'mother': 77092626},\n",
       "             77092646: {'mother': 77092627},\n",
       "             77092627: {'child': 77092646},\n",
       "             77092620: {'mother': 77092648},\n",
       "             77092648: {'child': 77092620},\n",
       "             77092629: {'spouse': 77092630, 'child': 77092640},\n",
       "             77092630: {'child': 77092640, 'spouse': 77092629},\n",
       "             77092640: {'mother': 77092629, 'father': 77092630},\n",
       "             77092639: {'mother': 77092631},\n",
       "             77092631: {'child': 77092639},\n",
       "             77092644: {'mother': 77092633},\n",
       "             77092633: {'child': 77092644},\n",
       "             77092637: {'mother': 77092634},\n",
       "             77092634: {'child': 77092637},\n",
       "             77092635: {'spouse': 77092636, 'child': 77092641},\n",
       "             77092636: {'spouse': 77092635, 'child': 77092641},\n",
       "             77092641: {'father': 77092635, 'mother': 77092636},\n",
       "             77092547: {'mother': 77092546},\n",
       "             77092546: {'child': 77092547},\n",
       "             77092538: {'mother': 77092540, 'child': 77092539},\n",
       "             77092539: {'mother': 77092540, 'father': 77092538},\n",
       "             77092540: {'child': 77092539, 'spouse': 77092538},\n",
       "             77092543: {'mother': 77092542, 'father': 77092541},\n",
       "             77092541: {'spouse': 77092542, 'child': 77092543},\n",
       "             77092542: {'spouse': 77092541, 'child': 77092543},\n",
       "             77092545: {'mother': 77092544},\n",
       "             77092544: {'child': 77092545},\n",
       "             77092586: {'child': 77092585},\n",
       "             77092585: {'mother': 77092586},\n",
       "             77092588: {'child': 77092587},\n",
       "             77092587: {'mother': 77092588},\n",
       "             77092582: {'mother': 77092581},\n",
       "             77092581: {'child': 77092582},\n",
       "             77092595: {'mother': 77092591,\n",
       "              'father': 77092592,\n",
       "              'sibling': 77092594},\n",
       "             77092593: {'mother': 77092591,\n",
       "              'father': 77092592,\n",
       "              'sibling': 77092594},\n",
       "             77092594: {'mother': 77092591,\n",
       "              'father': 77092592,\n",
       "              'sibling': 77092595},\n",
       "             77092591: {'spouse': 77092592, 'child': 77092595},\n",
       "             77092592: {'spouse': 77092591, 'child': 77092595},\n",
       "             77092611: {'mother': 77092610},\n",
       "             77092610: {'child': 77092611},\n",
       "             77092613: {'mother': 77092612},\n",
       "             77092612: {'child': 77092613},\n",
       "             77092606: {'mother': 77092605},\n",
       "             77092605: {'child': 77092606},\n",
       "             77092597: {'father': 77092596},\n",
       "             77092596: {'father': 7709597},\n",
       "             77092602: {'spouse': 77092600, 'child': 77092601},\n",
       "             77092601: {'father': 77092600, 'mother': 77092602},\n",
       "             77092600: {'\\xa0child': 77092601, 'spouse': 77092602},\n",
       "             77092614: {'spouse': 77092615, 'child': 77092616},\n",
       "             77092616: {'father': 77092615, 'mother': 77092614},\n",
       "             77092615: {'spouse': 77092614, 'child': 77092616},\n",
       "             77092628: {'mother': 77092645},\n",
       "             77092645: {'child': 77092628},\n",
       "             77092638: {'mother': 77092632},\n",
       "             77092632: {'child': 77092638},\n",
       "             77092608: {'child': 77092609, 'spouse': 77092607},\n",
       "             77092609: {'mother': 77092608, 'father': 77092607},\n",
       "             77092607: {'spouse': 77092608, 'child': 77092609},\n",
       "             77092599: {'mother': 77092598},\n",
       "             77092598: {'child': 77092599},\n",
       "             77092604: {'mother': 77092603},\n",
       "             77092603: {'child': 77092604},\n",
       "             77092617: {'mother': 77092618},\n",
       "             77092618: {'child': 77092617},\n",
       "             77092584: {'mother': 77092583},\n",
       "             77092583: {'child': 77092584},\n",
       "             77092590: {'mother': 77092589},\n",
       "             77092589: {'child': 77092590},\n",
       "             77092380: {'племянник': 77092467,\n",
       "              'spouse sibling': 77092468,\n",
       "              'sibling': 77092475},\n",
       "             77092367: {'mother': 77092366},\n",
       "             77092366: {'child': 77092367},\n",
       "             77092365: {'child': 77092364, 'spouse': 77092363},\n",
       "             77092363: {'spouse': 77092365, 'child': 77092364},\n",
       "             77092364: {'father': 77092363, 'mother': 77092365},\n",
       "             77092359: {'child': 77092362,\n",
       "              'mother': 77092360,\n",
       "              'spouse': 77092361},\n",
       "             77092362: {'mother': 77092359,\n",
       "              'father': 77092361,\n",
       "              'бабушка по материнской линии': 77092360},\n",
       "             77092360: {'внучка': 77092362, 'child': 77092359},\n",
       "             77092361: {'child': 77092362, 'spouse': 77092359},\n",
       "             77092490: {'sibling': 77092489,\n",
       "              'mother': 77092487,\n",
       "              'father': 77092488},\n",
       "             77092489: {'sibling': 77092490,\n",
       "              'mother': 77092487,\n",
       "              'father': 77092488},\n",
       "             77092488: {'child': 77092490, 'spouse': 77092487},\n",
       "             77092487: {'child': 77092490, 'spouse': 77092488},\n",
       "             77092345: {'father': 77092344, 'mother': 77092347},\n",
       "             77092347: {'child': 77092345, 'spouse': 77092344},\n",
       "             77092344: {'spouse': 77092347, 'child': 77092345},\n",
       "             77092327: {'child': 77092329},\n",
       "             77092329: {'mother': 77092327},\n",
       "             77092357: {'spouse': 77092351, 'child': 77092352},\n",
       "             77092351: {'spouse': 77092357, 'child': 77092358},\n",
       "             77092358: {'sibling': 77092352,\n",
       "              'mother': 77092351,\n",
       "              'father': 77092357},\n",
       "             77092352: {'father': 77092357,\n",
       "              'mother': 77092351,\n",
       "              'sibling': 77092358},\n",
       "             77092355: {'father': 77092354, 'mother': 77092353},\n",
       "             77092353: {'child': 77092355, 'spouse': 77092354},\n",
       "             77092354: {'child': 77092355, 'spouse': 77092353},\n",
       "             77092348: {'child': 77092346, 'spouse': 77092356},\n",
       "             77092346: {'father': 77092348, 'mother': 77092356},\n",
       "             77092356: {'spouse': 77092348, 'child': 77092346},\n",
       "             77092382: {'spouse': 77092430, 'child': 77092431},\n",
       "             77092533: {'child': 77092532, 'spouse': 77092534},\n",
       "             77092535: {'spouse': 77092536, 'child': 77092537},\n",
       "             77092537: {'mother': 77092536, 'father': 77092535},\n",
       "             77092536: {'spouse': 77092535, 'child': 77092537},\n",
       "             77092559: {'child': 77092560},\n",
       "             77092560: {'mother': 77092559},\n",
       "             77092557: {'child': 77092558},\n",
       "             77092558: {'mother': 77092557},\n",
       "             77092561: {'child': 77092562},\n",
       "             77092562: {'mother': 77092561},\n",
       "             77092568: {'child': 77092569},\n",
       "             77092569: {'mother': 77092568},\n",
       "             77092566: {'child': 77092567},\n",
       "             77092567: {'mother': 77092566},\n",
       "             77092574: {'child': 77092575},\n",
       "             77092323: {'child': 77092314, 'spouse': 77092315},\n",
       "             77092315: {'spouse': 77092323, 'child': 77092316},\n",
       "             77092316: {'father': 77092315,\n",
       "              'mother': 77092323,\n",
       "              'sibling': 77092314},\n",
       "             77092314: {'sibling': 77092316,\n",
       "              'father': 77092315,\n",
       "              'mother': 77092323},\n",
       "             77092308: {'spouse': 77092309, 'child': 77092306},\n",
       "             77092307: {'father': 77092308,\n",
       "              'sibling': 77092310,\n",
       "              'mother': 77092309},\n",
       "             77092310: {'father': 77092308,\n",
       "              'sibling': 77092306,\n",
       "              'mother': 77092309},\n",
       "             77092306: {'father': 77092308,\n",
       "              'sibling': 77092310,\n",
       "              'mother': 77092309},\n",
       "             77092309: {'spouse': 77092308, 'child': 77092306},\n",
       "             77092576: {'mother': 77092574, 'sibling': 77092575},\n",
       "             77092575: {'mother': 77092574, 'sibling': 77092576},\n",
       "             77092731: {'mother': 77092732},\n",
       "             77092732: {'child': 77092731},\n",
       "             77092320: {'spouse': 77092319, 'child': 77092321},\n",
       "             77092319: {'spouse': 77092320, 'child': 77092321},\n",
       "             77092321: {'father': 77092319, 'mother': 77092320},\n",
       "             77092841: {'sibling': 77092839,\n",
       "              'spouse': 77092838,\n",
       "              'child': 77092846},\n",
       "             77092839: {'sibling': 77092841, 'child_сестры': 77092837},\n",
       "             77092838: {'spouse': 77092841, 'child': 77092846},\n",
       "             77092840: {'mother': 77092841,\n",
       "              'father': 77092838,\n",
       "              'sibling': 77092846},\n",
       "             77092837: {'mother': 77092841,\n",
       "              'father': 77092838,\n",
       "              'sibling': 77092846,\n",
       "              'sibling_матери': 77092839},\n",
       "             77092846: {'mother': 77092841,\n",
       "              'father': 77092838,\n",
       "              'sibling': 77092837},\n",
       "             77092733: {'father': 77092734},\n",
       "             77092734: {'child': 77092733},\n",
       "             77092570: {'child': 77092571},\n",
       "             77092571: {'mother': 77092570},\n",
       "             77092340: {'child': 77092337, 'spouse': 77092328},\n",
       "             77092338: {'mother': 77092340,\n",
       "              'father': 77092328,\n",
       "              'sibling': 77092337},\n",
       "             77092328: {'spouse': 77092340, 'child': 77092337},\n",
       "             77092337: {'father': 77092328,\n",
       "              'sibling': 77092338,\n",
       "              'mother': 77092340},\n",
       "             77092330: {'sibling': 77092332, 'mother': 77092343},\n",
       "             77092350: {'sibling': 77092332, 'mother': 77092343},\n",
       "             77092332: {'sibling': 77092330, 'mother': 77092343},\n",
       "             77092331: {'spouse': 77092335, 'child': 77092336},\n",
       "             77092335: {'spouse': 77092331, 'child': 77092336},\n",
       "             77092336: {'father': 77092335, 'mother': 77092331},\n",
       "             77092322: {'child': 77092324},\n",
       "             77092324: {'mother': 77092322},\n",
       "             77092339: {'child': 77092342, 'spouse': 77092349},\n",
       "             77092342: {'sibling': 77092341,\n",
       "              'mother': 77092339,\n",
       "              'father': 77092349},\n",
       "             77092341: {'sibling': 77092342,\n",
       "              'mother': 77092339,\n",
       "              'father': 77092349},\n",
       "             77092349: {'child': 77092342, 'spouse': 77092339},\n",
       "             77092300: {'child': 77092325, 'spouse': 77092301},\n",
       "             77092325: {'sibling': 77092848,\n",
       "              'father': 77092301,\n",
       "              'mother': 77092300},\n",
       "             77092301: {'child': 77092325, 'spouse': 77092300},\n",
       "             77092848: {'father': 77092301,\n",
       "              'sibling': 77092325,\n",
       "              'mother': 77092300},\n",
       "             77092311: {'child': 77092313},\n",
       "             77092312: {'mother': 77092311, 'sibling': 77092313},\n",
       "             77092313: {'sibling': 77092312, 'mother': 77092311},\n",
       "             77092819: {'child': 77092816},\n",
       "             77092816: {'mother': 77092819},\n",
       "             77092823: {'sibling': 77092827, 'father': 77092829},\n",
       "             77092831: {'sibling': 77092827, 'father': 77092829},\n",
       "             77092830: {'sibling': 77092827, 'father': 77092829},\n",
       "             77092828: {'sibling': 77092827, 'father': 77092829},\n",
       "             77092827: {'sibling': 77092828, 'father': 77092829},\n",
       "             77092829: {'child': 77092827},\n",
       "             77092748: {'mother': 77092749},\n",
       "             77092749: {'mother': 77092748},\n",
       "             77092751: {'mother': 77092750},\n",
       "             77092750: {'child': 77092751},\n",
       "             77092553: {'mother': 77092752},\n",
       "             77092752: {'child': 77092553},\n",
       "             77092740: {'child': 77092739},\n",
       "             77092739: {'mother': 77092740},\n",
       "             77092742: {'mother': 77092741},\n",
       "             77092741: {'child': 77092742},\n",
       "             77092817: {'child': 77092367, 'spouse': 77092366},\n",
       "             77092738: {'child': 77092736, 'spouse': 77092737},\n",
       "             77092736: {'father': 77092738, 'mother': 77092737},\n",
       "             77092737: {'child': 77092736, 'spouse': 77092738},\n",
       "             77092554: {'child': 77092556, 'spouse': 77092555},\n",
       "             77092556: {'mother': 77092555, 'father': 77092554},\n",
       "             77092555: {'child': 77092556, 'spouse': 77092554},\n",
       "             77092747: {'mother': 77092746},\n",
       "             77092746: {'child': 77092747},\n",
       "             77092578: {'spouse': 77092579, 'child': 77092577},\n",
       "             77092579: {'spouse': 77092578, 'child': 77092577},\n",
       "             77092580: {'mother': 77092578,\n",
       "              'father': 77092579,\n",
       "              'sibling': 77092577},\n",
       "             77092577: {'father': 77092579,\n",
       "              'sibling': 77092580,\n",
       "              'mother': 77092578},\n",
       "             77092572: {'mother': 77092573},\n",
       "             77092573: {'child': 77092572},\n",
       "             77092744: {'child': 77092745, 'spouse': 77092743},\n",
       "             77092745: {'mother': 77092744, 'father': 77092743},\n",
       "             77092743: {'child': 77092745, 'mother': 77092744},\n",
       "             77092565: {'child': 77092563},\n",
       "             77092564: {'sibling': 77092563, 'mother': 77092565},\n",
       "             77092563: {'sibling': 77092564, 'mother': 77092565},\n",
       "             77092655: {'mother': 77092654},\n",
       "             77092654: {'child': 77092655},\n",
       "             77092653: {'child': 77092735},\n",
       "             77092735: {'mother': 77092653},\n",
       "             77092333: {'spouse': 77092326, 'child': 77092334},\n",
       "             77092326: {'child': 77092334, 'spouse': 77092333},\n",
       "             77092334: {'mother': 77092333, 'father': 77092326},\n",
       "             77092843: {'sibling': 77092851, 'mother': 77092850},\n",
       "             77092850: {'child': 77092851, 'spouse': 77092849},\n",
       "             77092851: {'sibling': 77092843,\n",
       "              'mother': 77092850,\n",
       "              'father': 77092849},\n",
       "             77092849: {'child': 77092851, 'spouse': 77092850},\n",
       "             77092847: {'mother': 77092842,\n",
       "              'sibling': 77092852,\n",
       "              'father': 77092844},\n",
       "             77092845: {'sibling': 77092852, 'mother': 77092842},\n",
       "             77092842: {'child': 77092852, 'spouse': 77092844},\n",
       "             77092844: {'spouse': 77092842, 'child': 77092847},\n",
       "             77092852: {'mother': 77092842, 'sibling': 77092847},\n",
       "             77092343: {'child': 77092332},\n",
       "             77092524: {'father': 77092522, 'mother': 77092523},\n",
       "             77092523: {'spouse': 77092522, 'child': 77092524},\n",
       "             77092522: {'spouse': 77092523, 'child': 77092524},\n",
       "             77092525: {'mother': 77092526,\n",
       "              'father': 77092528,\n",
       "              'sibling': 77092527},\n",
       "             77092527: {'mother': 77092526,\n",
       "              'father': 77092528,\n",
       "              'sibling': 77092525},\n",
       "             77092526: {'child': 77092527, 'spouse': 77092528},\n",
       "             77092528: {'child': 77092527, 'spouse': 77092526},\n",
       "             77092707: {'child': 77092706},\n",
       "             77092706: {'father': 77092707},\n",
       "             77092709: {'sibling': 77092708, 'mother': 77092711},\n",
       "             77092711: {'child': 77092709},\n",
       "             77092710: {'sibling': 77092709, 'mother': 77092711},\n",
       "             77092708: {'sibling': 77092709, 'mother': 77092710},\n",
       "             77092715: {'child': 77092716},\n",
       "             77092716: {'mother': 77092715},\n",
       "             77092712: {'child': 77092714},\n",
       "             77092713: {'mother': 77092712, 'sibling': 77092714},\n",
       "             77092714: {'mother': 77092712, 'sibling': 77092713},\n",
       "             77092717: {'child': 77092718},\n",
       "             77092718: {'mother': 77092717},\n",
       "             77092532: {'father': 77092533, 'mother': 77092534},\n",
       "             77092534: {'spouse': 77092533, 'child': 77092532},\n",
       "             77092719: {'child': 77092720},\n",
       "             77092720: {'mother': 77092719},\n",
       "             77092722: {'sibling': 77092723, 'mother': 77092721},\n",
       "             77092723: {'sibling': 77092722, 'mother': 77092721},\n",
       "             77092721: {'child': 77092723},\n",
       "             77092724: {'child': 77092726},\n",
       "             77092726: {'sibling': 77092725, 'mother': 77092724},\n",
       "             77092725: {'sibling': 77092726, 'mother': 77092724},\n",
       "             77092728: {'mother': 77092727},\n",
       "             77092727: {'child': 77092728},\n",
       "             77092729: {'child': 77092730},\n",
       "             77092730: {'mother': 77092729},\n",
       "             77092832: {'child': 77092827, 'spouse': 77092829},\n",
       "             77092808: {'sibling': 77092811,\n",
       "              'father': 77092810,\n",
       "              'mother': 77092809},\n",
       "             77092812: {'sibling': 77092811,\n",
       "              'father': 77092810,\n",
       "              'mother': 77092809},\n",
       "             77092811: {'sibling': 77092808,\n",
       "              'father': 77092810,\n",
       "              'mother': 77092809},\n",
       "             77092810: {'sibling': 77092812,\n",
       "              'child': 77092811,\n",
       "              'spouse': 77092809},\n",
       "             77092809: {'sibling': 77092812,\n",
       "              'child': 77092811,\n",
       "              'spouse': 77092810},\n",
       "             77092696: {'mother': 77092695},\n",
       "             77092695: {'child': 77092696},\n",
       "             77092692: {'mother': 77092691},\n",
       "             77092691: {'child': 77092692},\n",
       "             77092836: {'sibling': 77092834,\n",
       "              'mother': 77092835,\n",
       "              'father': 77092833},\n",
       "             77092834: {'sibling': 77092836,\n",
       "              'mother': 77092835,\n",
       "              'father': 77092833},\n",
       "             77092835: {'child': 77092834, 'father': 77092833},\n",
       "             77092833: {'child': 77092834, 'mother': 77092835},\n",
       "             77092818: {'sibling': 77092820,\n",
       "              'father': 77092821,\n",
       "              'mother': 77092825},\n",
       "             77092821: {'child': 77092820, 'spouse': 77092825},\n",
       "             77092820: {'sibling': 77092818,\n",
       "              'father': 77092821,\n",
       "              'mother': 77092825},\n",
       "             77092825: {'child': 77092820, 'spouse': 77092821},\n",
       "             77092826: {'child': 77092824, 'spouse': 77092822},\n",
       "             77092822: {'child': 77092824, 'spouse': 77092826},\n",
       "             77092824: {'father': 77092822, 'mother': 77092826},\n",
       "             77092807: {'mother': 77092806},\n",
       "             77092806: {'child': 77092807},\n",
       "             77092815: {'spouse': 77092813, 'child': 77092814},\n",
       "             77092813: {'spouse': 77092815, 'child': 77092814},\n",
       "             77092814: {'father': 77092815, 'mother': 77092813},\n",
       "             77092702: {'spouse': 77092703, 'child': 77092705},\n",
       "             77092703: {'spouse': 77092702, 'child': 77092705},\n",
       "             77092783: {'child': 77092780, 'spouse': 77092781},\n",
       "             77092782: {'sibling': 77092780,\n",
       "              'father': 77092781,\n",
       "              'mother': 77092783},\n",
       "             77092784: {'sibling': 77092780,\n",
       "              'father': 77092781,\n",
       "              'mother': 77092783},\n",
       "             77092780: {'sibling': 77092782,\n",
       "              'father': 77092781,\n",
       "              'mother': 77092783},\n",
       "             77092781: {'child': 77092782, 'mother': 77092783},\n",
       "             77092687: {'child': 77092688},\n",
       "             77092688: {'mother': 77092687},\n",
       "             77092689: {'child': 77092690},\n",
       "             77092690: {'mother': 77092689},\n",
       "             77092788: {'spouse': 77092787, 'child': 77092785},\n",
       "             77092787: {'child': 77092785, 'spouse': 77092788},\n",
       "             77092786: {'father': 77092787,\n",
       "              'sibling': 77092785,\n",
       "              'mother': 77092788},\n",
       "             77092785: {'father': 77092787,\n",
       "              'sibling': 77092786,\n",
       "              'mother': 77092788},\n",
       "             77092704: {'father': 77092703, 'sibling': 77092705},\n",
       "             77092705: {'father': 77092703, 'sibling': 77092704},\n",
       "             77092698: {'spouse': 77092697, 'child': 77092701},\n",
       "             77092699: {'mother': 77092697,\n",
       "              'father': 77092698,\n",
       "              'sibling': 77092701},\n",
       "             77092700: {'mother': 77092697,\n",
       "              'father': 77092698,\n",
       "              'sibling': 77092701},\n",
       "             77092701: {'mother': 77092697,\n",
       "              'father': 77092698,\n",
       "              'sibling': 77092700},\n",
       "             77092680: {'sibling': 77092682,\n",
       "              'mother': 77092683,\n",
       "              'father': 77092684},\n",
       "             77092681: {'sibling': 77092682,\n",
       "              'mother': 77092683,\n",
       "              'father': 77092684},\n",
       "             77092682: {'sibling': 77092680,\n",
       "              'mother': 77092683,\n",
       "              'father': 77092684},\n",
       "             77092683: {'child': 77092680, 'spouse': 77092684},\n",
       "             77092684: {'child': 77092680, 'spouse': 77092683},\n",
       "             77092685: {'mother': 77092686},\n",
       "             77092686: {'child': 77092685},\n",
       "             77092694: {'child': 77092693},\n",
       "             77092693: {'mother': 77092694},\n",
       "             77092778: {'child': 77092779, 'spouse': 77092776},\n",
       "             77092776: {'child': 77092779, 'spouse': 77092778},\n",
       "             77092779: {'sibling': 77092777,\n",
       "              'father': 77092776,\n",
       "              'mother': 77092778},\n",
       "             77092777: {'sibling': 77092779,\n",
       "              'father': 77092776,\n",
       "              'mother': 77092778},\n",
       "             77092789: {'spouse': 77092790, 'child': 77092792},\n",
       "             77092790: {'child': 77092792, 'spouse': 77092789},\n",
       "             77092792: {'mother': 77092790, 'father': 77092789},\n",
       "             77092791: {'mother': 77092790,\n",
       "              'sibling': 77092792,\n",
       "              'father': 77092789},\n",
       "             77092795: {'spouse': 77092793, 'child': 77092794},\n",
       "             77092793: {'child': 77092794, 'spouse': 77092795},\n",
       "             77092794: {'father': 77092793, 'mother': 77092795},\n",
       "             77092796: {'sibling': 77095355,\n",
       "              'mother': 77095356,\n",
       "              'father': 77095359},\n",
       "             77092799: {'spouse': 77092797, 'child': 77092798},\n",
       "             77092797: {'spouse': 77092799, 'child': 77092798},\n",
       "             77092798: {'mother': 77092799, 'father': 77092797},\n",
       "             77092801: {'child': 77092804,\n",
       "              'sibling': 77092802,\n",
       "              'spouse': 77092800},\n",
       "             77092802: {'sibling': 77092804,\n",
       "              'mother': 77092801,\n",
       "              'father': 77092800},\n",
       "             77092697: {'spouse': 77092698, 'child': 77092701},\n",
       "             77092803: {'sibling': 77092802,\n",
       "              'mother': 77092801,\n",
       "              'father': 77092800},\n",
       "             77092804: {'sibling': 77092802,\n",
       "              'mother': 77092801,\n",
       "              'father': 77092800},\n",
       "             77092773: {'father': 77092774,\n",
       "              'mother': 77092775,\n",
       "              'sibling': 77092771},\n",
       "             77092772: {'father': 77092774,\n",
       "              'mother': 77092775,\n",
       "              'sibling': 77092771},\n",
       "             77092771: {'father': 77092774,\n",
       "              'mother': 77092775,\n",
       "              'sibling': 77092773},\n",
       "             77092775: {'spouse': 77092774, 'child': 77092771},\n",
       "             77092774: {'spouse': 77092775, 'child': 77092771},\n",
       "             77092769: {'spouse': 77092770, 'child': 77092767},\n",
       "             77092770: {'spouse': 77092769, 'child': 77092767},\n",
       "             77092767: {'father': 77092769,\n",
       "              'mother': 77092770,\n",
       "              'sibling': 77092768},\n",
       "             77092768: {'father': 77092769,\n",
       "              'mother': 77092770,\n",
       "              'sibling': 77092767},\n",
       "             77092902: {'mother': 77092901},\n",
       "             77092901: {'child': 77092902},\n",
       "             77092766: {'mother': 77092813,\n",
       "              'father': 77092815,\n",
       "              'sibling': 77092814},\n",
       "             77092758: {'mother': 77092757,\n",
       "              'sibling': 77092754,\n",
       "              'father': 77092756},\n",
       "             77092757: {'spouse': 77092756, 'child': 77092758},\n",
       "             77092756: {'child': 77092758, 'spouse': 77092757},\n",
       "             77092755: {'sibling': 77092754,\n",
       "              'mother': 77092757,\n",
       "              'свобдный sibling по матери': 77092758},\n",
       "             77092754: {'sibling': 77092755,\n",
       "              'mother': 77092757,\n",
       "              'сводный sibling по матери': 77092758},\n",
       "             77092761: {'mother': 77092760,\n",
       "              'сводный sibling по матери': 77092759},\n",
       "             77092760: {'child': 77092759},\n",
       "             77092759: {'mother': 77092760,\n",
       "              'сводный sibling по матери': 77092761},\n",
       "             77092764: {'mother': 77092762, 'sibling': 77092765},\n",
       "             77092765: {'mother': 77092762, 'sibling': 77092764},\n",
       "             77092763: {'mother': 77092762, 'sibling': 77092765},\n",
       "             77092762: {'child': 77092765},\n",
       "             77095399: {'child': 77095397, 'spouse': 77092753},\n",
       "             77095397: {'sibling': 77095398,\n",
       "              'mother': 77095399,\n",
       "              'father': 77092753},\n",
       "             77092753: {'child': 77095397, 'spouse': 77095399},\n",
       "             77095398: {'father': 77092753,\n",
       "              'sibling': 77095397,\n",
       "              'mother': 77095399},\n",
       "             77092665: {'child': 77092664},\n",
       "             77092664: {'mother': 77092665},\n",
       "             77092677: {'mother': 77092679, 'father': 77092678},\n",
       "             77092679: {'child': 77092677, 'spouse': 77092678},\n",
       "             77092678: {'child': 77092677, 'spouse': 77092679},\n",
       "             77095386: {'sibling': 77095387,\n",
       "              'father': 77095388,\n",
       "              'mother': 77095385},\n",
       "             77095387: {'sibling': 77095386,\n",
       "              'father': 77095388,\n",
       "              'mother': 77095385},\n",
       "             77095390: {'sibling': 77095389},\n",
       "             77095389: {'sibling': 77095390},\n",
       "             77095379: {'child': 77095380},\n",
       "             77095380: {'mother': 77095379},\n",
       "             77095382: {'father': 77095383,\n",
       "              'mother': 77095381,\n",
       "              'sibling': 77095384},\n",
       "             77095383: {'child': 77095382,\n",
       "              'mother': 77095381,\n",
       "              'sibling': 77095384},\n",
       "             77095385: {'child': 77095386, 'spouse': 77095388},\n",
       "             77095384: {'father': 77095383,\n",
       "              'mother': 77095381,\n",
       "              'sibling': 77095382},\n",
       "             77095381: {'spouse': 77095383,\n",
       "              'child': 77095384,\n",
       "              'sibling': 77095382},\n",
       "             77095388: {'child': 77095387, 'spouse': 77095385},\n",
       "             77095365: {'sibling': 77095367,\n",
       "              'mother': 77095366,\n",
       "              'father': 77095368},\n",
       "             77095347: {'child': 77095349, 'spouse': 77095350},\n",
       "             77095349: {'mother': 77095347,\n",
       "              'sibling': 77095354,\n",
       "              'father': 77095350},\n",
       "             77095354: {'mother': 77095347,\n",
       "              'sibling': 77095349,\n",
       "              'father': 77095350},\n",
       "             77095350: {'spouse': 77095347, 'child': 77095354},\n",
       "             77095364: {'mother': 77095366,\n",
       "              'father': 77095368,\n",
       "              'sibling': 77095367},\n",
       "             77095366: {'child': 77095367, 'spouse': 77095368},\n",
       "             77095368: {'spouse': 77095366, 'child': 77095367},\n",
       "             77095367: {'mother': 77095366,\n",
       "              'father': 77095368,\n",
       "              'sibling': 77095364},\n",
       "             77095370: {'child': 77095369, 'spouse': 77095372},\n",
       "             77095371: {'mother': 77095370,\n",
       "              'father': 77095372,\n",
       "              'sibling': 77095369},\n",
       "             77095369: {'mother': 77095370,\n",
       "              'father': 77095372,\n",
       "              'sibling': 77095371},\n",
       "             77095372: {'mother': 77095370, 'child': 77095369},\n",
       "             77095353: {'mother': 77095352, 'father': 77095351},\n",
       "             77095352: {'child': 77095353, 'spouse': 77095351},\n",
       "             77095351: {'child': 77095353, 'spouse': 77095352},\n",
       "             77095355: {'sibling': 77092796,\n",
       "              'mother': 77095356,\n",
       "              'father': 77095359},\n",
       "             77095359: {'spouse': 77095356, 'child': 77092796},\n",
       "             77095356: {'spouse': 77095359, 'child': 77092796},\n",
       "             77095344: {'father': 77095343, 'mother': 77095342},\n",
       "             77095342: {'spouse': 77095343, 'child': 77095344},\n",
       "             77095343: {'spouse': 77095342, 'child': 77095344},\n",
       "             77095336: {'sibling': 77095335,\n",
       "              'mother': 77095334,\n",
       "              'father': 77095340},\n",
       "             77095335: {'sibling': 77095336,\n",
       "              'mother': 77095334,\n",
       "              'father': 77095340},\n",
       "             77095334: {'child': 77095335, 'spouse': 77095340},\n",
       "             77095340: {'child': 77095335, 'spouse': 77095334},\n",
       "             77095346: {'sibling': 77095345},\n",
       "             77095345: {'sibling': 77095346},\n",
       "             77095338: {'sibling': 77095337,\n",
       "              'father': 77095339,\n",
       "              'mother': 77095341},\n",
       "             77095339: {'child': 77095337, 'spouse': 77095341},\n",
       "             77095341: {'child': 77095337, 'spouse': 77095339},\n",
       "             77095337: {'sibling': 77095338,\n",
       "              'father': 77095339,\n",
       "              'mother': 77095341},\n",
       "             77095360: {'sibling': 77095358,\n",
       "              'father': 77095361,\n",
       "              'mother': 77095363},\n",
       "             77095362: {'sibling': 77095358,\n",
       "              'father': 77095361,\n",
       "              'mother': 77095363},\n",
       "             77095357: {'sibling': 77095358,\n",
       "              'father': 77095361,\n",
       "              'mother': 77095363},\n",
       "             77095358: {'sibling': 77095357,\n",
       "              'father': 77095361,\n",
       "              'mother': 77095363},\n",
       "             77095361: {'child': 77095358, 'spouse': 77095363},\n",
       "             77095363: {'child': 77095358, 'spouse': 77095361},\n",
       "             77095376: {'child': 77095377, 'spouse': 77095378},\n",
       "             77095375: {'sibling': 77095377,\n",
       "              'mother': 77095376,\n",
       "              'father': 77095378},\n",
       "             77095377: {'sibling': 77095375,\n",
       "              'mother': 77095376,\n",
       "              'father': 77095378},\n",
       "             77095378: {'spouse': 77095376, 'child': 77095377},\n",
       "             77095374: {'child': 77095373},\n",
       "             77095373: {'mother': 77095374},\n",
       "             77092531: {'father': 77092530, 'mother': 77092529},\n",
       "             77092530: {'spouse': 77092529, 'child': 77092531},\n",
       "             77092529: {'spouse': 77092530, 'child': 77092531},\n",
       "             77095333: {'mother': 77092432,\n",
       "              'sibling': 77092433,\n",
       "              'father': 77092429},\n",
       "             77095330: {'spouse': 77095331, 'child': 77095332},\n",
       "             77095331: {'child': 77095332, 'spouse': 77095330},\n",
       "             77095332: {'mother': 77095330, 'father': 77095331},\n",
       "             77095328: {'spouse': 77095329, 'child': 77095348},\n",
       "             77095329: {'spouse': 77095328, 'child': 77095348},\n",
       "             77095348: {'father': 77095328,\n",
       "              'mother': 77095329,\n",
       "              'sibling': 77095327},\n",
       "             77095327: {'father': 77095328,\n",
       "              'mother': 77095329,\n",
       "              'sibling': 77095348},\n",
       "             77095326: {'father': 77095328,\n",
       "              'mother': 77095329,\n",
       "              'sibling': 77095348},\n",
       "             77092897: {'child': 77092898},\n",
       "             77092898: {'mother': 77092897},\n",
       "             77095322: {'child': 77095320, 'spouse': 77095321},\n",
       "             77095321: {'child': 77095320, 'spouse': 77095322},\n",
       "             77095320: {'mother': 77095321, 'father': 77095322},\n",
       "             77095325: {'child': 77095323},\n",
       "             77095324: {'father': 77095325, 'sibling': 77095323},\n",
       "             77095323: {'father': 77095325, 'sibling': 77095324},\n",
       "             77095319: {'spouse': 77095318, 'child': 77095316},\n",
       "             77095318: {'spouse': 77095319, 'child': 77095316},\n",
       "             77095317: {'father': 77095319,\n",
       "              'mother': 77095318,\n",
       "              'sibling': 77095316},\n",
       "             77095316: {'father': 77095319,\n",
       "              'mother': 77095318,\n",
       "              'sibling': 77095317},\n",
       "             77095313: {'spouse': 77095312, 'child': 77095311},\n",
       "             77095312: {'spouse': 77095313, 'child': 77095311},\n",
       "             77095311: {'mother': 77095313,\n",
       "              'father': 77095312,\n",
       "              'sibling': 77095315},\n",
       "             77095315: {'mother': 77095313,\n",
       "              'father': 77095312,\n",
       "              'sibling': 77095311},\n",
       "             77095314: {'mother': 77095313,\n",
       "              'father': 77095312,\n",
       "              'sibling': 77095311}})"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trios=defaultdict(dict)\n",
    "for idx, row in trio_info.iterrows():\n",
    "    if row['relatives']:\n",
    "        relatives=[x for x in row['relatives'].split('; ')]\n",
    "        for i in relatives:\n",
    "            person=i.split(':')\n",
    "            if len(person)>1:\n",
    "                trios[row['sample']][person[1]] = int(person[0])\n",
    "trios"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "files_list= pd.read_excel('/storage/haplotypes/resolve_conf.xlsx', index_col=0)\n",
    "calling_region = [32037643, 32041345]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_vcf(link, mode=None): \n",
    "    vcf = pd.read_csv(link, sep='\\t', comment='#', header=None,\n",
    "          names=['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT', 'SAMPLE'], index_col=False) \n",
    "    vcf = vcf.assign(GT=vcf['SAMPLE'].map(lambda x: x.split(':')[0]))\n",
    "    if mode == 'hc':\n",
    "        vcf = vcf.assign(AD=vcf['SAMPLE'].map(lambda x: x.split(':')[1]))\n",
    "        vcf['DP'] = vcf['AD'].map(lambda x: sum(map(int, x.split(','))) if pd.notna(x) else None)\n",
    "        vcf['DP'] = vcf['DP'].astype(int)\n",
    "        vcf['GT'] = vcf['GT'].str.replace('/', '|', regex=False)\n",
    "    if mode == 'dv':\n",
    "        vcf = vcf.assign(AD=vcf['SAMPLE'].map(lambda x: x.split(':')[3]))\n",
    "        vcf['DP'] = vcf['AD'].map(lambda x: sum(map(int, x.split(','))) if pd.notna(x) else None)\n",
    "        vcf['GT'] = vcf['GT'].str.replace('|', '/', regex=False)\n",
    "        vcf['DP'] = vcf['DP'].astype(int)\n",
    "    if mode == 'bam':\n",
    "        vcf = vcf.assign(AD=vcf['SAMPLE'].map(lambda x: x.split(':')[5]))\n",
    "        vcf['DP'] = vcf['AD'].map(lambda x: sum(map(int, x.split(','))) if pd.notna(x) else None)\n",
    "        vcf['DP'] = vcf['DP'].astype(int)\n",
    "    \n",
    "    return vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def gt_vectors(vcf, mode = None):\n",
    "    gt_vector={}\n",
    "    for idx, row in vcf.iterrows(): \n",
    "        if ',' in row['ALT']: \n",
    "            dp=int(row['AD'].split(',')[0])+int(row['AD'].split(',')[1]) + int(row['AD'].split(',')[2])\n",
    "            if row['POS'] == 32038855:\n",
    "                position = '32038855' + '_' + 'T' + \"_\" + 'TTTG'\n",
    "                if mode == 'hc':\n",
    "                    gt = '1|1'\n",
    "                else:\n",
    "                    gt='1/1'\n",
    "                vaf=(int(row['AD'].split(',')[1]) + int(row['AD'].split(',')[2]))/dp\n",
    "                gt_vector[position] = [gt, vaf, dp]\n",
    "            else:\n",
    "                for num, alt in enumerate(row['ALT'].split(',')):\n",
    "                    position = str(row['POS']) + '_' + row['REF'] + \"_\" + alt\n",
    "                    if mode == 'hc':\n",
    "                        gt = '1|0'\n",
    "                    else:\n",
    "                        gt='1/0'\n",
    "                    vaf=int(row['AD'].split(',')[num+1])/dp\n",
    "                    gt_vector[position] = [gt, vaf, row['DP']]\n",
    "                    \n",
    "        else: \n",
    "            position=str(row['POS']) + '_' + row['REF'] + \"_\" + row['ALT']\n",
    "            gt=row['GT']\n",
    "            vaf=int(row['AD'].split(',')[1])/row['DP']\n",
    "            gt_vector[position] = [gt, vaf, row['DP']]\n",
    "    return gt_vector"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#do I need DP checking here???\n",
    "def check_relative_variant(position, hc_gen_info, dv_gen_info):\n",
    "    print(f'checking relative variant with {hc_gen_info} AND {dv_gen_info}')\n",
    "    hc_genotype=hc_gen_info[0]\n",
    "    hc=hc_genotype.split('|')\n",
    "    hc=int(hc[0])+int(hc[1])\n",
    "\n",
    "    dv_genotype=dv_gen_info[0]\n",
    "    dv=dv_genotype.split('/')\n",
    "    dv=int(dv[0])+int(dv[1])\n",
    "\n",
    "    if hc==dv:\n",
    "        return hc_genotype\n",
    "    else:\n",
    "        if hc == 2:\n",
    "            if hc_gen_info[1]>=0.85:\n",
    "                if hc_gen_info[2] >= 30:\n",
    "                    print(f'hc right with {hc_genotype}, VAF - {hc_gen_info[1]}, DP - {hc_gen_info[2]}')\n",
    "                    return hc_genotype\n",
    "                else:\n",
    "                    print(f'hc call for {hc_genotype} with right VAF - {hc_gen_info[1]}, but low DP - {hc_gen_info[2]}')\n",
    "                    return None\n",
    "            if dv ==1 and (dv_gen_info[1] >= 0.2 and dv_gen_info[1] < 0.85):\n",
    "                if dv_gen_info[2] >= 30:\n",
    "                    print(f'dv right with {dv_genotype}, VAF - {dv_gen_info[1]}, DP - {dv_gen_info[2]}')\n",
    "                    return dv_genotype\n",
    "                else:\n",
    "                    print(f'dv call for {dv_genotype} with right VAF - {dv_gen_info[1]}, but low DP - {dv_gen_info[2]}')\n",
    "                    return None\n",
    "            else:\n",
    "                print(f'both callers have low qual variant in relative; HC: {hc_gen_info}, DV: {dv_gen_info}')\n",
    "                return None\n",
    "        if hc == 1:\n",
    "            if (hc_gen_info[1] >= 0.2 and hc_gen_info[1] < 0.85):\n",
    "                if hc_gen_info[2] >= 30:\n",
    "                    print(f'hc_right with {hc_genotype}, VAF - {hc_gen_info[1]}, DP - {hc_gen_info[2]}')\n",
    "                    return hc_genotype\n",
    "                else:\n",
    "                    print(f'hc call for {hc_genotype} with right VAF - {hc_gen_info[1]}, but low DP - {hc_gen_info[2]}')\n",
    "                    return None\n",
    "            if dv == 2 and dv_gen_info[1] >= 0.85:\n",
    "                if dv_gen_info[2] >= 30:\n",
    "                    print(f'dv right with {dv_genotype}, VAF - {dv_gen_info[1]}, DP - {dv_gen_info[2]}')\n",
    "                    return dv_genotype\n",
    "                else:\n",
    "                    print(f'dv call for {dv_genotype} with right VAF - {dv_gen_info[1]}, but low DP - {dv_gen_info[2]}')\n",
    "                    return None\n",
    "            else:\n",
    "                print(f'both callers have low qual variant in relative; HC: {hc_gen_info}, DV: {dv_gen_info}')\n",
    "                return None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_inheritance(patient, pos, trios, gt_hc, gt_dv, gt_bam=None):\n",
    "    found_in_both_parents=0\n",
    "    if trios[patient] != {}:\n",
    "        print(trios[patient])\n",
    "    else:\n",
    "        print(f'{patient} does not have relatives')\n",
    "    inheritance=[]\n",
    "    for rel in trios[patient].keys():\n",
    "        if rel == 'child' or rel == 'mother' or rel == 'father':\n",
    "            rel_num=int(trios[patient][rel])\n",
    "            try:\n",
    "                print(f'found {pos} in {rel_num} HC data - {gt_hc[rel_num][pos]}')\n",
    "            except KeyError:\n",
    "                print(f'{pos} not found in gt_hc of {rel_num} - the {rel} of {patient}')\n",
    "                try:\n",
    "                    print(f'found {pos} in {rel_num} DV data - {gt_dv[rel_num][pos]}')\n",
    "                except KeyError:\n",
    "                    print(f'{pos} not found in gt_dv of {rel_num} - the {rel} of {patient}')\n",
    "\n",
    "            if pos in gt_hc[rel_num]:\n",
    "                print(f'{pos} found in HC data of {rel}')\n",
    "                hc_genotype=gt_hc[rel_num][pos][0]\n",
    "                hc=hc_genotype.split('|')\n",
    "                hc=int(hc[0])+int(hc[1])\n",
    "                if hc == 2 and gt_hc[rel_num][pos][1]>=0.85:\n",
    "                    if gt_hc[rel_num][pos][2]>=30:\n",
    "                        print(f'found a homozygote {pos} in {rel} with VAF {gt_hc[rel_num][pos][1]}')\n",
    "                        inheritance.append(hc_genotype)\n",
    "                        if rel == 'mother' or rel == 'father':\n",
    "                            found_in_both_parents+=1\n",
    "                    else:\n",
    "                        print(f'found a lowqual homozygote {pos} in {rel} with right VAF {gt_hc[rel_num][pos][1]}, but low DP - {gt_hc[rel_num][pos][2]}')\n",
    "                        inheritance.append(hc_genotype)\n",
    "                        if rel == 'mother' or rel == 'father':\n",
    "                            found_in_both_parents+=1\n",
    "                if hc == 1 and (gt_hc[rel_num][pos][1]>=0.2 and gt_hc[rel_num][pos][1]<0.85):\n",
    "                    if gt_hc[rel_num][pos][2]>=30:\n",
    "                        print(f'found a heterozygote {pos} in {rel} with VAF {gt_hc[rel_num][pos][1]}')\n",
    "                        inheritance.append(hc_genotype)\n",
    "                        if rel == 'mother' or rel == 'father':\n",
    "                            found_in_both_parents+=1\n",
    "                    else:\n",
    "                        print(f'found a lowqual heterozygote {pos} in {rel} with right VAF {gt_hc[rel_num][pos][1]}, but low DP - {gt_hc[rel_num][pos][2]}')\n",
    "                        inheritance.append(hc_genotype)\n",
    "                        if rel == 'mother' or rel == 'father':\n",
    "                            found_in_both_parents+=1\n",
    "                else:\n",
    "                    print(f'found only low qual {pos} in {rel} with {gt_hc[rel_num][pos][1]} and {gt_hc[rel_num][pos][2]}, probably an artefact')\n",
    "            if pos in gt_dv[rel_num] and pos not in gt_hc[rel_num]:\n",
    "                print(f'{pos} found only in DV data of {rel}')\n",
    "                dv_genotype=gt_dv[rel_num][pos][0]\n",
    "                dv=dv_genotype.split('/')\n",
    "                dv=int(dv[0])+int(dv[1])\n",
    "                if pos in gt_bam[rel_num].keys():\n",
    "                    pp_value=gt_bam[rel_num][pos][0]\n",
    "                    pp=pp_value.split('/')\n",
    "                    pp=int(pp[0])+int(pp[1])\n",
    "                    if pp==2 and gt_bam[rel_num][pos][1]>=0.85:\n",
    "                        print(f'found a homozygote DV variant in {rel_num}')\n",
    "                        inheritance.append(pp_value)\n",
    "                        if rel == 'mother' or rel == 'father':\n",
    "                            found_in_both_parents+=1\n",
    "                    if pp==1 and (gt_bam[rel_num][pos][1]>=0.2 and gt_bam[rel_num][pos][1]<0.85):\n",
    "                        print(f'found a heterozygote DV variant in {rel_num}')\n",
    "                        inheritance.append(pp_value)\n",
    "                        if rel == 'mother' or rel == 'father':\n",
    "                            found_in_both_parents+=1\n",
    "                # if dv == 2 and gt_dv[rel_num][pos][1]>=0.85:\n",
    "                #     if gt_dv[rel_num][pos][2]>=30:\n",
    "                #         print(f'found a homozygote {pos} in {rel} with VAF {gt_dv[rel_num][pos][1]}')\n",
    "                #         inheritance.append(dv_genotype)\n",
    "                #     else:\n",
    "                #         print(f'found a lowqual homozygote {pos} in {rel} with right VAF {gt_dv[rel_num][pos][1]}, but low DP - {gt_dv[rel_num][pos][2]}')\n",
    "                #         inheritance.append(dv_genotype)\n",
    "                # if dv == 1 and (gt_dv[rel_num][pos][1]>=0.2 and gt_dv[rel_num][pos][1]<0.85):\n",
    "                #     if gt_dv[rel_num][pos][2]>=30:\n",
    "                #         print(f'found a heterozygote {pos} in {rel} with VAF {gt_dv[rel_num][pos][1]}')\n",
    "                #         inheritance.append(dv_genotype)\n",
    "                #     else:\n",
    "                #         print(f'found a lowqual heterozygote {pos} in {rel} with right VAF {gt_dv[rel_num][pos][1]}, but low DP - {gt_dv[rel_num][pos][2]}')\n",
    "                #         inheritance.append(dv_genotype)\n",
    "                else:\n",
    "                    print(f'cannot support found DV variant by bam, probably an artefact')\n",
    "\n",
    "            if pos not in gt_hc[rel_num] and pos not in gt_dv[rel_num]:\n",
    "                print(f'no {pos} found in {rel}')\n",
    "    \n",
    "    print(inheritance)\n",
    "    if len(inheritance)>0:\n",
    "        return True\n",
    "    else:\n",
    "        return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_inheritance(patient, pos, trios, gt_hc, gt_dv):\n",
    "    print(trios[patient])\n",
    "    inheritance=[]\n",
    "    for rel in trios[patient].keys():\n",
    "        if rel == 'child' or rel == 'mother' or rel == 'father':\n",
    "            rel_num=int(trios[patient][rel])\n",
    "            try:\n",
    "                print(f'found {pos} in {rel_num} HC data - {gt_hc[rel_num][pos]}')\n",
    "            except KeyError:\n",
    "                print(f'{pos} not found in gt_hc of {rel_num} - the {rel} of {patient}')\n",
    "                try:\n",
    "                    print(f'found {pos} in {rel_num} DV data - {gt_dv[rel_num][pos]}')\n",
    "                except KeyError:\n",
    "                    print(f'{pos} not found in gt_dv of {rel_num} - the {rel} of {patient}')\n",
    "            if pos in gt_hc[rel_num] and pos in gt_dv[rel_num]:\n",
    "                rel_gt_hc=gt_hc[rel_num][pos]\n",
    "                rel_gt_dv = gt_dv[rel_num][pos]\n",
    "                variant=check_relative_variant(pos, rel_gt_hc, rel_gt_dv)\n",
    "                if variant!=None:\n",
    "                    print(f'found {pos} in {rel} with genotype {variant}')\n",
    "                    inheritance.append(variant)\n",
    "            else:\n",
    "                if pos in gt_hc[rel_num]:\n",
    "                    print(f'{pos} found only in HC data of {rel}')\n",
    "                    hc_genotype=gt_hc[rel_num][pos][0]\n",
    "                    hc=hc_genotype.split('|')\n",
    "                    hc=int(hc[0])+int(hc[1])\n",
    "                    if hc == 2 and gt_hc[rel_num][pos][1]>=0.85:\n",
    "                        if gt_hc[rel_num][pos][2]>=30:\n",
    "                            print(f'found a homozygote {pos} in {rel} with VAF {gt_hc[rel_num][pos][1]}')\n",
    "                            inheritance.append(hc_genotype)\n",
    "                        else:\n",
    "                            print(f'found a lowqual homozygote {pos} in {rel} with right VAF {gt_hc[rel_num][pos][1]}, but low DP - {gt_hc[rel_num][pos][2]}')\n",
    "                            inheritance.append(hc_genotype)\n",
    "                    if hc == 1 and (gt_hc[rel_num][pos][1]>=0.2 and gt_hc[rel_num][pos][1]<0.85):\n",
    "                        if gt_hc[rel_num][pos][2]>=30:\n",
    "                            print(f'found a heterozygote {pos} in {rel} with VAF {gt_hc[rel_num][pos][1]}')\n",
    "                            inheritance.append(hc_genotype)\n",
    "                        else:\n",
    "                            print(f'found a lowqual heterozygote {pos} in {rel} with right VAF {gt_hc[rel_num][pos][1]}, but low DP - {gt_hc[rel_num][pos][2]}')\n",
    "                            inheritance.append(hc_genotype)\n",
    "                    else:\n",
    "                        print(f'found only low qual {pos} in {rel} with {gt_hc[rel_num][pos][1]} and {gt_hc[rel_num][pos][2]}, probably an artefact')\n",
    "                if pos in gt_dv[rel_num]:\n",
    "                    print(f'{pos} found only in DV data of {rel}')\n",
    "                    dv_genotype=gt_dv[rel_num][pos][0]\n",
    "                    dv=dv_genotype.split('/')\n",
    "                    dv=int(dv[0])+int(dv[1])\n",
    "                    if dv == 2 and gt_dv[rel_num][pos][1]>=0.85:\n",
    "                        if gt_dv[rel_num][pos][2]>=30:\n",
    "                            print(f'found a homozygote {pos} in {rel} with VAF {gt_dv[rel_num][pos][1]}')\n",
    "                            inheritance.append(dv_genotype)\n",
    "                        else:\n",
    "                            print(f'found a lowqual homozygote {pos} in {rel} with right VAF {gt_dv[rel_num][pos][1]}, but low DP - {gt_dv[rel_num][pos][2]}')\n",
    "                            inheritance.append(dv_genotype)\n",
    "                    if dv == 1 and (gt_dv[rel_num][pos][1]>=0.2 and gt_dv[rel_num][pos][1]<0.85):\n",
    "                        if gt_dv[rel_num][pos][2]>=30:\n",
    "                            print(f'found a heterozygote {pos} in {rel} with VAF {gt_dv[rel_num][pos][1]}')\n",
    "                            inheritance.append(dv_genotype)\n",
    "                        else:\n",
    "                            print(f'found a lowqual heterozygote {pos} in {rel} with right VAF {gt_dv[rel_num][pos][1]}, but low DP - {gt_dv[rel_num][pos][2]}')\n",
    "                            inheritance.append(dv_genotype)\n",
    "                    else:\n",
    "                        print(f'found only low qual {pos} in {rel} with {gt_dv[rel_num][pos][1]} and {gt_dv[rel_num][pos][2]}, probably an artefact')\n",
    "\n",
    "                if pos not in gt_hc[rel_num] and pos not in gt_dv[rel_num]:\n",
    "                    print(f'no {pos} found in {rel}')\n",
    "    \n",
    "    print(inheritance)\n",
    "    if len(inheritance)>0:\n",
    "        return True\n",
    "    else:\n",
    "        return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#add DP checking\n",
    "def check_variant(sample, pos, hc_genotype, dv_genotype, sample_hc_variants, sample_dv_variants, trios, gt_hc, gt_dv):\n",
    "    hc=hc_genotype.split('|')\n",
    "    hc=int(hc[0])+int(hc[1])\n",
    "    dv=dv_genotype.split('/')\n",
    "    dv=int(dv[0])+int(dv[1])\n",
    "    if hc==dv:\n",
    "        # print(f'{pos} are the same')\n",
    "        return hc_genotype\n",
    "    else:\n",
    "        print(f'for {idx} comparing {pos} with HC - {hc_genotype} and DV - {dv_genotype}')\n",
    "        if hc == 2:\n",
    "            if sample_hc_variants[pos][1]>=0.85 and sample_hc_variants[pos][2]>=30:\n",
    "                print(f'hc right with {hc_genotype}, VAF - {sample_hc_variants[pos][1]}, DP-{sample_hc_variants[pos][2]}')\n",
    "                return hc_genotype\n",
    "            if dv ==1 and (sample_dv_variants[pos][1] >= 0.2 and sample_dv_variants[pos][1] < 0.85) and sample_dv_variants[pos][2]>=30:\n",
    "                print(f'dv right with {dv_genotype}, VAF - {sample_dv_variants[pos][1]}, DP-{sample_dv_variants[pos][2]}')\n",
    "                return dv_genotype\n",
    "            else:\n",
    "                print(f'both wrong, checking inheritance; variant info: {hc_genotype}, {sample_hc_variants[pos][1]}, {sample_hc_variants[pos][2]} AND {dv_genotype}, {sample_dv_variants[pos][1]}, {sample_dv_variants[pos][2]}')\n",
    "                inheritance = check_inheritance(sample, pos, hc_genotype, dv_genotype, sample_hc_variants, sample_dv_variants, trios, gt_hc, gt_dv)\n",
    "                if inheritance == True:\n",
    "                    print('saving this variant even with bad QUAL')\n",
    "                    return hc_genotype\n",
    "                else:\n",
    "                    print('droping this variant, probably artefact')\n",
    "        if hc == 1:\n",
    "            if (sample_hc_variants[pos][1] >= 0.2 and sample_hc_variants[pos][1] < 0.85) and sample_hc_variants[pos][2]>=30:\n",
    "                print(f'hc right with {hc_genotype}, VAF - {sample_hc_variants[pos][1]}, DP-{sample_hc_variants[pos][2]}')\n",
    "                return hc_genotype\n",
    "            if dv == 2 and sample_dv_variants[pos][1] >= 0.85 and sample_dv_variants[pos][2]>=30:\n",
    "                print(f'dv right with {dv_genotype}, VAF - {sample_dv_variants[pos][1]}, DP-{sample_dv_variants[pos][2]}')\n",
    "                return dv_genotype\n",
    "            else:\n",
    "                print(f'both wrong, checking inheritance; variant info: {hc_genotype}, {sample_hc_variants[pos][1]}, {sample_hc_variants[pos][2]} AND {dv_genotype}, {sample_dv_variants[pos][1]}, {sample_dv_variants[pos][2]}')\n",
    "                inheritance = check_inheritance(sample, pos, trios, gt_hc, gt_dv)\n",
    "                if inheritance == True:\n",
    "                    print('saving this variant even with bad QUAL')\n",
    "                    return hc_genotype\n",
    "                else:\n",
    "                    print('droping this variant, probably artefact')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "gt_hc=defaultdict(dict)\n",
    "gt_dv=defaultdict(dict)\n",
    "gt_bam=defaultdict(dict)\n",
    "for idx, row in files_list.iterrows():\n",
    "    # dv\n",
    "    vcf = read_vcf(row['dv'], mode = 'dv')\n",
    "    vcf = vcf[vcf['FILTER'] == 'PASS']\n",
    "    vcf = vcf[(vcf['POS'] >= calling_region[0]) & (vcf['POS'] <= calling_region[1])]\n",
    "    gt_dv[idx]=gt_vectors(vcf, mode = 'dv')\n",
    "    # hc \n",
    "    vcf = read_vcf(row['hc'], mode='hc')\n",
    "    vcf = vcf[(vcf['POS'] >= calling_region[0]) & (vcf['POS'] <= calling_region[1])]\n",
    "    gt_hc[idx]=gt_vectors(vcf, mode= 'hc')\n",
    "    #bam\n",
    "    vcf = read_vcf(row['pileup'], mode='bam')\n",
    "    vcf = vcf[(vcf['POS'] >= calling_region[0]) & (vcf['POS'] <= calling_region[1])]\n",
    "    gt_bam[idx]=gt_vectors(vcf, mode= 'dv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "checking variants for 77050386\n",
      "checking variants for 77091048\n",
      "checking variants for 77092306\n",
      "checking variants for 77092307\n",
      "checking variants for 77092308\n",
      "checking variants for 77092309\n",
      "checking variants for 77092310\n",
      "checking variants for 77092311\n",
      "checking variants for 77092312\n",
      "32039081_C_G in 77092312 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.997539975399754, 1626]\n",
      "checking variants for 77092313\n",
      "checking variants for 77092314\n",
      "checking variants for 77092315\n",
      "checking variants for 77092316\n",
      "checking variants for 77092317\n",
      "checking variants for 77092319\n",
      "checking variants for 77092320\n",
      "checking variants for 77092321\n",
      "checking variants for 77092322\n",
      "checking variants for 77092323\n",
      "checking variants for 77092324\n",
      "checking variants for 77092327\n",
      "checking variants for 77092329\n",
      "checking variants for 77092344\n",
      "checking variants for 77092345\n",
      "checking variants for 77092346\n",
      "checking variants for 77092347\n",
      "checking variants for 77092348\n",
      "checking variants for 77092351\n",
      "checking variants for 77092352\n",
      "checking variants for 77092353\n",
      "checking variants for 77092354\n",
      "checking variants for 77092355\n",
      "checking variants for 77092356\n",
      "checking variants for 77092357\n",
      "checking variants for 77092358\n",
      "checking variants for 77092359\n",
      "checking variants for 77092360\n",
      "checking variants for 77092361\n",
      "checking variants for 77092362\n",
      "checking variants for 77092363\n",
      "checking variants for 77092364\n",
      "checking variants for 77092365\n",
      "checking variants for 77092366\n",
      "checking variants for 77092367\n",
      "checking variants for 77092368\n",
      "checking variants for 77092370\n",
      "check 32038949_T_G: HC - ['1|1', 1.0, 25], DV - 1/0, bam - ['1/0', 0.35555555555555557, 90]\n",
      "HC genotype is not supported by bam, saving a new genotype\n",
      "['1/0', 0.35555555555555557, 90]\n",
      "check 32038962_A_ATCC: HC - ['0|1', 0.8823529411764706, 17], DV - 1/1, Not found in bam\n",
      "need to check inheritance\n",
      "{'father': 77092368, 'mother': 77092371}\n",
      "found 32038962_A_ATCC in 77092368 HC data - ['1|0', 0.6018518518518519, 108]\n",
      "32038962_A_ATCC found in HC data of father\n",
      "found a heterozygote 32038962_A_ATCC in father with VAF 0.6018518518518519\n",
      "found 32038962_A_ATCC in 77092371 HC data - ['1|0', 0.5107340720221607, 2888]\n",
      "32038962_A_ATCC found in HC data of mother\n",
      "found a heterozygote 32038962_A_ATCC in mother with VAF 0.5107340720221607\n",
      "['1|0', '1|0']\n",
      "saving this variant even with bad QUAL\n",
      "77092370\n",
      "check 32039001_G_C: HC - ['1|1', 1.0, 13], DV - not called, bam - ['1/1', 0.9290322580645162, 155]\n",
      "genotype is supported by bam\n",
      "32038949_T_C in 77092370 found in DV, but not in HC, checking it\n",
      "32038949_T_C not called by HC, saving it with 1/0, ['1/0', 0.5666666666666667, 90]\n",
      "checking variants for 77092371\n",
      "checking variants for 77092374\n",
      "checking variants for 77092375\n",
      "checking variants for 77092376\n",
      "32038895_A_G in 77092376 found in DV, but not in HC, checking it\n",
      "32038895_A_G not called by HC, saving it with 0/1, ['0/1', 0.45, 680]\n",
      "checking variants for 77092377\n",
      "77092377\n",
      "check 32039006_G_A: HC - ['0|1', 0.16883116883116883, 77], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'mother': 77092375, 'sibling': 77092376}\n",
      "32039006_G_A not found in gt_hc of 77092375 - the mother of 77092377\n",
      "32039006_G_A not found in gt_dv of 77092375 - the mother of 77092377\n",
      "no 32039006_G_A found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "checking variants for 77092378\n",
      "checking variants for 77092379\n",
      "checking variants for 77092380\n",
      "checking variants for 77092381\n",
      "checking variants for 77092382\n",
      "checking variants for 77092383\n",
      "checking variants for 77092384\n",
      "77092384\n",
      "check 32039046_A_G: HC - ['1|0', 0.8666666666666667, 15], DV - not called, ['0/1', 0.5272727272727272, 165]\n",
      "HC genotype is supported by bam\n",
      "checking variants for 77092385\n",
      "checking variants for 77092386\n",
      "checking variants for 77092387\n",
      "checking variants for 77092388\n",
      "32038949_T_C in 77092388 found in DV, but not in HC, checking it\n",
      "32038949_T_C not called by HC, saving it with 1/0, ['1/0', 0.5176470588235295, 85]\n",
      "checking variants for 77092389\n",
      "checking variants for 77092390\n",
      "checking variants for 77092391\n",
      "checking variants for 77092392\n",
      "checking variants for 77092393\n",
      "77092393\n",
      "check 32039006_G_A: HC - ['1|0', 0.08333333333333333, 72], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'mother': 77092394, 'father': 77092395}\n",
      "32039006_G_A not found in gt_hc of 77092394 - the mother of 77092393\n",
      "32039006_G_A not found in gt_dv of 77092394 - the mother of 77092393\n",
      "no 32039006_G_A found in mother\n",
      "32039006_G_A not found in gt_hc of 77092395 - the father of 77092393\n",
      "32039006_G_A not found in gt_dv of 77092395 - the father of 77092393\n",
      "no 32039006_G_A found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "checking variants for 77092394\n",
      "checking variants for 77092395\n",
      "checking variants for 77092396\n",
      "checking variants for 77092397\n",
      "checking variants for 77092398\n",
      "77092398\n",
      "check 32038515_G_A: HC - ['1|0', 0.10059171597633136, 169], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'mother': 77092396, 'sibling': 77092397, 'father': 77092399}\n",
      "32038515_G_A not found in gt_hc of 77092396 - the mother of 77092398\n",
      "32038515_G_A not found in gt_dv of 77092396 - the mother of 77092398\n",
      "no 32038515_G_A found in mother\n",
      "32038515_G_A not found in gt_hc of 77092399 - the father of 77092398\n",
      "32038515_G_A not found in gt_dv of 77092399 - the father of 77092398\n",
      "no 32038515_G_A found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "checking variants for 77092399\n",
      "checking variants for 77092400\n",
      "checking variants for 77092401\n",
      "32038820_C_T in 77092401 found in DV, but not in HC, checking it\n",
      "32038820_C_T not called by HC, saving it with 0/1, ['0/1', 0.5477178423236515, 482]\n",
      "32038844_A_C in 77092401 found in DV, but not in HC, checking it\n",
      "32038844_A_C not called by HC, saving it with 1/1, ['1/1', 0.9639423076923077, 416]\n",
      "32038855_T_TTTG in 77092401 found in DV, but not in HC, checking it\n",
      "low qual 32038855_T_TTTG in 77092401, checking inheritance; variant info: 1/1\n",
      "{'father': 77092400, 'mother': 77092402, 'sibling': 77092403}\n",
      "found 32038855_T_TTTG in 77092400 HC data - ['0|1', 0.42424242424242425, 165]\n",
      "32038855_T_TTTG found in HC data of father\n",
      "found a heterozygote 32038855_T_TTTG in father with VAF 0.42424242424242425\n",
      "found 32038855_T_TTTG in 77092402 HC data - ['0|1', 0.41420118343195267, 169]\n",
      "32038855_T_TTTG found in HC data of mother\n",
      "found a heterozygote 32038855_T_TTTG in mother with VAF 0.41420118343195267\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038857_A_T in 77092401 found in DV, but not in HC, checking it\n",
      "32038857_A_T not called by HC, saving it with 1/1, ['1/1', 0.9815668202764977, 217]\n",
      "32038867_T_G in 77092401 found in DV, but not in HC, checking it\n",
      "32038867_T_G not called by HC, saving it with 1/1, ['1/1', 0.9822784810126582, 395]\n",
      "32038878_C_T in 77092401 found in DV, but not in HC, checking it\n",
      "32038878_C_T not called by HC, saving it with 1/1, ['1/1', 0.9975, 400]\n",
      "32038895_A_G in 77092401 found in DV, but not in HC, checking it\n",
      "32038895_A_G not called by HC, saving it with 1/1, ['1/1', 1.0, 378]\n",
      "32038949_T_C in 77092401 found in DV, but not in HC, checking it\n",
      "32038949_T_C not called by HC, saving it with 1/0, ['1/0', 0.4482758620689655, 87]\n",
      "checking variants for 77092402\n",
      "checking variants for 77092403\n",
      "checking variants for 77092404\n",
      "checking variants for 77092405\n",
      "checking variants for 77092406\n",
      "checking variants for 77092407\n",
      "checking variants for 77092408\n",
      "checking variants for 77092409\n",
      "checking variants for 77092410\n",
      "checking variants for 77092411\n",
      "checking variants for 77092412\n",
      "checking variants for 77092413\n",
      "checking variants for 77092414\n",
      "checking variants for 77092415\n",
      "checking variants for 77092416\n",
      "checking variants for 77092417\n",
      "32039081_C_G in 77092417 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.9924859110832811, 1597]\n",
      "checking variants for 77092418\n",
      "32039081_C_G in 77092418 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.9878190255220418, 1724]\n",
      "checking variants for 77092419\n",
      "checking variants for 77092420\n",
      "checking variants for 77092421\n",
      "checking variants for 77092422\n",
      "checking variants for 77092423\n",
      "checking variants for 77092424\n",
      "checking variants for 77092425\n",
      "checking variants for 77092426\n",
      "checking variants for 77092428\n",
      "checking variants for 77092430\n",
      "checking variants for 77092431\n",
      "checking variants for 77092434\n",
      "checking variants for 77092435\n",
      "checking variants for 77092436\n",
      "32039081_C_G in 77092436 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.9953947368421052, 1520]\n",
      "checking variants for 77092437\n",
      "77092437\n",
      "check 32038878_C_T: HC - ['0|1', 0.17341040462427745, 519], DV - not called, ['0/1', 0.470954356846473, 964]\n",
      "HC genotype is supported by bam\n",
      "77092437\n",
      "check 32038895_A_G: HC - ['0|1', 0.14285714285714285, 469], DV - not called, ['0/1', 0.46652267818574517, 926]\n",
      "HC genotype is supported by bam\n",
      "checking variants for 77092438\n",
      "checking variants for 77092439\n",
      "checking variants for 77092440\n",
      "checking variants for 77092441\n",
      "checking variants for 77092442\n",
      "checking variants for 77092443\n",
      "checking variants for 77092444\n",
      "checking variants for 77092445\n",
      "checking variants for 77092446\n",
      "checking variants for 77092447\n",
      "checking variants for 77092448\n",
      "checking variants for 77092449\n",
      "checking variants for 77092450\n",
      "checking variants for 77092451\n",
      "checking variants for 77092452\n",
      "checking variants for 77092453\n",
      "checking variants for 77092454\n",
      "checking variants for 77092455\n",
      "checking variants for 77092456\n",
      "checking variants for 77092457\n",
      "checking variants for 77092458\n",
      "checking variants for 77092459\n",
      "checking variants for 77092460\n",
      "checking variants for 77092461\n",
      "checking variants for 77092462\n",
      "checking variants for 77092463\n",
      "checking variants for 77092464\n",
      "checking variants for 77092465\n",
      "checking variants for 77092466\n",
      "checking variants for 77092467\n",
      "checking variants for 77092468\n",
      "checking variants for 77092469\n",
      "checking variants for 77092470\n",
      "checking variants for 77092471\n",
      "checking variants for 77092472\n",
      "checking variants for 77092473\n",
      "checking variants for 77092474\n",
      "checking variants for 77092475\n",
      "checking variants for 77092476\n",
      "checking variants for 77092477\n",
      "checking variants for 77092478\n",
      "checking variants for 77092479\n",
      "checking variants for 77092480\n",
      "checking variants for 77092481\n",
      "checking variants for 77092482\n",
      "checking variants for 77092483\n",
      "checking variants for 77092484\n",
      "32039081_C_G in 77092484 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.994413407821229, 1611]\n",
      "checking variants for 77092487\n",
      "checking variants for 77092488\n",
      "checking variants for 77092489\n",
      "checking variants for 77092490\n",
      "checking variants for 77092491\n",
      "checking variants for 77092492\n",
      "32039081_C_G in 77092492 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.9910501193317423, 1676]\n",
      "checking variants for 77092493\n",
      "checking variants for 77092494\n",
      "checking variants for 77092495\n",
      "checking variants for 77092496\n",
      "checking variants for 77092497\n",
      "checking variants for 77092498\n",
      "checking variants for 77092499\n",
      "checking variants for 77092500\n",
      "checking variants for 77092501\n",
      "checking variants for 77092502\n",
      "checking variants for 77092522\n",
      "checking variants for 77092523\n",
      "checking variants for 77092524\n",
      "checking variants for 77092525\n",
      "checking variants for 77092526\n",
      "checking variants for 77092527\n",
      "checking variants for 77092528\n",
      "checking variants for 77092529\n",
      "77092529\n",
      "check 32038903_A_G: HC - ['0|1', 0.12, 25], DV - not called, ['0/1', 0.44663677130044843, 1115]\n",
      "HC genotype is supported by bam\n",
      "77092529\n",
      "check 32038948_T_TC: HC - ['0|1', 0.18518518518518517, 27], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038948_T_TC not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32038948_T_TC in 77092531 DV data - ['1/1', 0.9016393442622951, 61]\n",
      "32038948_T_TC found only in DV data of child\n",
      "cannot support found DV variant by bam, probably an artefact\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092529\n",
      "check 32038949_T_G: HC - ['0|1', 0.17857142857142858, 28], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038949_T_G not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32038949_T_G in 77092531 DV data - ['1/1', 1.0, 55]\n",
      "32038949_T_G found only in DV data of child\n",
      "cannot support found DV variant by bam, probably an artefact\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092529\n",
      "check 32038955_A_T: HC - ['0|1', 0.16666666666666666, 30], DV - not called, ['0/1', 0.32908163265306123, 784]\n",
      "HC genotype is supported by bam\n",
      "77092529\n",
      "check 32038962_A_ATCC: HC - ['0|1', 0.19230769230769232, 26], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038962_A_ATCC not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32038962_A_ATCC in 77092531 DV data - ['1/1', 0.9333333333333333, 60]\n",
      "32038962_A_ATCC found only in DV data of child\n",
      "cannot support found DV variant by bam, probably an artefact\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092529\n",
      "check 32038971_C_A: HC - ['0|1', 0.19230769230769232, 26], DV - not called, ['0/1', 0.35544041450777203, 965]\n",
      "HC genotype is supported by bam\n",
      "77092529\n",
      "check 32038983_TG_T: HC - ['0|1', 0.125, 24], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038983_TG_T not found in gt_hc of 77092531 - the child of 77092529\n",
      "32038983_TG_T not found in gt_dv of 77092531 - the child of 77092529\n",
      "no 32038983_TG_T found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092529\n",
      "check 32038997_TGG_T: HC - ['0|1', 0.11538461538461539, 26], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038997_TGG_T not found in gt_hc of 77092531 - the child of 77092529\n",
      "32038997_TGG_T not found in gt_dv of 77092531 - the child of 77092529\n",
      "no 32038997_TGG_T found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092529\n",
      "check 32039001_G_C: HC - ['0|1', 0.125, 24], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32039001_G_C not found in gt_hc of 77092531 - the child of 77092529\n",
      "32039001_G_C not found in gt_dv of 77092531 - the child of 77092529\n",
      "no 32039001_G_C found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092529\n",
      "check 32039002_G_A: HC - ['0|1', 0.125, 24], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32039002_G_A not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32039002_G_A in 77092531 DV data - ['1/1', 0.9487179487179487, 39]\n",
      "32039002_G_A found only in DV data of child\n",
      "cannot support found DV variant by bam, probably an artefact\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "checking variants for 77092530\n",
      "77092530\n",
      "check 32038983_TG_T: HC - ['0|1', 0.1875, 32], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77092529, 'child': 77092531}\n",
      "32038983_TG_T not found in gt_hc of 77092531 - the child of 77092530\n",
      "32038983_TG_T not found in gt_dv of 77092531 - the child of 77092530\n",
      "no 32038983_TG_T found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "checking variants for 77092531\n",
      "check 32038139_A_G: HC - ['1|0', 0.8936170212765957, 47], DV - 1/1, ['1/1', 0.9912225705329154, 1595]\n",
      "HC genotype is not supported by bam, saving a new genotype\n",
      "['1/1', 0.9912225705329154, 1595]\n",
      "check 32038141_T_G: HC - ['1|0', 0.8958333333333334, 48], DV - 1/1, ['1/1', 0.9878493317132442, 1646]\n",
      "HC genotype is not supported by bam, saving a new genotype\n",
      "['1/1', 0.9878493317132442, 1646]\n",
      "check 32038844_A_C: HC - ['1|0', 0.38461538461538464, 13], DV - 1/1, ['1/1', 0.8616044090630741, 1633]\n",
      "HC genotype is not supported by bam, saving a new genotype\n",
      "['1/1', 0.8616044090630741, 1633]\n",
      "77092531\n",
      "check 32038847_A_ATTT: HC - ['1|1', 0.8, 10], DV - not called, bam - Not found in bam\n",
      "need to check inheritance\n",
      "{'father': 77092530, 'mother': 77092529}\n",
      "32038847_A_ATTT not found in gt_hc of 77092530 - the father of 77092531\n",
      "32038847_A_ATTT not found in gt_dv of 77092530 - the father of 77092531\n",
      "no 32038847_A_ATTT found in father\n",
      "32038847_A_ATTT not found in gt_hc of 77092529 - the mother of 77092531\n",
      "32038847_A_ATTT not found in gt_dv of 77092529 - the mother of 77092531\n",
      "no 32038847_A_ATTT found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092531\n",
      "check 32039081_C_A: HC - ['1|1', 1.0, 9], DV - not called, bam - Not found in bam\n",
      "need to check inheritance\n",
      "{'father': 77092530, 'mother': 77092529}\n",
      "found 32039081_C_A in 77092530 HC data - ['1|0', 0.5331325301204819, 332]\n",
      "32039081_C_A found in HC data of father\n",
      "found a heterozygote 32039081_C_A in father with VAF 0.5331325301204819\n",
      "found 32039081_C_A in 77092529 HC data - ['1|0', 0.8709677419354839, 93]\n",
      "32039081_C_A found in HC data of mother\n",
      "found only low qual 32039081_C_A in mother with 0.8709677419354839 and 93, probably an artefact\n",
      "['1|0']\n",
      "saving this variant even with bad QUAL\n",
      "32038855_T_TTTG in 77092531 found in DV, but not in HC, checking it\n",
      "low qual 32038855_T_TTTG in 77092531, checking inheritance; variant info: 1/1\n",
      "{'father': 77092530, 'mother': 77092529}\n",
      "found 32038855_T_TTTG in 77092530 HC data - ['0|1', 0.40540540540540543, 111]\n",
      "32038855_T_TTTG found in HC data of father\n",
      "found a heterozygote 32038855_T_TTTG in father with VAF 0.40540540540540543\n",
      "found 32038855_T_TTTG in 77092529 HC data - ['0|1', 0.3805309734513274, 113]\n",
      "32038855_T_TTTG found in HC data of mother\n",
      "found a heterozygote 32038855_T_TTTG in mother with VAF 0.3805309734513274\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038867_T_G in 77092531 found in DV, but not in HC, checking it\n",
      "32038867_T_G not called by HC, saving it with 1/1, ['1/1', 0.9936421435059037, 1101]\n",
      "32038895_A_G in 77092531 found in DV, but not in HC, checking it\n",
      "32038895_A_G not called by HC, saving it with 1/1, ['1/1', 0.9889807162534435, 1089]\n",
      "32038903_A_G in 77092531 found in DV, but not in HC, checking it\n",
      "32038903_A_G not called by HC, saving it with 1/1, ['1/1', 0.9932301740812379, 1034]\n",
      "32038911_A_G in 77092531 found in DV, but not in HC, checking it\n",
      "32038911_A_G not called by HC, saving it with 1/1, ['1/1', 0.9841740850642928, 1011]\n",
      "32038927_A_G in 77092531 found in DV, but not in HC, checking it\n",
      "32038927_A_G not called by HC, saving it with 1/1, ['1/1', 0.9926082365364308, 947]\n",
      "32038938_T_G in 77092531 found in DV, but not in HC, checking it\n",
      "32038938_T_G not called by HC, saving it with 1/1, ['1/1', 0.9907940161104718, 869]\n",
      "32038948_T_TC in 77092531 found in DV, but not in HC, checking it\n",
      "low qual 32038948_T_TC in 77092531, checking inheritance; variant info: 1/1\n",
      "{'father': 77092530, 'mother': 77092529}\n",
      "found 32038948_T_TC in 77092530 HC data - ['0|1', 0.4576271186440678, 59]\n",
      "32038948_T_TC found in HC data of father\n",
      "found a heterozygote 32038948_T_TC in father with VAF 0.4576271186440678\n",
      "found 32038948_T_TC in 77092529 HC data - ['0|1', 0.18518518518518517, 27]\n",
      "32038948_T_TC found in HC data of mother\n",
      "found only low qual 32038948_T_TC in mother with 0.18518518518518517 and 27, probably an artefact\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038949_T_G in 77092531 found in DV, but not in HC, checking it\n",
      "low qual 32038949_T_G in 77092531, checking inheritance; variant info: 1/1\n",
      "{'father': 77092530, 'mother': 77092529}\n",
      "found 32038949_T_G in 77092530 HC data - ['0|1', 0.5254237288135594, 59]\n",
      "32038949_T_G found in HC data of father\n",
      "found a heterozygote 32038949_T_G in father with VAF 0.5254237288135594\n",
      "found 32038949_T_G in 77092529 HC data - ['0|1', 0.17857142857142858, 28]\n",
      "32038949_T_G found in HC data of mother\n",
      "found only low qual 32038949_T_G in mother with 0.17857142857142858 and 28, probably an artefact\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038955_A_T in 77092531 found in DV, but not in HC, checking it\n",
      "32038955_A_T not called by HC, saving it with 1/1, ['1/1', 0.988394584139265, 517]\n",
      "32038962_A_ATCC in 77092531 found in DV, but not in HC, checking it\n",
      "low qual 32038962_A_ATCC in 77092531, checking inheritance; variant info: 1/1\n",
      "{'father': 77092530, 'mother': 77092529}\n",
      "found 32038962_A_ATCC in 77092530 HC data - ['0|1', 0.36585365853658536, 41]\n",
      "32038962_A_ATCC found in HC data of father\n",
      "found a heterozygote 32038962_A_ATCC in father with VAF 0.36585365853658536\n",
      "found 32038962_A_ATCC in 77092529 HC data - ['0|1', 0.19230769230769232, 26]\n",
      "32038962_A_ATCC found in HC data of mother\n",
      "found only low qual 32038962_A_ATCC in mother with 0.19230769230769232 and 26, probably an artefact\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038971_C_A in 77092531 found in DV, but not in HC, checking it\n",
      "32038971_C_A not called by HC, saving it with 1/1, ['1/1', 0.8599033816425121, 828]\n",
      "32039002_G_A in 77092531 found in DV, but not in HC, checking it\n",
      "low qual 32039002_G_A in 77092531, checking inheritance; variant info: 1/1\n",
      "{'father': 77092530, 'mother': 77092529}\n",
      "found 32039002_G_A in 77092530 HC data - ['0|1', 0.23255813953488372, 43]\n",
      "32039002_G_A found in HC data of father\n",
      "found a heterozygote 32039002_G_A in father with VAF 0.23255813953488372\n",
      "found 32039002_G_A in 77092529 HC data - ['0|1', 0.125, 24]\n",
      "32039002_G_A found in HC data of mother\n",
      "found only low qual 32039002_G_A in mother with 0.125 and 24, probably an artefact\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32039015_G_T in 77092531 found in DV, but not in HC, checking it\n",
      "32039015_G_T not called by HC, saving it with 1/1, ['1/1', 0.904487917146145, 869]\n",
      "32039027_C_A in 77092531 found in DV, but not in HC, checking it\n",
      "32039027_C_A not called by HC, saving it with 1/1, ['1/1', 0.9360967184801382, 1158]\n",
      "32039046_A_G in 77092531 found in DV, but not in HC, checking it\n",
      "32039046_A_G not called by HC, saving it with 1/1, ['1/1', 0.9910931174089069, 1235]\n",
      "32039050_G_T in 77092531 found in DV, but not in HC, checking it\n",
      "32039050_G_T not called by HC, saving it with 1/1, ['1/1', 0.9918962722852512, 1234]\n",
      "32039055_C_G in 77092531 found in DV, but not in HC, checking it\n",
      "32039055_C_G not called by HC, saving it with 1/1, ['1/1', 0.9903225806451613, 1240]\n",
      "32039081_C_G in 77092531 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.9939718687206965, 1493]\n",
      "checking variants for 77092532\n",
      "checking variants for 77092533\n",
      "checking variants for 77092534\n",
      "checking variants for 77092535\n",
      "checking variants for 77092536\n",
      "checking variants for 77092537\n",
      "checking variants for 77092538\n",
      "checking variants for 77092539\n",
      "checking variants for 77092540\n",
      "checking variants for 77092541\n",
      "checking variants for 77092542\n",
      "checking variants for 77092543\n",
      "checking variants for 77092544\n",
      "checking variants for 77092545\n",
      "checking variants for 77092546\n",
      "checking variants for 77092547\n",
      "checking variants for 77092548\n",
      "checking variants for 77092549\n",
      "checking variants for 77092550\n",
      "checking variants for 77092551\n",
      "checking variants for 77092552\n",
      "checking variants for 77092553\n",
      "checking variants for 77092554\n",
      "checking variants for 77092555\n",
      "checking variants for 77092556\n",
      "checking variants for 77092561\n",
      "checking variants for 77092562\n",
      "checking variants for 77092563\n",
      "checking variants for 77092564\n",
      "77092564\n",
      "check 32038948_T_TC: HC - ['0|1', 0.14285714285714285, 140], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32038948_T_TC not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32038948_T_TC not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32038948_T_TC found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092564\n",
      "check 32038949_T_G: HC - ['0|1', 0.14285714285714285, 140], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32038949_T_G not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32038949_T_G not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32038949_T_G found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092564\n",
      "check 32038955_A_T: HC - ['0|1', 0.14814814814814814, 135], DV - not called, ['0/1', 0.2916666666666667, 768]\n",
      "HC genotype is supported by bam\n",
      "77092564\n",
      "check 32038962_A_ATCC: HC - ['0|1', 0.08620689655172414, 58], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32038962_A_ATCC not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32038962_A_ATCC not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32038962_A_ATCC found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092564\n",
      "check 32038983_TG_T: HC - ['0|1', 0.1111111111111111, 36], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32038983_TG_T not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32038983_TG_T not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32038983_TG_T found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092564\n",
      "check 32038997_TGG_T: HC - ['0|1', 0.1282051282051282, 39], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32038997_TGG_T not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32038997_TGG_T not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32038997_TGG_T found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092564\n",
      "check 32039001_G_C: HC - ['0|1', 0.15151515151515152, 33], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32039001_G_C not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32039001_G_C not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32039001_G_C found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092564\n",
      "check 32039002_G_A: HC - ['0|1', 0.15151515151515152, 33], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32039002_G_A not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32039002_G_A not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32039002_G_A found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "32038903_A_G in 77092564 found in DV, but not in HC, checking it\n",
      "32038903_A_G not called by HC, saving it with 0/1, ['0/1', 0.44709626093874305, 1257]\n",
      "32038911_A_G in 77092564 found in DV, but not in HC, checking it\n",
      "32038911_A_G not called by HC, saving it with 0/1, ['0/1', 0.42502004811547717, 1247]\n",
      "32038938_T_G in 77092564 found in DV, but not in HC, checking it\n",
      "32038938_T_G not called by HC, saving it with 0/1, ['0/1', 0.4619826756496631, 1039]\n",
      "checking variants for 77092565\n",
      "checking variants for 77092566\n",
      "checking variants for 77092567\n",
      "checking variants for 77092568\n",
      "checking variants for 77092569\n",
      "checking variants for 77092570\n",
      "checking variants for 77092571\n",
      "checking variants for 77092572\n",
      "checking variants for 77092573\n",
      "checking variants for 77092574\n",
      "checking variants for 77092575\n",
      "checking variants for 77092576\n",
      "checking variants for 77092577\n",
      "checking variants for 77092578\n",
      "checking variants for 77092579\n",
      "checking variants for 77092580\n",
      "checking variants for 77092581\n",
      "checking variants for 77092582\n",
      "checking variants for 77092583\n",
      "checking variants for 77092584\n",
      "checking variants for 77092587\n",
      "checking variants for 77092588\n",
      "checking variants for 77092589\n",
      "checking variants for 77092590\n",
      "checking variants for 77092591\n",
      "checking variants for 77092592\n",
      "checking variants for 77092593\n",
      "32039081_C_G in 77092593 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.9974874371859297, 398]\n",
      "checking variants for 77092594\n",
      "32038878_C_T in 77092594 found in DV, but not in HC, checking it\n",
      "32038878_C_T not called by HC, saving it with 1/1, ['1/1', 0.9777777777777777, 360]\n",
      "32038895_A_G in 77092594 found in DV, but not in HC, checking it\n",
      "32038895_A_G not called by HC, saving it with 1/1, ['1/1', 0.993920972644377, 329]\n",
      "32038903_A_G in 77092594 found in DV, but not in HC, checking it\n",
      "32038903_A_G not called by HC, saving it with 1/1, ['1/1', 1.0, 331]\n",
      "32038949_T_C in 77092594 found in DV, but not in HC, checking it\n",
      "32038949_T_C not called by HC, saving it with 1/0, ['1/0', 0.46551724137931033, 58]\n",
      "checking variants for 77092595\n",
      "32038949_T_C in 77092595 found in DV, but not in HC, checking it\n",
      "32038949_T_C not called by HC, saving it with 1/0, ['1/0', 0.5, 96]\n",
      "checking variants for 77092596\n",
      "checking variants for 77092597\n",
      "checking variants for 77092598\n",
      "checking variants for 77092599\n",
      "checking variants for 77092600\n",
      "checking variants for 77092601\n",
      "checking variants for 77092602\n",
      "checking variants for 77092607\n",
      "checking variants for 77092608\n",
      "checking variants for 77092609\n",
      "checking variants for 77092610\n",
      "checking variants for 77092611\n",
      "checking variants for 77092612\n",
      "checking variants for 77092613\n",
      "checking variants for 77092614\n",
      "checking variants for 77092615\n",
      "checking variants for 77092616\n",
      "checking variants for 77092617\n",
      "checking variants for 77092618\n",
      "checking variants for 77092619\n",
      "checking variants for 77092620\n",
      "checking variants for 77092621\n",
      "checking variants for 77092622\n",
      "checking variants for 77092623\n",
      "checking variants for 77092624\n",
      "checking variants for 77092625\n",
      "checking variants for 77092626\n",
      "checking variants for 77092627\n",
      "checking variants for 77092628\n",
      "checking variants for 77092629\n",
      "checking variants for 77092630\n",
      "checking variants for 77092631\n",
      "checking variants for 77092632\n",
      "checking variants for 77092633\n",
      "77092633\n",
      "check 32039006_G_A: HC - ['0|1', 0.14705882352941177, 68], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'child': 77092644}\n",
      "32039006_G_A not found in gt_hc of 77092644 - the child of 77092633\n",
      "32039006_G_A not found in gt_dv of 77092644 - the child of 77092633\n",
      "no 32039006_G_A found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "checking variants for 77092634\n",
      "checking variants for 77092635\n",
      "77092635\n",
      "check 32039027_C_A: HC - ['0|1', 0.8571428571428571, 35], DV - not called, ['0/1', 0.5701754385964912, 342]\n",
      "HC genotype is supported by bam\n",
      "77092635\n",
      "check 32039046_A_G: HC - ['0|1', 0.8717948717948718, 39], DV - not called, ['0/1', 0.5794392523364486, 321]\n",
      "HC genotype is supported by bam\n",
      "checking variants for 77092636\n",
      "checking variants for 77092637\n",
      "checking variants for 77092638\n",
      "checking variants for 77092639\n",
      "checking variants for 77092640\n",
      "check 32038949_T_G: HC - ['1|1', 1.0, 24], DV - 1/0, bam - Not found in bam\n",
      "need to check inheritance\n",
      "{'mother': 77092629, 'father': 77092630}\n",
      "found 32038949_T_G in 77092629 HC data - ['0|1', 0.4861111111111111, 288]\n",
      "32038949_T_G found in HC data of mother\n",
      "found a heterozygote 32038949_T_G in mother with VAF 0.4861111111111111\n",
      "32038949_T_G not found in gt_hc of 77092630 - the father of 77092640\n",
      "32038949_T_G not found in gt_dv of 77092630 - the father of 77092640\n",
      "no 32038949_T_G found in father\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "77092640\n",
      "check 32039001_G_C: HC - ['1|1', 1.0, 27], DV - not called, bam - ['1/1', 0.95, 120]\n",
      "genotype is supported by bam\n",
      "32038949_T_C in 77092640 found in DV, but not in HC, checking it\n",
      "checking variants for 77092641\n",
      "checking variants for 77092644\n",
      "checking variants for 77092645\n",
      "checking variants for 77092646\n",
      "checking variants for 77092647\n",
      "checking variants for 77092648\n",
      "checking variants for 77092649\n",
      "checking variants for 77092650\n",
      "checking variants for 77092651\n",
      "checking variants for 77092652\n",
      "checking variants for 77092653\n",
      "checking variants for 77092654\n",
      "checking variants for 77092655\n",
      "checking variants for 77092657\n",
      "checking variants for 77092658\n",
      "checking variants for 77092659\n",
      "checking variants for 77092660\n",
      "checking variants for 77092661\n",
      "checking variants for 77092662\n",
      "checking variants for 77092663\n",
      "checking variants for 77092664\n",
      "checking variants for 77092665\n",
      "checking variants for 77092666\n",
      "77092666\n",
      "check 32038948_T_TC: HC - ['0|1', 0.14285714285714285, 14], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77092666 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092666\n",
      "check 32038949_T_G: HC - ['0|1', 0.14285714285714285, 14], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77092666 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092666\n",
      "check 32038955_A_T: HC - ['0|1', 0.125, 16], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77092666 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "checking variants for 77092667\n",
      "checking variants for 77092668\n",
      "checking variants for 77092669\n",
      "checking variants for 77092670\n",
      "checking variants for 77092671\n",
      "77092671\n",
      "check 32038962_A_ATCC: HC - ['0|1', 0.11678832116788321, 137], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77092671 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092671\n",
      "check 32038971_C_A: HC - ['0|1', 0.11594202898550725, 138], DV - not called, ['0/1', 0.3330739299610895, 1285]\n",
      "HC genotype is supported by bam\n",
      "32038895_A_G in 77092671 found in DV, but not in HC, checking it\n",
      "32038895_A_G not called by HC, saving it with 0/1, ['0/1', 0.39581881533101043, 1435]\n",
      "32038903_A_G in 77092671 found in DV, but not in HC, checking it\n",
      "32038903_A_G not called by HC, saving it with 0/1, ['0/1', 0.41710427606901723, 1333]\n",
      "32038911_A_G in 77092671 found in DV, but not in HC, checking it\n",
      "32038911_A_G not called by HC, saving it with 0/1, ['0/1', 0.3890160183066362, 1311]\n",
      "32038927_A_G in 77092671 found in DV, but not in HC, checking it\n",
      "32038927_A_G not called by HC, saving it with 0/1, ['0/1', 0.41051805337519626, 1274]\n",
      "32038938_T_G in 77092671 found in DV, but not in HC, checking it\n",
      "32038938_T_G not called by HC, saving it with 0/1, ['0/1', 0.4178628389154705, 1254]\n",
      "checking variants for 77092672\n",
      "checking variants for 77092673\n",
      "32038878_C_T in 77092673 found in DV, but not in HC, checking it\n",
      "32038878_C_T not called by HC, saving it with 1/1, ['1/1', 0.9970980847359258, 1723]\n",
      "checking variants for 77092674\n",
      "77092674\n",
      "check 32038539_C_CTTG: HC - ['1|0', 0.1694915254237288, 59], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77092674 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092674\n",
      "check 32038997_TGG_T: HC - ['0|1', 0.15873015873015872, 63], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77092674 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092674\n",
      "check 32039001_G_C: HC - ['0|1', 0.15517241379310345, 58], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77092674 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092674\n",
      "check 32039002_G_A: HC - ['0|1', 0.15789473684210525, 57], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77092674 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092674\n",
      "check 32040112_G_GCA: HC - ['0|1', 0.12643678160919541, 174], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77092674 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "32038844_A_C in 77092674 found in DV, but not in HC, checking it\n",
      "32038844_A_C not called by HC, saving it with 0/1, ['0/1', 0.39467592592592593, 1728]\n",
      "32038855_T_TTTG in 77092674 found in DV, but not in HC, checking it\n",
      "low qual 32038855_T_TTTG in 77092674, checking inheritance; variant info: 0/1\n",
      "77092674 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "32038857_A_T in 77092674 found in DV, but not in HC, checking it\n",
      "low qual 32038857_A_T in 77092674, checking inheritance; variant info: 0/1\n",
      "77092674 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "32038867_T_G in 77092674 found in DV, but not in HC, checking it\n",
      "32038867_T_G not called by HC, saving it with 0/1, ['0/1', 0.42654364797728883, 1409]\n",
      "32038878_C_T in 77092674 found in DV, but not in HC, checking it\n",
      "32038878_C_T not called by HC, saving it with 0/1, ['0/1', 0.4298245614035088, 1368]\n",
      "32038895_A_G in 77092674 found in DV, but not in HC, checking it\n",
      "32038895_A_G not called by HC, saving it with 0/1, ['0/1', 0.42243975903614456, 1328]\n",
      "checking variants for 77092675\n",
      "checking variants for 77092676\n",
      "checking variants for 77092677\n",
      "checking variants for 77092678\n",
      "checking variants for 77092679\n",
      "checking variants for 77092687\n",
      "checking variants for 77092688\n",
      "checking variants for 77092689\n",
      "checking variants for 77092690\n",
      "checking variants for 77092691\n",
      "checking variants for 77092692\n",
      "checking variants for 77092693\n",
      "checking variants for 77092694\n",
      "checking variants for 77092695\n",
      "checking variants for 77092696\n",
      "checking variants for 77092697\n",
      "checking variants for 77092702\n",
      "checking variants for 77092703\n",
      "checking variants for 77092704\n",
      "checking variants for 77092705\n",
      "32039081_C_G in 77092705 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.9977285633162976, 1761]\n",
      "checking variants for 77092706\n",
      "checking variants for 77092707\n",
      "checking variants for 77092708\n",
      "checking variants for 77092712\n",
      "checking variants for 77092719\n",
      "checking variants for 77092720\n",
      "checking variants for 77092721\n",
      "checking variants for 77092722\n",
      "checking variants for 77092723\n",
      "checking variants for 77092725\n",
      "checking variants for 77092727\n",
      "checking variants for 77092728\n",
      "checking variants for 77092731\n",
      "checking variants for 77092732\n",
      "checking variants for 77092733\n",
      "checking variants for 77092734\n",
      "checking variants for 77092735\n",
      "checking variants for 77092736\n",
      "checking variants for 77092737\n",
      "checking variants for 77092738\n",
      "checking variants for 77092739\n",
      "checking variants for 77092740\n",
      "checking variants for 77092741\n",
      "checking variants for 77092742\n",
      "checking variants for 77092743\n",
      "checking variants for 77092744\n",
      "checking variants for 77092745\n",
      "checking variants for 77092746\n",
      "checking variants for 77092747\n",
      "checking variants for 77092748\n",
      "checking variants for 77092749\n",
      "checking variants for 77092750\n",
      "checking variants for 77092751\n",
      "checking variants for 77092752\n",
      "checking variants for 77092754\n",
      "checking variants for 77092755\n",
      "checking variants for 77092756\n",
      "checking variants for 77092757\n",
      "checking variants for 77092758\n",
      "32039081_C_G in 77092758 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 1.0, 928]\n",
      "checking variants for 77092759\n",
      "checking variants for 77092760\n",
      "checking variants for 77092761\n",
      "checking variants for 77092762\n",
      "checking variants for 77092763\n",
      "checking variants for 77092764\n",
      "checking variants for 77092765\n",
      "checking variants for 77092767\n",
      "checking variants for 77092768\n",
      "checking variants for 77092769\n",
      "checking variants for 77092770\n",
      "checking variants for 77092771\n",
      "checking variants for 77092772\n",
      "checking variants for 77092773\n",
      "checking variants for 77092774\n",
      "checking variants for 77092775\n",
      "checking variants for 77092776\n",
      "checking variants for 77092777\n",
      "checking variants for 77092778\n",
      "checking variants for 77092779\n",
      "checking variants for 77092780\n",
      "32038949_T_C in 77092780 found in DV, but not in HC, checking it\n",
      "32038949_T_C not called by HC, saving it with 1/0, ['1/0', 0.36324786324786323, 234]\n",
      "32039081_C_G in 77092780 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.9982378854625551, 1135]\n",
      "checking variants for 77092781\n",
      "checking variants for 77092782\n",
      "32039081_C_G in 77092782 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.998262380538662, 1151]\n",
      "checking variants for 77092783\n",
      "checking variants for 77092784\n",
      "checking variants for 77092785\n",
      "checking variants for 77092787\n",
      "checking variants for 77092789\n",
      "checking variants for 77092790\n",
      "checking variants for 77092791\n",
      "checking variants for 77092792\n",
      "checking variants for 77092793\n",
      "checking variants for 77092794\n",
      "checking variants for 77092795\n",
      "checking variants for 77092797\n",
      "checking variants for 77092798\n",
      "checking variants for 77092799\n",
      "checking variants for 77092804\n",
      "checking variants for 77092808\n",
      "checking variants for 77092809\n",
      "checking variants for 77092810\n",
      "checking variants for 77092811\n",
      "checking variants for 77092812\n",
      "checking variants for 77092816\n",
      "checking variants for 77092817\n",
      "checking variants for 77092819\n",
      "checking variants for 77092823\n",
      "checking variants for 77092827\n",
      "checking variants for 77092828\n",
      "checking variants for 77092829\n",
      "checking variants for 77092830\n",
      "checking variants for 77092831\n",
      "checking variants for 77092832\n",
      "checking variants for 77092835\n",
      "checking variants for 77092837\n",
      "checking variants for 77092838\n",
      "checking variants for 77092839\n",
      "checking variants for 77092840\n",
      "checking variants for 77092841\n",
      "checking variants for 77092843\n",
      "checking variants for 77092846\n",
      "checking variants for 77092849\n",
      "checking variants for 77092850\n",
      "checking variants for 77092851\n",
      "checking variants for 77092896\n",
      "77092896\n",
      "check 32039007_A_AAG: HC - ['0|1', 0.1935483870967742, 31], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77092896 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "checking variants for 77092897\n",
      "checking variants for 77092898\n",
      "check 32038844_A_C: HC - ['1|0', 0.45454545454545453, 22], DV - 1/1, ['1/1', 0.8655083655083655, 1554]\n",
      "HC genotype is not supported by bam, saving a new genotype\n",
      "['1/1', 0.8655083655083655, 1554]\n",
      "check 32038855_T_TTTG: HC - ['1|0', 0.45454545454545453, 22], DV - 1/1, Not found in bam\n",
      "need to check inheritance\n",
      "{'mother': 77092897}\n",
      "found 32038855_T_TTTG in 77092897 HC data - ['0|1', 0.576271186440678, 59]\n",
      "32038855_T_TTTG found in HC data of mother\n",
      "found a heterozygote 32038855_T_TTTG in mother with VAF 0.576271186440678\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "check 32038857_A_T: HC - ['1|0', 0.7692307692307693, 13], DV - 1/1, ['1/1', 0.9485294117647058, 544]\n",
      "HC genotype is not supported by bam, saving a new genotype\n",
      "['1/1', 0.9485294117647058, 544]\n",
      "77092898\n",
      "check 32039001_G_C: HC - ['1|1', 1.0, 20], DV - not called, bam - ['1/1', 0.9280575539568345, 278]\n",
      "genotype is supported by bam\n",
      "77092898\n",
      "check 32039007_A_AAG: HC - ['1|1', 1.0, 15], DV - not called, bam - Not found in bam\n",
      "need to check inheritance\n",
      "{'mother': 77092897}\n",
      "found 32039007_A_AAG in 77092897 HC data - ['0|1', 0.6, 15]\n",
      "32039007_A_AAG found in HC data of mother\n",
      "found a lowqual heterozygote 32039007_A_AAG in mother with right VAF 0.6, but low DP - 15\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038867_T_G in 77092898 found in DV, but not in HC, checking it\n",
      "32038867_T_G not called by HC, saving it with 1/1, ['1/1', 0.9952076677316294, 1252]\n",
      "32038878_C_T in 77092898 found in DV, but not in HC, checking it\n",
      "32038878_C_T not called by HC, saving it with 1/1, ['1/1', 0.9951923076923077, 1248]\n",
      "32038895_A_G in 77092898 found in DV, but not in HC, checking it\n",
      "32038895_A_G not called by HC, saving it with 1/1, ['1/1', 0.9974619289340102, 1182]\n",
      "32038903_A_G in 77092898 found in DV, but not in HC, checking it\n",
      "32038903_A_G not called by HC, saving it with 1/1, ['1/1', 0.9982062780269059, 1115]\n",
      "checking variants for 77092899\n",
      "checking variants for 77092900\n",
      "77092900\n",
      "check 32038962_A_ATCC: HC - ['1|0', 0.07246376811594203, 138], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77092900 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77092900\n",
      "check 32038971_C_A: HC - ['1|0', 0.10416666666666667, 96], DV - not called, ['0/1', 0.380997177798683, 1063]\n",
      "HC genotype is supported by bam\n",
      "77092900\n",
      "check 32039050_G_T: HC - ['1|0', 0.16666666666666666, 12], DV - not called, ['0/1', 0.40584415584415584, 1232]\n",
      "HC genotype is supported by bam\n",
      "77092900\n",
      "check 32039056_A_G: HC - ['1|0', 0.16666666666666666, 12], DV - not called, ['0/1', 0.38384615384615384, 1300]\n",
      "HC genotype is supported by bam\n",
      "32038938_T_G in 77092900 found in DV, but not in HC, checking it\n",
      "32038938_T_G not called by HC, saving it with 0/1, ['0/1', 0.43699731903485256, 1119]\n",
      "checking variants for 77092901\n",
      "checking variants for 77092902\n",
      "checking variants for 77093060\n",
      "checking variants for 77093072\n",
      "checking variants for 77093073\n",
      "checking variants for 77093074\n",
      "checking variants for 77093075\n",
      "checking variants for 77093076\n",
      "checking variants for 77093077\n",
      "checking variants for 77093109\n",
      "checking variants for 77093110\n",
      "checking variants for 77093111\n",
      "checking variants for 77093137\n",
      "checking variants for 77093138\n",
      "checking variants for 77093139\n",
      "checking variants for 77093142\n",
      "checking variants for 77093143\n",
      "checking variants for 77093144\n",
      "checking variants for 77093145\n",
      "checking variants for 77093146\n",
      "checking variants for 77093147\n",
      "checking variants for 77093148\n",
      "checking variants for 77093149\n",
      "checking variants for 77093150\n",
      "checking variants for 77093153\n",
      "checking variants for 77093155\n",
      "checking variants for 77093156\n",
      "checking variants for 77093157\n",
      "checking variants for 77093158\n",
      "checking variants for 77093192\n",
      "checking variants for 77093202\n",
      "checking variants for 77093203\n",
      "checking variants for 77093204\n",
      "checking variants for 77093207\n",
      "checking variants for 77093216\n",
      "checking variants for 77093217\n",
      "checking variants for 77093249\n",
      "checking variants for 77093250\n",
      "checking variants for 77093251\n",
      "checking variants for 77093257\n",
      "checking variants for 77093258\n",
      "checking variants for 77093259\n",
      "checking variants for 77093451\n",
      "checking variants for 77093452\n",
      "checking variants for 77093453\n",
      "checking variants for 77093459\n",
      "checking variants for 77093476\n",
      "checking variants for 77093477\n",
      "checking variants for 77093480\n",
      "checking variants for 77093483\n",
      "checking variants for 77093491\n",
      "checking variants for 77093492\n",
      "checking variants for 77093493\n",
      "checking variants for 77093494\n",
      "checking variants for 77093496\n",
      "checking variants for 77093619\n",
      "checking variants for 77093620\n",
      "checking variants for 77093621\n",
      "checking variants for 77093640\n",
      "checking variants for 77093641\n",
      "checking variants for 77093643\n",
      "checking variants for 77093646\n",
      "checking variants for 77093647\n",
      "checking variants for 77093648\n",
      "checking variants for 77095100\n",
      "77095100\n",
      "check 32038878_C_T: HC - ['0|1', 0.12365591397849462, 186], DV - not called, ['0/1', 0.4177708495713172, 1283]\n",
      "HC genotype is supported by bam\n",
      "77095100\n",
      "check 32038971_C_A: HC - ['0|1', 0.19047619047619047, 42], DV - not called, ['0/1', 0.33186813186813185, 910]\n",
      "HC genotype is supported by bam\n",
      "77095100\n",
      "check 32039001_G_C: HC - ['0|1', 0.19444444444444445, 36], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77095100 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "checking variants for 77095101\n",
      "77095101\n",
      "check 32039046_A_G: HC - ['0|1', 0.13636363636363635, 22], DV - not called, ['0/1', 0.4199424736337488, 1043]\n",
      "HC genotype is supported by bam\n",
      "77095101\n",
      "check 32039055_C_G: HC - ['0|1', 0.11475409836065574, 61], DV - not called, ['0/1', 0.3966725043782837, 1142]\n",
      "HC genotype is supported by bam\n",
      "77095101\n",
      "check 32039056_A_G: HC - ['0|1', 0.11475409836065574, 61], DV - not called, ['0/1', 0.398576512455516, 1124]\n",
      "HC genotype is supported by bam\n",
      "32039081_C_G in 77095101 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/0, ['1/0', 0.4518581081081081, 1184]\n",
      "checking variants for 77095102\n",
      "checking variants for 77095103\n",
      "32038855_T_TTTG in 77095103 found in DV, but not in HC, checking it\n",
      "low qual 32038855_T_TTTG in 77095103, checking inheritance; variant info: 0/1\n",
      "77095103 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "32038857_A_T in 77095103 found in DV, but not in HC, checking it\n",
      "low qual 32038857_A_T in 77095103, checking inheritance; variant info: 0/1\n",
      "77095103 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "checking variants for 77095104\n",
      "checking variants for 77095105\n",
      "checking variants for 77095106\n",
      "77095106\n",
      "check 32038948_T_TC: HC - ['1|0', 0.14130434782608695, 184], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77095106 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095106\n",
      "check 32038949_T_G: HC - ['1|0', 0.14130434782608695, 184], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77095106 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095106\n",
      "check 32038955_A_T: HC - ['1|0', 0.1452513966480447, 179], DV - not called, ['0/1', 0.2923728813559322, 944]\n",
      "HC genotype is supported by bam\n",
      "77095106\n",
      "check 32038962_A_ATCC: HC - ['1|0', 0.0958904109589041, 73], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77095106 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095106\n",
      "check 32038971_C_A: HC - ['1|0', 0.08823529411764706, 68], DV - not called, ['0/1', 0.32129277566539927, 1052]\n",
      "HC genotype is supported by bam\n",
      "77095106\n",
      "check 32039027_C_A: HC - ['1|0', 0.17391304347826086, 23], DV - not called, ['0/1', 0.4174454828660436, 1284]\n",
      "HC genotype is supported by bam\n",
      "77095106\n",
      "check 32039046_A_G: HC - ['1|0', 0.13043478260869565, 23], DV - not called, ['0/1', 0.40408805031446543, 1272]\n",
      "HC genotype is supported by bam\n",
      "77095106\n",
      "check 32039050_G_T: HC - ['1|0', 0.16, 25], DV - not called, ['0/1', 0.4121306376360809, 1286]\n",
      "HC genotype is supported by bam\n",
      "77095106\n",
      "check 32039055_C_G: HC - ['1|0', 0.15384615384615385, 26], DV - not called, ['0/1', 0.3904059040590406, 1355]\n",
      "HC genotype is supported by bam\n",
      "77095106\n",
      "check 32039056_A_G: HC - ['1|0', 0.15384615384615385, 26], DV - not called, ['0/1', 0.39538346984363365, 1343]\n",
      "HC genotype is supported by bam\n",
      "checking variants for 77095107\n",
      "77095107\n",
      "check 32038949_T_C: HC - ['0|1', 0.10526315789473684, 190], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77095107 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095107\n",
      "check 32038954_C_G: HC - ['0|1', 0.10582010582010581, 189], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77095107 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095107\n",
      "check 32038955_A_C: HC - ['0|1', 0.10638297872340426, 188], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77095107 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095107\n",
      "check 32038962_A_ATCC: HC - ['0|1', 0.13846153846153847, 65], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77095107 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095107\n",
      "check 32038971_C_A: HC - ['0|1', 0.15254237288135594, 59], DV - not called, ['0/1', 0.35664335664335667, 858]\n",
      "HC genotype is supported by bam\n",
      "77095107\n",
      "check 32039056_A_G: HC - ['0|1', 0.14285714285714285, 14], DV - not called, ['0/1', 0.36381709741550694, 1006]\n",
      "HC genotype is supported by bam\n",
      "32038844_A_C in 77095107 found in DV, but not in HC, checking it\n",
      "32038844_A_C not called by HC, saving it with 0/1, ['0/1', 0.37524557956778, 1527]\n",
      "32038855_T_TTTG in 77095107 found in DV, but not in HC, checking it\n",
      "low qual 32038855_T_TTTG in 77095107, checking inheritance; variant info: 0/1\n",
      "77095107 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "32038857_A_T in 77095107 found in DV, but not in HC, checking it\n",
      "low qual 32038857_A_T in 77095107, checking inheritance; variant info: 0/1\n",
      "77095107 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "32038867_T_G in 77095107 found in DV, but not in HC, checking it\n",
      "32038867_T_G not called by HC, saving it with 0/1, ['0/1', 0.41487839771101576, 1398]\n",
      "32038911_A_G in 77095107 found in DV, but not in HC, checking it\n",
      "32038911_A_G not called by HC, saving it with 0/1, ['0/1', 0.41625615763546797, 1218]\n",
      "32039050_G_T in 77095107 found in DV, but not in HC, checking it\n",
      "32039050_G_T not called by HC, saving it with 0/1, ['0/1', 0.37991718426501037, 966]\n",
      "checking variants for 77095108\n",
      "77095108\n",
      "check 32038938_T_G: HC - ['1|0', 0.17647058823529413, 17], DV - not called, ['0/1', 0.5263702171664943, 967]\n",
      "HC genotype is supported by bam\n",
      "checking variants for 77095109\n",
      "77095109\n",
      "check 32039001_G_C: HC - ['0|1', 0.18181818181818182, 22], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77095109 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095109\n",
      "check 32039002_G_A: HC - ['0|1', 0.18181818181818182, 22], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77095109 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095109\n",
      "check 32039007_A_AAG: HC - ['0|1', 0.17391304347826086, 23], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "77095109 does not have relatives\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095109\n",
      "check 32039015_G_T: HC - ['0|1', 0.13793103448275862, 29], DV - not called, ['0/1', 0.36205128205128206, 975]\n",
      "HC genotype is supported by bam\n",
      "77095109\n",
      "check 32039055_C_G: HC - ['0|1', 0.15789473684210525, 152], DV - not called, ['0/1', 0.4088628762541806, 1196]\n",
      "HC genotype is supported by bam\n",
      "checking variants for 77095110\n",
      "checking variants for 77095111\n",
      "checking variants for 77095112\n",
      "checking variants for 77095330\n",
      "checking variants for 77095331\n",
      "checking variants for 77095332\n",
      "checking variants for 77095333\n",
      "checking variants for 77095334\n",
      "32038878_C_T in 77095334 found in DV, but not in HC, checking it\n",
      "32038878_C_T not called by HC, saving it with 0/1, ['0/1', 0.5082089552238805, 1340]\n",
      "checking variants for 77095335\n",
      "check 32038844_A_C: HC - ['1|0', 0.5882352941176471, 17], DV - 1/1, ['1/1', 0.856091030789826, 1494]\n",
      "HC genotype is not supported by bam, saving a new genotype\n",
      "['1/1', 0.856091030789826, 1494]\n",
      "check 32038855_T_TTTG: HC - ['1|0', 0.4444444444444444, 18], DV - 1/1, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77095336, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038855_T_TTTG in 77095334 HC data - ['0|1', 0.4225352112676056, 71]\n",
      "32038855_T_TTTG found in HC data of mother\n",
      "found a heterozygote 32038855_T_TTTG in mother with VAF 0.4225352112676056\n",
      "found 32038855_T_TTTG in 77095340 HC data - ['0|1', 0.49230769230769234, 130]\n",
      "32038855_T_TTTG found in HC data of father\n",
      "found a heterozygote 32038855_T_TTTG in father with VAF 0.49230769230769234\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "check 32038857_A_T: HC - ['1|0', 0.4444444444444444, 18], DV - 1/1, ['1/1', 0.926873857404022, 547]\n",
      "HC genotype is not supported by bam, saving a new genotype\n",
      "['1/1', 0.926873857404022, 547]\n",
      "77095335\n",
      "check 32038983_TG_T: HC - ['1|1', 1.0, 16], DV - not called, bam - Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77095336, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038983_TG_T in 77095334 HC data - ['0|1', 0.4883720930232558, 43]\n",
      "32038983_TG_T found in HC data of mother\n",
      "found a heterozygote 32038983_TG_T in mother with VAF 0.4883720930232558\n",
      "found 32038983_TG_T in 77095340 HC data - ['0|1', 0.6296296296296297, 27]\n",
      "32038983_TG_T found in HC data of father\n",
      "found a lowqual heterozygote 32038983_TG_T in father with right VAF 0.6296296296296297, but low DP - 27\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "77095335\n",
      "check 32038997_TGG_T: HC - ['1|1', 1.0, 8], DV - not called, bam - Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77095336, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038997_TGG_T in 77095334 HC data - ['0|1', 0.4444444444444444, 27]\n",
      "32038997_TGG_T found in HC data of mother\n",
      "found a lowqual heterozygote 32038997_TGG_T in mother with right VAF 0.4444444444444444, but low DP - 27\n",
      "found 32038997_TGG_T in 77095340 HC data - ['0|1', 0.47058823529411764, 17]\n",
      "32038997_TGG_T found in HC data of father\n",
      "found a lowqual heterozygote 32038997_TGG_T in father with right VAF 0.47058823529411764, but low DP - 17\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "77095335\n",
      "check 32039001_G_C: HC - ['1|1', 1.0, 4], DV - not called, bam - ['1/1', 0.9395348837209302, 215]\n",
      "genotype is supported by bam\n",
      "77095335\n",
      "check 32039007_A_AAG: HC - ['1|1', 1.0, 4], DV - not called, bam - Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77095336, 'mother': 77095334, 'father': 77095340}\n",
      "found 32039007_A_AAG in 77095334 HC data - ['0|1', 0.6666666666666666, 24]\n",
      "32039007_A_AAG found in HC data of mother\n",
      "found a lowqual heterozygote 32039007_A_AAG in mother with right VAF 0.6666666666666666, but low DP - 24\n",
      "found 32039007_A_AAG in 77095340 HC data - ['0|1', 0.5, 12]\n",
      "32039007_A_AAG found in HC data of father\n",
      "found a lowqual heterozygote 32039007_A_AAG in father with right VAF 0.5, but low DP - 12\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038867_T_G in 77095335 found in DV, but not in HC, checking it\n",
      "32038867_T_G not called by HC, saving it with 1/1, ['1/1', 0.9941666666666666, 1200]\n",
      "32038878_C_T in 77095335 found in DV, but not in HC, checking it\n",
      "32038878_C_T not called by HC, saving it with 1/1, ['1/1', 0.9899328859060402, 1192]\n",
      "32038895_A_G in 77095335 found in DV, but not in HC, checking it\n",
      "32038895_A_G not called by HC, saving it with 1/1, ['1/1', 0.9955035971223022, 1112]\n",
      "32038903_A_G in 77095335 found in DV, but not in HC, checking it\n",
      "32038903_A_G not called by HC, saving it with 1/1, ['1/1', 0.9980372914622179, 1019]\n",
      "checking variants for 77095336\n",
      "77095336\n",
      "check 32038818_G_T: HC - ['0|1', 0.13636363636363635, 22], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "32038818_G_T not found in gt_hc of 77095334 - the mother of 77095336\n",
      "32038818_G_T not found in gt_dv of 77095334 - the mother of 77095336\n",
      "no 32038818_G_T found in mother\n",
      "32038818_G_T not found in gt_hc of 77095340 - the father of 77095336\n",
      "32038818_G_T not found in gt_dv of 77095340 - the father of 77095336\n",
      "no 32038818_G_T found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "check 32038844_A_C: HC - ['1|0', 0.5294117647058824, 17], DV - 1/1, ['1/1', 0.8657587548638133, 1542]\n",
      "HC genotype is not supported by bam, saving a new genotype\n",
      "['1/1', 0.8657587548638133, 1542]\n",
      "77095336\n",
      "check 32038863_TGG_T: HC - ['1|1', 1.0, 3], DV - not called, bam - Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "32038863_TGG_T not found in gt_hc of 77095334 - the mother of 77095336\n",
      "32038863_TGG_T not found in gt_dv of 77095334 - the mother of 77095336\n",
      "no 32038863_TGG_T found in mother\n",
      "32038863_TGG_T not found in gt_hc of 77095340 - the father of 77095336\n",
      "32038863_TGG_T not found in gt_dv of 77095340 - the father of 77095336\n",
      "no 32038863_TGG_T found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "32038855_T_TTTG in 77095336 found in DV, but not in HC, checking it\n",
      "low qual 32038855_T_TTTG in 77095336, checking inheritance; variant info: 1/1\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038855_T_TTTG in 77095334 HC data - ['0|1', 0.4225352112676056, 71]\n",
      "32038855_T_TTTG found in HC data of mother\n",
      "found a heterozygote 32038855_T_TTTG in mother with VAF 0.4225352112676056\n",
      "found 32038855_T_TTTG in 77095340 HC data - ['0|1', 0.49230769230769234, 130]\n",
      "32038855_T_TTTG found in HC data of father\n",
      "found a heterozygote 32038855_T_TTTG in father with VAF 0.49230769230769234\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038857_A_T in 77095336 found in DV, but not in HC, checking it\n",
      "32038857_A_T not called by HC, saving it with 1/1, ['1/1', 0.9351145038167938, 524]\n",
      "32038867_T_G in 77095336 found in DV, but not in HC, checking it\n",
      "32038867_T_G not called by HC, saving it with 1/1, ['1/1', 0.9975, 1200]\n",
      "32038878_C_T in 77095336 found in DV, but not in HC, checking it\n",
      "32038878_C_T not called by HC, saving it with 1/1, ['1/1', 0.9983221476510067, 1192]\n",
      "32038895_A_G in 77095336 found in DV, but not in HC, checking it\n",
      "32038895_A_G not called by HC, saving it with 1/1, ['1/1', 0.994768962510898, 1147]\n",
      "32038903_A_G in 77095336 found in DV, but not in HC, checking it\n",
      "32038903_A_G not called by HC, saving it with 1/1, ['1/1', 0.9971936389148737, 1069]\n",
      "32038911_A_G in 77095336 found in DV, but not in HC, checking it\n",
      "32038911_A_G not called by HC, saving it with 1/1, ['1/1', 0.9952335557673975, 1049]\n",
      "32038927_A_G in 77095336 found in DV, but not in HC, checking it\n",
      "32038927_A_G not called by HC, saving it with 1/1, ['1/1', 1.0, 950]\n",
      "32038938_T_G in 77095336 found in DV, but not in HC, checking it\n",
      "32038938_T_G not called by HC, saving it with 1/1, ['1/1', 0.9976717112922002, 859]\n",
      "32038948_T_TC in 77095336 found in DV, but not in HC, checking it\n",
      "low qual 32038948_T_TC in 77095336, checking inheritance; variant info: 1/1\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038948_T_TC in 77095334 HC data - ['0|1', 0.38349514563106796, 206]\n",
      "32038948_T_TC found in HC data of mother\n",
      "found a heterozygote 32038948_T_TC in mother with VAF 0.38349514563106796\n",
      "found 32038948_T_TC in 77095340 HC data - ['0|1', 0.3615819209039548, 177]\n",
      "32038948_T_TC found in HC data of father\n",
      "found a heterozygote 32038948_T_TC in father with VAF 0.3615819209039548\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038949_T_G in 77095336 found in DV, but not in HC, checking it\n",
      "low qual 32038949_T_G in 77095336, checking inheritance; variant info: 1/1\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038949_T_G in 77095334 HC data - ['0|1', 0.4708333333333333, 240]\n",
      "32038949_T_G found in HC data of mother\n",
      "found a heterozygote 32038949_T_G in mother with VAF 0.4708333333333333\n",
      "found 32038949_T_G in 77095340 HC data - ['0|1', 0.3989071038251366, 183]\n",
      "32038949_T_G found in HC data of father\n",
      "found a heterozygote 32038949_T_G in father with VAF 0.3989071038251366\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038955_A_T in 77095336 found in DV, but not in HC, checking it\n",
      "32038955_A_T not called by HC, saving it with 1/1, ['1/1', 1.0, 463]\n",
      "32038962_A_ATCC in 77095336 found in DV, but not in HC, checking it\n",
      "low qual 32038962_A_ATCC in 77095336, checking inheritance; variant info: 1/1\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038962_A_ATCC in 77095334 HC data - ['0|1', 0.42592592592592593, 108]\n",
      "32038962_A_ATCC found in HC data of mother\n",
      "found a heterozygote 32038962_A_ATCC in mother with VAF 0.42592592592592593\n",
      "found 32038962_A_ATCC in 77095340 HC data - ['0|1', 0.45977011494252873, 87]\n",
      "32038962_A_ATCC found in HC data of father\n",
      "found a heterozygote 32038962_A_ATCC in father with VAF 0.45977011494252873\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038971_C_A in 77095336 found in DV, but not in HC, checking it\n",
      "32038971_C_A not called by HC, saving it with 1/1, ['1/1', 0.8795888399412628, 681]\n",
      "32038983_TG_T in 77095336 found in DV, but not in HC, checking it\n",
      "low qual 32038983_TG_T in 77095336, checking inheritance; variant info: 1/1\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038983_TG_T in 77095334 HC data - ['0|1', 0.4883720930232558, 43]\n",
      "32038983_TG_T found in HC data of mother\n",
      "found a heterozygote 32038983_TG_T in mother with VAF 0.4883720930232558\n",
      "found 32038983_TG_T in 77095340 HC data - ['0|1', 0.6296296296296297, 27]\n",
      "32038983_TG_T found in HC data of father\n",
      "found a lowqual heterozygote 32038983_TG_T in father with right VAF 0.6296296296296297, but low DP - 27\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038997_TGG_T in 77095336 found in DV, but not in HC, checking it\n",
      "low qual 32038997_TGG_T in 77095336, checking inheritance; variant info: 1/1\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038997_TGG_T in 77095334 HC data - ['0|1', 0.4444444444444444, 27]\n",
      "32038997_TGG_T found in HC data of mother\n",
      "found a lowqual heterozygote 32038997_TGG_T in mother with right VAF 0.4444444444444444, but low DP - 27\n",
      "found 32038997_TGG_T in 77095340 HC data - ['0|1', 0.47058823529411764, 17]\n",
      "32038997_TGG_T found in HC data of father\n",
      "found a lowqual heterozygote 32038997_TGG_T in father with right VAF 0.47058823529411764, but low DP - 17\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32039002_G_A in 77095336 found in DV, but not in HC, checking it\n",
      "low qual 32039002_G_A in 77095336, checking inheritance; variant info: 1/1\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32039002_G_A in 77095334 HC data - ['0|1', 0.5217391304347826, 23]\n",
      "32039002_G_A found in HC data of mother\n",
      "found a lowqual heterozygote 32039002_G_A in mother with right VAF 0.5217391304347826, but low DP - 23\n",
      "found 32039002_G_A in 77095340 HC data - ['0|1', 0.5, 14]\n",
      "32039002_G_A found in HC data of father\n",
      "found a lowqual heterozygote 32039002_G_A in father with right VAF 0.5, but low DP - 14\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32039007_A_AAG in 77095336 found in DV, but not in HC, checking it\n",
      "low qual 32039007_A_AAG in 77095336, checking inheritance; variant info: 1/1\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32039007_A_AAG in 77095334 HC data - ['0|1', 0.6666666666666666, 24]\n",
      "32039007_A_AAG found in HC data of mother\n",
      "found a lowqual heterozygote 32039007_A_AAG in mother with right VAF 0.6666666666666666, but low DP - 24\n",
      "found 32039007_A_AAG in 77095340 HC data - ['0|1', 0.5, 12]\n",
      "32039007_A_AAG found in HC data of father\n",
      "found a lowqual heterozygote 32039007_A_AAG in father with right VAF 0.5, but low DP - 12\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32039015_G_T in 77095336 found in DV, but not in HC, checking it\n",
      "32039015_G_T not called by HC, saving it with 1/1, ['1/1', 0.889922480620155, 645]\n",
      "32039020_G_A in 77095336 found in DV, but not in HC, checking it\n",
      "32039020_G_A not called by HC, saving it with 1/1, ['1/1', 0.9424460431654677, 695]\n",
      "32039027_C_A in 77095336 found in DV, but not in HC, checking it\n",
      "32039027_C_A not called by HC, saving it with 1/1, ['1/1', 0.9324796274738067, 859]\n",
      "32039046_A_G in 77095336 found in DV, but not in HC, checking it\n",
      "32039046_A_G not called by HC, saving it with 1/1, ['1/1', 1.0, 926]\n",
      "32039050_G_T in 77095336 found in DV, but not in HC, checking it\n",
      "32039050_G_T not called by HC, saving it with 1/1, ['1/1', 0.9989212513484358, 927]\n",
      "32039055_C_G in 77095336 found in DV, but not in HC, checking it\n",
      "32039055_C_G not called by HC, saving it with 1/1, ['1/1', 1.0, 923]\n",
      "32039056_A_G in 77095336 found in DV, but not in HC, checking it\n",
      "32039056_A_G not called by HC, saving it with 1/1, ['1/1', 1.0, 923]\n",
      "32039081_C_G in 77095336 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.9982300884955753, 1130]\n",
      "32039109_G_A in 77095336 found in DV, but not in HC, checking it\n",
      "32039109_G_A not called by HC, saving it with 1/1, ['1/1', 0.9527638190954774, 995]\n",
      "32039128_C_G in 77095336 found in DV, but not in HC, checking it\n",
      "32039128_C_G not called by HC, saving it with 1/1, ['1/1', 0.9793233082706767, 532]\n",
      "32039132_GGAGACTAC_G in 77095336 found in DV, but not in HC, checking it\n",
      "low qual 32039132_GGAGACTAC_G in 77095336, checking inheritance; variant info: 1/1\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32039132_GGAGACTAC_G in 77095334 HC data - ['0|1', 0.4037854889589905, 317]\n",
      "32039132_GGAGACTAC_G found in HC data of mother\n",
      "found a heterozygote 32039132_GGAGACTAC_G in mother with VAF 0.4037854889589905\n",
      "found 32039132_GGAGACTAC_G in 77095340 HC data - ['0|1', 0.3870967741935484, 248]\n",
      "32039132_GGAGACTAC_G found in HC data of father\n",
      "found a heterozygote 32039132_GGAGACTAC_G in father with VAF 0.3870967741935484\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32039143_C_T in 77095336 found in DV, but not in HC, checking it\n",
      "checking variants for 77095337\n",
      "checking variants for 77095338\n",
      "77095338\n",
      "check 32038997_TGG_T: HC - ['1|0', 0.16279069767441862, 43], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77095337, 'father': 77095339, 'mother': 77095341}\n",
      "found 32038997_TGG_T in 77095339 HC data - ['0|1', 0.4722222222222222, 36]\n",
      "32038997_TGG_T found in HC data of father\n",
      "found a heterozygote 32038997_TGG_T in father with VAF 0.4722222222222222\n",
      "32038997_TGG_T not found in gt_hc of 77095341 - the mother of 77095338\n",
      "32038997_TGG_T not found in gt_dv of 77095341 - the mother of 77095338\n",
      "no 32038997_TGG_T found in mother\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "77095338\n",
      "check 32039001_G_C: HC - ['1|0', 0.15625, 32], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77095337, 'father': 77095339, 'mother': 77095341}\n",
      "found 32039001_G_C in 77095339 HC data - ['0|1', 0.37777777777777777, 45]\n",
      "32039001_G_C found in HC data of father\n",
      "found a heterozygote 32039001_G_C in father with VAF 0.37777777777777777\n",
      "32039001_G_C not found in gt_hc of 77095341 - the mother of 77095338\n",
      "32039001_G_C not found in gt_dv of 77095341 - the mother of 77095338\n",
      "no 32039001_G_C found in mother\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "77095338\n",
      "check 32039002_G_A: HC - ['1|0', 0.16129032258064516, 31], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77095337, 'father': 77095339, 'mother': 77095341}\n",
      "found 32039002_G_A in 77095339 HC data - ['0|1', 0.3695652173913043, 46]\n",
      "32039002_G_A found in HC data of father\n",
      "found a heterozygote 32039002_G_A in father with VAF 0.3695652173913043\n",
      "32039002_G_A not found in gt_hc of 77095341 - the mother of 77095338\n",
      "32039002_G_A not found in gt_dv of 77095341 - the mother of 77095338\n",
      "no 32039002_G_A found in mother\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "77095338\n",
      "check 32039015_G_T: HC - ['1|0', 0.18518518518518517, 27], DV - not called, ['0/1', 0.3539518900343643, 1164]\n",
      "HC genotype is supported by bam\n",
      "77095338\n",
      "check 32039020_G_A: HC - ['1|0', 0.18518518518518517, 27], DV - not called, ['0/1', 0.3851851851851852, 1215]\n",
      "HC genotype is supported by bam\n",
      "77095338\n",
      "check 32039027_C_A: HC - ['1|0', 0.14705882352941177, 34], DV - not called, ['0/1', 0.4237947122861586, 1286]\n",
      "HC genotype is supported by bam\n",
      "77095338\n",
      "check 32039056_A_G: HC - ['1|0', 0.17647058823529413, 136], DV - not called, ['0/1', 0.39257673090649536, 1401]\n",
      "HC genotype is supported by bam\n",
      "checking variants for 77095339\n",
      "checking variants for 77095340\n",
      "checking variants for 77095341\n",
      "checking variants for 77095342\n",
      "77095342\n",
      "check 32038962_A_ATCC: HC - ['0|1', 0.17857142857142858, 28], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77095343, 'child': 77095344}\n",
      "found 32038962_A_ATCC in 77095344 HC data - ['1|0', 0.35, 40]\n",
      "32038962_A_ATCC found in HC data of child\n",
      "found a heterozygote 32038962_A_ATCC in child with VAF 0.35\n",
      "['1|0']\n",
      "saving this variant even with bad QUAL\n",
      "77095342\n",
      "check 32038983_TG_T: HC - ['0|1', 0.10714285714285714, 28], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77095343, 'child': 77095344}\n",
      "32038983_TG_T not found in gt_hc of 77095344 - the child of 77095342\n",
      "32038983_TG_T not found in gt_dv of 77095344 - the child of 77095342\n",
      "no 32038983_TG_T found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095342\n",
      "check 32038997_TGG_T: HC - ['0|1', 0.1282051282051282, 39], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77095343, 'child': 77095344}\n",
      "32038997_TGG_T not found in gt_hc of 77095344 - the child of 77095342\n",
      "32038997_TGG_T not found in gt_dv of 77095344 - the child of 77095342\n",
      "no 32038997_TGG_T found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095342\n",
      "check 32039001_G_C: HC - ['0|1', 0.15384615384615385, 39], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77095343, 'child': 77095344}\n",
      "32039001_G_C not found in gt_hc of 77095344 - the child of 77095342\n",
      "32039001_G_C not found in gt_dv of 77095344 - the child of 77095342\n",
      "no 32039001_G_C found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095342\n",
      "check 32039002_G_A: HC - ['0|1', 0.15384615384615385, 39], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77095343, 'child': 77095344}\n",
      "32039002_G_A not found in gt_hc of 77095344 - the child of 77095342\n",
      "32039002_G_A not found in gt_dv of 77095344 - the child of 77095342\n",
      "no 32039002_G_A found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "checking variants for 77095343\n",
      "checking variants for 77095344\n",
      "77095344\n",
      "check 32039015_G_T: HC - ['1|0', 0.15384615384615385, 13], DV - not called, ['0/1', 0.3511705685618729, 897]\n",
      "HC genotype is supported by bam\n",
      "32038514_C_T in 77095344 found in DV, but not in HC, checking it\n",
      "32038514_C_T not called by HC, saving it with 0/1, ['0/1', 0.5012345679012346, 810]\n",
      "32038844_A_C in 77095344 found in DV, but not in HC, checking it\n",
      "32038844_A_C not called by HC, saving it with 0/1, ['0/1', 0.4449789536981359, 1663]\n",
      "32038878_C_T in 77095344 found in DV, but not in HC, checking it\n",
      "32038878_C_T not called by HC, saving it with 0/1, ['0/1', 0.49463135289906945, 1397]\n",
      "checking variants for 77095345\n",
      "checking variants for 77095346\n",
      "check 32038844_A_C: HC - ['1|0', 0.8571428571428571, 7], DV - 1/1, ['1/1', 0.8812580231065469, 1558]\n",
      "HC genotype is not supported by bam, saving a new genotype\n",
      "['1/1', 0.8812580231065469, 1558]\n",
      "check 32038855_T_TTTG: HC - ['1|0', 0.3, 10], DV - 1/1, Not found in bam\n",
      "need to check inheritance\n",
      "{'sibling': 77095345}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "check 32038857_A_T: HC - ['1|0', 0.3, 10], DV - 1/1, ['1/1', 0.9518716577540107, 561]\n",
      "HC genotype is not supported by bam, saving a new genotype\n",
      "['1/1', 0.9518716577540107, 561]\n",
      "32038820_C_T in 77095346 found in DV, but not in HC, checking it\n",
      "32038820_C_T not called by HC, saving it with 0/1, ['0/1', 0.4819976771196283, 1722]\n",
      "32038855_T_TTTG in 77095346 found in DV, but not in HC, checking it\n",
      "low qual 32038855_T_TTTG in 77095346, checking inheritance; variant info: 1/1\n",
      "{'sibling': 77095345}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "32038867_T_G in 77095346 found in DV, but not in HC, checking it\n",
      "32038867_T_G not called by HC, saving it with 1/1, ['1/1', 0.9959839357429718, 1245]\n",
      "32038878_C_T in 77095346 found in DV, but not in HC, checking it\n",
      "32038878_C_T not called by HC, saving it with 1/1, ['1/1', 0.9943865276663993, 1247]\n",
      "32038895_A_G in 77095346 found in DV, but not in HC, checking it\n",
      "32038895_A_G not called by HC, saving it with 1/1, ['1/1', 0.9958333333333333, 1200]\n",
      "32038903_A_G in 77095346 found in DV, but not in HC, checking it\n",
      "32038903_A_G not called by HC, saving it with 1/1, ['1/1', 0.9991039426523297, 1116]\n",
      "32038949_T_C in 77095346 found in DV, but not in HC, checking it\n",
      "32038949_T_C not called by HC, saving it with 1/0, ['0/1', 0.6363636363636364, 44]\n",
      "checking variants for 77095347\n",
      "checking variants for 77095349\n",
      "checking variants for 77095350\n",
      "77095350\n",
      "check 32038948_T_TC: HC - ['0|1', 0.15675675675675677, 185], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77095347, 'child': 77095354}\n",
      "32038948_T_TC not found in gt_hc of 77095354 - the child of 77095350\n",
      "32038948_T_TC not found in gt_dv of 77095354 - the child of 77095350\n",
      "no 32038948_T_TC found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095350\n",
      "check 32038949_T_G: HC - ['0|1', 0.15675675675675677, 185], DV - not called, Not found in bam\n",
      "need to check inheritance\n",
      "{'spouse': 77095347, 'child': 77095354}\n",
      "32038949_T_G not found in gt_hc of 77095354 - the child of 77095350\n",
      "32038949_T_G not found in gt_dv of 77095354 - the child of 77095350\n",
      "no 32038949_T_G found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "77095350\n",
      "check 32038955_A_T: HC - ['0|1', 0.16111111111111112, 180], DV - not called, ['0/1', 0.2797281993204983, 883]\n",
      "HC genotype is supported by bam\n",
      "check 32039081_C_A: HC - ['1|1', 1.0, 21], DV - 1/0, bam - ['1/0', 0.5308720560152769, 1571]\n",
      "HC genotype is not supported by bam, saving a new genotype\n",
      "['1/0', 0.5308720560152769, 1571]\n",
      "32038878_C_T in 77095350 found in DV, but not in HC, checking it\n",
      "32038878_C_T not called by HC, saving it with 0/1, ['0/1', 0.4304253785147801, 1387]\n",
      "32039050_G_T in 77095350 found in DV, but not in HC, checking it\n",
      "32039050_G_T not called by HC, saving it with 0/1, ['0/1', 0.4042402826855124, 1415]\n",
      "32039056_A_G in 77095350 found in DV, but not in HC, checking it\n",
      "32039056_A_G not called by HC, saving it with 0/1, ['0/1', 0.3830074173971679, 1483]\n",
      "32039081_C_G in 77095350 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/0, ['1/0', 0.46085295989815406, 1571]\n",
      "checking variants for 77095351\n",
      "checking variants for 77095352\n",
      "checking variants for 77095353\n",
      "checking variants for 77095354\n",
      "32038855_T_TTTG in 77095354 found in DV, but not in HC, checking it\n",
      "low qual 32038855_T_TTTG in 77095354, checking inheritance; variant info: 0/1\n",
      "{'mother': 77095347, 'sibling': 77095349, 'father': 77095350}\n",
      "32038855_T_TTTG not found in gt_hc of 77095347 - the mother of 77095354\n",
      "32038855_T_TTTG not found in gt_dv of 77095347 - the mother of 77095354\n",
      "no 32038855_T_TTTG found in mother\n",
      "found 32038855_T_TTTG in 77095350 HC data - ['0|1', 0.23478260869565218, 115]\n",
      "32038855_T_TTTG found in HC data of father\n",
      "found a heterozygote 32038855_T_TTTG in father with VAF 0.23478260869565218\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "32038911_A_G in 77095354 found in DV, but not in HC, checking it\n",
      "32038911_A_G not called by HC, saving it with 0/1, ['0/1', 0.4220110847189232, 1263]\n",
      "32039020_G_A in 77095354 found in DV, but not in HC, checking it\n",
      "32039020_G_A not called by HC, saving it with 0/1, ['0/1', 0.3322751322751323, 945]\n",
      "32039027_C_A in 77095354 found in DV, but not in HC, checking it\n",
      "32039027_C_A not called by HC, saving it with 1/0, ['1/0', 0.3713420787083754, 991]\n",
      "32039046_A_G in 77095354 found in DV, but not in HC, checking it\n",
      "32039046_A_G not called by HC, saving it with 0/1, ['0/1', 0.3666997026759167, 1009]\n",
      "32039050_G_T in 77095354 found in DV, but not in HC, checking it\n",
      "32039050_G_T not called by HC, saving it with 0/1, ['0/1', 0.3646944713870029, 1031]\n",
      "32039081_C_G in 77095354 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/0, ['1/0', 0.40606060606060607, 1155]\n",
      "32039109_G_A in 77095354 found in DV, but not in HC, checking it\n",
      "32039109_G_A not called by HC, saving it with 0/1, ['0/1', 0.3791304347826087, 1150]\n",
      "checking variants for 77095355\n",
      "checking variants for 77095356\n",
      "checking variants for 77095357\n",
      "checking variants for 77095358\n",
      "checking variants for 77095359\n",
      "checking variants for 77095360\n",
      "checking variants for 77095361\n",
      "checking variants for 77095362\n",
      "checking variants for 77095363\n",
      "checking variants for 77095364\n",
      "checking variants for 77095365\n",
      "checking variants for 77095366\n",
      "checking variants for 77095367\n",
      "checking variants for 77095368\n",
      "checking variants for 77095373\n",
      "checking variants for 77095374\n",
      "checking variants for 77095375\n",
      "checking variants for 77095376\n",
      "checking variants for 77095377\n",
      "checking variants for 77095378\n",
      "checking variants for 77095379\n",
      "checking variants for 77095380\n",
      "checking variants for 77095381\n",
      "checking variants for 77095382\n",
      "checking variants for 77095383\n",
      "checking variants for 77095384\n",
      "32039081_C_G in 77095384 found in DV, but not in HC, checking it\n",
      "32039081_C_G not called by HC, saving it with 1/1, ['1/1', 0.9991769547325103, 1215]\n",
      "checking variants for 77095385\n",
      "checking variants for 77095386\n",
      "checking variants for 77095387\n",
      "checking variants for 77095388\n",
      "checking variants for 77095389\n",
      "checking variants for 77095390\n"
     ]
    }
   ],
   "source": [
    "res = defaultdict(dict)\n",
    "for idx, row in files_list.iterrows():\n",
    "    # dv\n",
    "    vcf = read_vcf(row['dv'], mode = 'dv')\n",
    "    vcf = vcf[vcf['FILTER'] == 'PASS']\n",
    "    vcf = vcf[(vcf['POS'] >= calling_region[0]) & (vcf['POS'] <= calling_region[1])]\n",
    "    sample_dv=gt_vectors(vcf, mode = 'dv')\n",
    "    # hc \n",
    "    vcf = read_vcf(row['hc'], mode='hc')\n",
    "    vcf = vcf[(vcf['POS'] >= calling_region[0]) & (vcf['POS'] <= calling_region[1])]\n",
    "    sample_hc=gt_vectors(vcf, mode= 'hc')\n",
    "\n",
    "    #pileup\n",
    "    vcf = read_vcf(row['pileup'], mode='bam')\n",
    "    vcf = vcf[(vcf['POS'] >= calling_region[0]) & (vcf['POS'] <= calling_region[1])]\n",
    "    sample_pp=gt_vectors(vcf, mode= 'dv')\n",
    "\n",
    "    print(f'checking variants for {idx}')\n",
    "    for position in sample_hc.keys():\n",
    "        hc_value=sample_hc[position][0]\n",
    "        hc=hc_value.split('|')\n",
    "        hc=int(hc[0])+int(hc[1])\n",
    "        if position in sample_dv.keys():\n",
    "            dv_value=sample_dv[position][0]\n",
    "            dv=dv_value.split('/')\n",
    "            dv=int(dv[0])+int(dv[1])\n",
    "            if hc == dv:\n",
    "                # print(f'no conflict for {position} in {idx}, saving HC genotype')\n",
    "                res[idx][position] = hc_value\n",
    "            if hc!=dv:\n",
    "                if hc == 2:\n",
    "                    if sample_hc[position][1] > 0.85 and sample_hc[position][2]>=30:\n",
    "                        res[idx][position] = hc_value\n",
    "                    else:\n",
    "                        print(f'check {position}: HC - {sample_hc[position]}, DV - {dv_value}, bam - {sample_pp.get(position, 'Not found in bam')}')\n",
    "                        if position in sample_pp.keys():\n",
    "                            pp_value=sample_pp[position][0]\n",
    "                            pp=pp_value.split('/')\n",
    "                            pp=int(pp[0])+int(pp[1])\n",
    "                            if pp==2 and sample_pp[position][1]>=0.85:\n",
    "                                res[idx][position]  = hc_value\n",
    "                                print(f'HC genotype is supported by bam')\n",
    "                            if pp==1 and (sample_pp[position][1]>=0.2 and sample_pp[position][1]<0.85):\n",
    "                                print(f'HC genotype is not supported by bam, saving a new genotype')\n",
    "                                print(sample_pp[position])\n",
    "                                res[idx][position] = pp_value\n",
    "                        else:\n",
    "                            print('need to check inheritance')\n",
    "                            inheritance = check_inheritance(idx, position, trios, gt_hc, gt_dv, gt_bam)\n",
    "                            if inheritance == True:\n",
    "                                print('saving this variant even with bad QUAL')\n",
    "                                res[idx][position]  = hc_value\n",
    "                            else:\n",
    "                                print('droping this variant, probably artefact')\n",
    "                                \n",
    "                if hc == 1: \n",
    "                    if (sample_hc[position][1] >= 0.2 and sample_hc[position][1] < 0.85) and sample_hc[position][2]>=30:\n",
    "                        res[idx][position]  = hc_value\n",
    "                    else:\n",
    "                        print(f'check {position}: HC - {sample_hc[position]}, DV - {dv_value}, {sample_pp.get(position, 'Not found in bam')}')\n",
    "                        if position in sample_pp.keys():\n",
    "                            pp_value=sample_pp[position][0]\n",
    "                            pp=pp_value.split('/')\n",
    "                            pp=int(pp[0])+int(pp[1])\n",
    "                            if pp==1 and (sample_pp[position][1]>=0.2 and sample_pp[position][1]<0.85):\n",
    "                                res[idx][position]  = hc_value\n",
    "                                print(f'HC genotype is supported by bam')\n",
    "                            if pp==2 and sample_pp[position][1]>=0.85:\n",
    "                                print(f'HC genotype is not supported by bam, saving a new genotype')\n",
    "                                print(sample_pp[position])\n",
    "                                res[idx][position] = pp_value\n",
    "                        else:\n",
    "                            print('need to check inheritance')\n",
    "                            inheritance = check_inheritance(idx, position, trios, gt_hc, gt_dv, gt_bam)\n",
    "                            if inheritance == True:\n",
    "                                print('saving this variant even with bad QUAL')\n",
    "                                res[idx][position]  = hc_value\n",
    "                            else:\n",
    "                                print('droping this variant, probably artefact')\n",
    "        else:\n",
    "            hc_value=sample_hc[position][0]\n",
    "            hc=hc_value.split('|')\n",
    "            hc=int(hc[0])+int(hc[1])\n",
    "            if hc == 2:\n",
    "                if sample_hc[position][1] >= 0.85 and sample_hc[position][2]>=30:\n",
    "                    res[idx][position]  = hc_value\n",
    "                else:\n",
    "                    print(idx)\n",
    "                    print(f'check {position}: HC - {sample_hc[position]}, DV - not called, bam - {sample_pp.get(position, 'Not found in bam')}')\n",
    "                    if position in sample_pp.keys():\n",
    "                        pp_value=sample_pp[position][0]\n",
    "                        pp=pp_value.split('/')\n",
    "                        pp=int(pp[0])+int(pp[1])\n",
    "                        if pp==2 and sample_pp[position][1]>=0.85:\n",
    "                            res[idx][position]  = hc_value\n",
    "                            print(f'genotype is supported by bam')\n",
    "                        if pp==1 and (sample_pp[position][1]>=0.2 and sample_pp[position][1]<0.85):\n",
    "                            print(f'HC genotype is not supported by bam, saving a new genotype')\n",
    "                            res[idx][position]  = pp_value\n",
    "                    else:\n",
    "                        print('need to check inheritance')\n",
    "                        inheritance = check_inheritance(idx, position, trios, gt_hc, gt_dv, gt_bam)\n",
    "                        if inheritance == True:\n",
    "                            print('saving this variant even with bad QUAL')\n",
    "                            res[idx][position]  = hc_value\n",
    "                        else:\n",
    "                            print('droping this variant, probably artefact')\n",
    "            if hc == 1:\n",
    "                if (sample_hc[position][1] >= 0.2 and sample_hc[position][1] < 0.85):\n",
    "                    res[idx][position]  = hc_value\n",
    "                else:             \n",
    "                    print(idx)\n",
    "                    print(f'check {position}: HC - {sample_hc[position]}, DV - not called, {sample_pp.get(position, 'Not found in bam')}')\n",
    "                    if position in sample_pp.keys():\n",
    "                        pp_value=sample_pp[position][0]\n",
    "                        pp=pp_value.split('/')\n",
    "                        pp=int(pp[0])+int(pp[1])\n",
    "                        if hc == pp and (sample_pp[position][1] >= 0.2 and sample_pp[position][1]<0.85):\n",
    "                            res[idx][position]  = hc_value\n",
    "                            print(f'HC genotype is supported by bam')\n",
    "                        else:\n",
    "                            print(f'HC genotype is not supported by bam, saving a new genotype')\n",
    "                            print(sample_pp[position])\n",
    "                            res[idx][position]  = pp_value\n",
    "                    else:\n",
    "                        print('need to check inheritance')\n",
    "                        inheritance = check_inheritance(idx, position, trios, gt_hc, gt_dv, gt_bam)\n",
    "                        if inheritance == True:\n",
    "                            print('saving this variant even with bad QUAL')\n",
    "                            res[idx][position]  = hc_value\n",
    "                        else:\n",
    "                            print('droping this variant, probably artefact')\n",
    "    for position in sample_dv.keys():\n",
    "        if position not in res[idx].keys():\n",
    "            print(f'{position} in {idx} found in DV, but not in HC, checking it')\n",
    "            dv_value=sample_dv[position][0]\n",
    "            dv=dv_value.split('/')\n",
    "            dv=int(dv[0])+int(dv[1])\n",
    "            if dv == 2:\n",
    "                if position in sample_pp.keys():\n",
    "                    if sample_pp[position][1]> 0.85:\n",
    "                        print(f'{position} not called by HC, saving it with {dv_value}, {sample_pp[position]}')\n",
    "                        res[idx][position] = dv_value\n",
    "                else:\n",
    "                    print(f'low qual {position} in {idx}, checking inheritance; variant info: {dv_value}')\n",
    "                    inheritance = check_inheritance(idx, position, trios, gt_hc, gt_dv, gt_bam)\n",
    "                    if inheritance == True:\n",
    "                        print('saving this variant even with bad QUAL')\n",
    "                        res[idx][position] = dv_value\n",
    "                    else:\n",
    "                        print('droping this variant, probably artefact')\n",
    "            if dv == 1:\n",
    "                if position in sample_pp.keys():\n",
    "                    if (sample_pp[position][1] >= 0.2 and sample_pp[position][1] < 0.85):\n",
    "                        print(f'{position} not called by HC, saving it with {dv_value}, {sample_pp[position]}')\n",
    "                        res[idx][position] = dv_value\n",
    "                else:\n",
    "                    print(f'low qual {position} in {idx}, checking inheritance; variant info: {dv_value}')\n",
    "                    inheritance = check_inheritance(idx, position, trios, gt_hc, gt_dv, gt_bam)\n",
    "                    if inheritance == True:\n",
    "                        print('saving this variant even with bad QUAL')\n",
    "                        res[idx][position] = dv_value\n",
    "                    else:\n",
    "                        print('droping this variant, probably artefact')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'father': 77092368, 'mother': 77092371}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trios[77092370]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mKeyboardInterrupt\u001b[39m                         Traceback (most recent call last)",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniforge3/envs/haplotypes/lib/python3.12/site-packages/IPython/core/async_helpers.py:128\u001b[39m, in \u001b[36m_pseudo_sync_runner\u001b[39m\u001b[34m(coro)\u001b[39m\n\u001b[32m    120\u001b[39m \u001b[38;5;250m\u001b[39m\u001b[33;03m\"\"\"\u001b[39;00m\n\u001b[32m    121\u001b[39m \u001b[33;03mA runner that does not really allow async execution, and just advance the coroutine.\u001b[39;00m\n\u001b[32m    122\u001b[39m \n\u001b[32m   (...)\u001b[39m\u001b[32m    125\u001b[39m \u001b[33;03mCredit to Nathaniel Smith\u001b[39;00m\n\u001b[32m    126\u001b[39m \u001b[33;03m\"\"\"\u001b[39;00m\n\u001b[32m    127\u001b[39m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[32m--> \u001b[39m\u001b[32m128\u001b[39m     \u001b[43mcoro\u001b[49m\u001b[43m.\u001b[49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[32m    129\u001b[39m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mStopIteration\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m exc:\n\u001b[32m    130\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m exc.value\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniforge3/envs/haplotypes/lib/python3.12/site-packages/IPython/core/interactiveshell.py:3256\u001b[39m, in \u001b[36mInteractiveShell.run_cell_async\u001b[39m\u001b[34m(self, raw_cell, store_history, silent, shell_futures, transformed_cell, preprocessing_exc_tuple, cell_id)\u001b[39m\n\u001b[32m   3254\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m store_history:\n\u001b[32m   3255\u001b[39m     \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m.history_manager \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[32m-> \u001b[39m\u001b[32m3256\u001b[39m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mhistory_manager\u001b[49m\u001b[43m.\u001b[49m\u001b[43mstore_inputs\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mexecution_count\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcell\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mraw_cell\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m   3257\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m silent:\n\u001b[32m   3258\u001b[39m     \u001b[38;5;28mself\u001b[39m.logger.log(cell, raw_cell)\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniforge3/envs/haplotypes/lib/python3.12/site-packages/IPython/core/history.py:951\u001b[39m, in \u001b[36mHistoryManager.store_inputs\u001b[39m\u001b[34m(self, line_num, source, source_raw)\u001b[39m\n\u001b[32m    948\u001b[39m \u001b[38;5;28mself\u001b[39m.input_hist_parsed.append(source)\n\u001b[32m    949\u001b[39m \u001b[38;5;28mself\u001b[39m.input_hist_raw.append(source_raw)\n\u001b[32m--> \u001b[39m\u001b[32m951\u001b[39m \u001b[43m\u001b[49m\u001b[38;5;28;43;01mwith\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mdb_input_cache_lock\u001b[49m\u001b[43m:\u001b[49m\n\u001b[32m    952\u001b[39m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mdb_input_cache\u001b[49m\u001b[43m.\u001b[49m\u001b[43mappend\u001b[49m\u001b[43m(\u001b[49m\u001b[43m(\u001b[49m\u001b[43mline_num\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msource\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msource_raw\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    953\u001b[39m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m# Trigger to flush cache and write to DB.\u001b[39;49;00m\n",
      "\u001b[31mKeyboardInterrupt\u001b[39m: "
     ]
    }
   ],
   "source": [
    "print(gt_hc[77092370]['32038949_T_G'], 'NONE', gt_dv[77092370]['32038949_T_G'], gt_dv[77092370]['32038949_T_C'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'32038115_G_C': ['1/1', 0.9778085991678225, 1442],\n",
       " '32038127_T_C': ['1/1', 0.9961180124223602, 1288],\n",
       " '32038128_A_C': ['1/1', 0.9992223950233281, 1286],\n",
       " '32038139_A_G': ['1/1', 0.996177370030581, 1308],\n",
       " '32038141_T_G': ['1/1', 1.0, 1365],\n",
       " '32038213_T_C': ['1/1', 0.9944328462073765, 1437],\n",
       " '32038224_C_T': ['1/1', 0.996395097332372, 1387],\n",
       " '32038297_C_T': ['1/1', 0.9927826784282278, 1247],\n",
       " '32038310_G_A': ['1/1', 0.9694589877835951, 1146],\n",
       " '32038313_T_C': ['1/1', 1.0, 1117],\n",
       " '32038320_A_G': ['1/1', 0.9873843566021867, 1189],\n",
       " '32038419_C_T': ['1/1', 0.9975339087546239, 811],\n",
       " '32038514_C_T': ['1/1', 0.9549653579676675, 866],\n",
       " '32038844_A_C': ['1/1', 0.8655083655083655, 1554],\n",
       " '32038857_A_T': ['1/1', 0.9485294117647058, 544],\n",
       " '32038867_T_G': ['1/1', 0.9952076677316294, 1252],\n",
       " '32038878_C_T': ['1/1', 0.9951923076923077, 1248],\n",
       " '32038895_A_G': ['1/1', 0.9974619289340102, 1182],\n",
       " '32038903_A_G': ['1/1', 0.9982062780269059, 1115],\n",
       " '32038911_A_G': ['1/1', 0.9943872778297475, 1069],\n",
       " '32038927_A_G': ['1/1', 0.9979338842975206, 968],\n",
       " '32038938_T_G': ['1/1', 0.9988479262672811, 868],\n",
       " '32038949_T_C': ['0/1', 0.7575757575757576, 33],\n",
       " '32038955_A_T': ['1/1', 1.0, 481],\n",
       " '32038971_C_A': ['1/1', 0.874828060522696, 727],\n",
       " '32038998_G_T': ['1/1', 1.0, 16],\n",
       " '32038999_G_C': ['1/1', 1.0, 17],\n",
       " '32039001_G_C': ['1/1', 0.9280575539568345, 278],\n",
       " '32039002_G_A': ['1/1', 0.925, 280],\n",
       " '32039015_G_T': ['1/1', 0.8982785602503912, 639],\n",
       " '32039020_G_A': ['1/1', 0.9498567335243553, 698],\n",
       " '32039027_C_A': ['1/1', 0.9505882352941176, 850],\n",
       " '32039046_A_G': ['1/1', 1.0, 907],\n",
       " '32039050_G_T': ['1/1', 1.0, 905],\n",
       " '32039055_C_G': ['1/1', 1.0, 909],\n",
       " '32039056_A_G': ['1/1', 0.9988925802879292, 903],\n",
       " '32039081_C_G': ['1/1', 0.9925925925925926, 1080],\n",
       " '32039109_G_A': ['1/1', 0.9621676891615542, 978],\n",
       " '32039128_C_G': ['1/1', 0.9812889812889813, 481],\n",
       " '32039143_C_T': ['1/1', 0.76, 225],\n",
       " '32039847_A_G': ['1/1', 0.9990272373540856, 1028],\n",
       " '32039848_C_T': ['1/1', 0.9950884086444007, 1018],\n",
       " '32040013_C_G': ['1/1', 0.9963369963369964, 819],\n",
       " '32040110_G_T': ['1/1', 0.9990990990990991, 1110],\n",
       " '32041186_C_T': ['1/1', 0.9588414634146342, 656]}"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gt_bam[77092898]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "for 77092312 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8401396160558464, DP-2868\n",
      "low qual 32038855_T_TTG in 77092312, checking inheritance; variant info: 1|0, 0.15986038394415358, 2868\n",
      "{'mother': 77092311, 'sibling': 77092313}\n",
      "32038855_T_TTG not found in gt_hc of 77092311 - the mother of 77092312\n",
      "32038855_T_TTG not found in gt_dv of 77092311 - the mother of 77092312\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038855_T_TTG in 77092329, checking inheritance; variant info: 1|0, 0.18454661558109833, 1581\n",
      "{'mother': 77092327}\n",
      "32038855_T_TTG not found in gt_hc of 77092327 - the mother of 77092329\n",
      "32038855_T_TTG not found in gt_dv of 77092327 - the mother of 77092329\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092329 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8154533844189017, DP-1581\n",
      "for 77092370 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.6842105263157895, DP-96\n",
      "for 77092370 comparing 32038949_T_G with HC - 1|1 and DV - 1/0\n",
      "dv right with 1/0, VAF - 0.8, DP-168\n",
      "for 77092370 comparing 32038962_A_ATCC with HC - 0|1 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.9246575342465754, DP-146\n",
      "low qual 32039001_G_C in 77092370, checking inheritance; variant info: 1|1, 1.0, 13\n",
      "{'father': 77092368, 'mother': 77092371}\n",
      "found 32039001_G_C in 77092368 HC data - ['1|0', 0.5068493150684932, 73]\n",
      "32039001_G_C found only in HC data of father\n",
      "found a heterozygote 32039001_G_C in father with VAF 0.5068493150684932\n",
      "found 32039001_G_C in 77092371 HC data - ['1|0', 0.4712892741061755, 2769]\n",
      "32039001_G_C found only in HC data of mother\n",
      "found a heterozygote 32039001_G_C in mother with VAF 0.4712892741061755\n",
      "['1|0', '1|0']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039006_G_A in 77092377, checking inheritance; variant info: 0|1, 0.16883116883116883, 77\n",
      "{'mother': 77092375, 'sibling': 77092376}\n",
      "32039006_G_A not found in gt_hc of 77092375 - the mother of 77092377\n",
      "32039006_G_A not found in gt_dv of 77092375 - the mother of 77092377\n",
      "no 32039006_G_A found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039007_A_AAG in 77092384, checking inheritance; variant info: 1|0, 0.75, 20\n",
      "{'child': 77092383, 'spouse': 77092386}\n",
      "found 32039007_A_AAG in 77092383 HC data - ['0|1', 0.6153846153846154, 130]\n",
      "32039007_A_AAG found only in HC data of child\n",
      "found a heterozygote 32039007_A_AAG in child with VAF 0.6153846153846154\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039015_G_T in 77092384, checking inheritance; variant info: 1|0, 0.75, 20\n",
      "{'child': 77092383, 'spouse': 77092386}\n",
      "found 32039015_G_T in 77092383 HC data - ['0|1', 0.5894039735099338, 151]\n",
      "32039015_G_T found only in HC data of child\n",
      "found a heterozygote 32039015_G_T in child with VAF 0.5894039735099338\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039020_G_A in 77092384, checking inheritance; variant info: 1|0, 0.75, 20\n",
      "{'child': 77092383, 'spouse': 77092386}\n",
      "found 32039020_G_A in 77092383 HC data - ['0|1', 0.5947712418300654, 153]\n",
      "32039020_G_A found only in HC data of child\n",
      "found a heterozygote 32039020_G_A in child with VAF 0.5947712418300654\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039046_A_G in 77092384, checking inheritance; variant info: 1|0, 0.8666666666666667, 15\n",
      "{'child': 77092383, 'spouse': 77092386}\n",
      "found 32039046_A_G in 77092383 HC data - ['0|1', 0.553072625698324, 179]\n",
      "checking relative variant with ['0|1', 0.553072625698324, 179] AND ['0/1', 0.47282608695652173, 184]\n",
      "found 32039046_A_G in child with genotype 0|1\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039050_G_T in 77092384, checking inheritance; variant info: 1|0, 0.6071428571428571, 28\n",
      "{'child': 77092383, 'spouse': 77092386}\n",
      "found 32039050_G_T in 77092383 HC data - ['0|1', 0.4327731092436975, 238]\n",
      "32039050_G_T found only in HC data of child\n",
      "found a heterozygote 32039050_G_T in child with VAF 0.4327731092436975\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039056_A_G in 77092384, checking inheritance; variant info: 1|0, 0.6206896551724138, 29\n",
      "{'child': 77092383, 'spouse': 77092386}\n",
      "found 32039056_A_G in 77092383 HC data - ['0|1', 0.43983402489626555, 241]\n",
      "checking relative variant with ['0|1', 0.43983402489626555, 241] AND ['0/1', 0.4189189189189189, 222]\n",
      "found 32039056_A_G in child with genotype 0|1\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "for 77092388 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8064516129032258, DP-97\n",
      "low qual 32038855_T_TTG in 77092388, checking inheritance; variant info: 1|0, 0.1935483870967742, 97\n",
      "{'mother': 77092391, 'sibling': 77092392, 'father': 77092389}\n",
      "32038855_T_TTG not found in gt_hc of 77092391 - the mother of 77092388\n",
      "32038855_T_TTG not found in gt_dv of 77092391 - the mother of 77092388\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092389 - the father of 77092388\n",
      "32038855_T_TTG not found in gt_dv of 77092389 - the father of 77092388\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092388 comparing 32038949_T_G with HC - 1|1 and DV - 1/0\n",
      "hc right with 1|1, VAF - 1.0, DP-161\n",
      "low qual 32038949_T_C in 77092388, checking inheritance; variant info: 1/0, 0.1724137931034483, 146\n",
      "{'mother': 77092391, 'sibling': 77092392, 'father': 77092389}\n",
      "32038949_T_C not found in gt_hc of 77092391 - the mother of 77092388\n",
      "32038949_T_C not found in gt_dv of 77092391 - the mother of 77092388\n",
      "no 32038949_T_C found in mother\n",
      "32038949_T_C not found in gt_hc of 77092389 - the father of 77092388\n",
      "32038949_T_C not found in gt_dv of 77092389 - the father of 77092388\n",
      "no 32038949_T_C found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038855_T_TTG in 77092392, checking inheritance; variant info: 1|0, 0.1889763779527559, 129\n",
      "{'mother': 77092391, 'sibling': 77092388, 'father': 77092389}\n",
      "32038855_T_TTG not found in gt_hc of 77092391 - the mother of 77092392\n",
      "32038855_T_TTG not found in gt_dv of 77092391 - the mother of 77092392\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092389 - the father of 77092392\n",
      "32038855_T_TTG not found in gt_dv of 77092389 - the father of 77092392\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092392 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8110236220472441, DP-129\n",
      "low qual 32039006_G_A in 77092393, checking inheritance; variant info: 1|0, 0.08333333333333333, 72\n",
      "{'mother': 77092394, 'father': 77092395}\n",
      "32039006_G_A not found in gt_hc of 77092394 - the mother of 77092393\n",
      "32039006_G_A not found in gt_dv of 77092394 - the mother of 77092393\n",
      "no 32039006_G_A found in mother\n",
      "32039006_G_A not found in gt_hc of 77092395 - the father of 77092393\n",
      "32039006_G_A not found in gt_dv of 77092395 - the father of 77092393\n",
      "no 32039006_G_A found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038515_G_A in 77092398, checking inheritance; variant info: 1|0, 0.10059171597633136, 169\n",
      "{'mother': 77092396, 'sibling': 77092397, 'father': 77092399}\n",
      "32038515_G_A not found in gt_hc of 77092396 - the mother of 77092398\n",
      "32038515_G_A not found in gt_dv of 77092396 - the mother of 77092398\n",
      "no 32038515_G_A found in mother\n",
      "32038515_G_A not found in gt_hc of 77092399 - the father of 77092398\n",
      "32038515_G_A not found in gt_dv of 77092399 - the father of 77092398\n",
      "no 32038515_G_A found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092401 comparing 32038949_T_G with HC - 1|1 and DV - 1/0\n",
      "hc right with 1|1, VAF - 1.0, DP-120\n",
      "low qual 32038855_T_TTTG in 77092401, checking inheritance; variant info: 1/1, 0.5949367088607594, 237\n",
      "{'father': 77092400, 'mother': 77092402, 'sibling': 77092403}\n",
      "found 32038855_T_TTTG in 77092400 HC data - ['0|1', 0.42424242424242425, 165]\n",
      "32038855_T_TTTG found only in HC data of father\n",
      "found a heterozygote 32038855_T_TTTG in father with VAF 0.42424242424242425\n",
      "found 32038855_T_TTTG in 77092402 HC data - ['0|1', 0.41420118343195267, 169]\n",
      "32038855_T_TTTG found only in HC data of mother\n",
      "found a heterozygote 32038855_T_TTTG in mother with VAF 0.41420118343195267\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038949_T_C in 77092401, checking inheritance; variant info: 1/0, 0.12738853503184713, 159\n",
      "{'father': 77092400, 'mother': 77092402, 'sibling': 77092403}\n",
      "32038949_T_C not found in gt_hc of 77092400 - the father of 77092401\n",
      "32038949_T_C not found in gt_dv of 77092400 - the father of 77092401\n",
      "no 32038949_T_C found in father\n",
      "32038949_T_C not found in gt_hc of 77092402 - the mother of 77092401\n",
      "32038949_T_C not found in gt_dv of 77092402 - the mother of 77092401\n",
      "no 32038949_T_C found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092417 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8213507625272332, DP-1383\n",
      "low qual 32038855_T_TTG in 77092417, checking inheritance; variant info: 1|0, 0.1786492374727669, 1383\n",
      "{'mother': 77092416, 'father': 77092415}\n",
      "32038855_T_TTG not found in gt_hc of 77092416 - the mother of 77092417\n",
      "32038855_T_TTG not found in gt_dv of 77092416 - the mother of 77092417\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092415 - the father of 77092417\n",
      "32038855_T_TTG not found in gt_dv of 77092415 - the father of 77092417\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038855_T_TTG in 77092418, checking inheritance; variant info: 1|0, 0.18398268398268397, 1871\n",
      "{'mother': 77092428, 'father': 77092426}\n",
      "32038855_T_TTG not found in gt_hc of 77092428 - the mother of 77092418\n",
      "32038855_T_TTG not found in gt_dv of 77092428 - the mother of 77092418\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092426 - the father of 77092418\n",
      "32038855_T_TTG not found in gt_dv of 77092426 - the father of 77092418\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092418 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.816017316017316, DP-1871\n",
      "low qual 32038855_T_TTG in 77092436, checking inheritance; variant info: 1|0, 0.17109634551495018, 1214\n",
      "{'father': 77092438, 'mother': 77092437}\n",
      "32038855_T_TTG not found in gt_hc of 77092438 - the father of 77092436\n",
      "32038855_T_TTG not found in gt_dv of 77092438 - the father of 77092436\n",
      "no 32038855_T_TTG found in father\n",
      "32038855_T_TTG not found in gt_hc of 77092437 - the mother of 77092436\n",
      "32038855_T_TTG not found in gt_dv of 77092437 - the mother of 77092436\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092436 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8289036544850499, DP-1214\n",
      "low qual 32038878_C_T in 77092437, checking inheritance; variant info: 0|1, 0.17341040462427745, 519\n",
      "{'spouse': 77092438, 'child': 77092436}\n",
      "found 32038878_C_T in 77092436 HC data - ['1|1', 0.9966969446738233, 1211]\n",
      "checking relative variant with ['1|1', 0.9966969446738233, 1211] AND ['1/1', 1.0, 271]\n",
      "found 32038878_C_T in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038895_A_G in 77092437, checking inheritance; variant info: 0|1, 0.14285714285714285, 469\n",
      "{'spouse': 77092438, 'child': 77092436}\n",
      "found 32038895_A_G in 77092436 HC data - ['1|1', 1.0, 1161]\n",
      "checking relative variant with ['1|1', 1.0, 1161] AND ['1/1', 1.0, 261]\n",
      "found 32038895_A_G in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038855_T_TTG in 77092450, checking inheritance; variant info: 1|0, 0.17721518987341772, 2449\n",
      "{'mother': 77092449, 'father': 77092448}\n",
      "32038855_T_TTG not found in gt_hc of 77092449 - the mother of 77092450\n",
      "32038855_T_TTG not found in gt_dv of 77092449 - the mother of 77092450\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092448 - the father of 77092450\n",
      "32038855_T_TTG not found in gt_dv of 77092448 - the father of 77092450\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092450 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8227848101265823, DP-2449\n",
      "low qual 32038855_T_TTG in 77092472, checking inheritance; variant info: 1|0, 0.16150576806314512, 1661\n",
      "{'mother': 77092473, 'father': 77092474}\n",
      "32038855_T_TTG not found in gt_hc of 77092473 - the mother of 77092472\n",
      "32038855_T_TTG not found in gt_dv of 77092473 - the mother of 77092472\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092474 - the father of 77092472\n",
      "32038855_T_TTG not found in gt_dv of 77092474 - the father of 77092472\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038855_T_TTG in 77092500, checking inheritance; variant info: 1|0, 0.18298653610771115, 1639\n",
      "{'mother': 77092501, 'father': 77092502}\n",
      "32038855_T_TTG not found in gt_hc of 77092501 - the mother of 77092500\n",
      "32038855_T_TTG not found in gt_dv of 77092501 - the mother of 77092500\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092502 - the father of 77092500\n",
      "32038855_T_TTG not found in gt_dv of 77092502 - the father of 77092500\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092500 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8170134638922889, DP-1639\n",
      "low qual 32038903_A_G in 77092529, checking inheritance; variant info: 0|1, 0.12, 25\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038903_A_G not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32038903_A_G in 77092531 DV data - ['1/1', 1.0, 137]\n",
      "32038903_A_G found only in DV data of child\n",
      "found a homozygote 32038903_A_G in child with VAF 1.0\n",
      "found only low qual 32038903_A_G in child with 1.0 and 137, probably an artefact\n",
      "['1/1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038948_T_TC in 77092529, checking inheritance; variant info: 0|1, 0.18518518518518517, 27\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038948_T_TC not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32038948_T_TC in 77092531 DV data - ['1/1', 0.9016393442622951, 61]\n",
      "32038948_T_TC found only in DV data of child\n",
      "found a homozygote 32038948_T_TC in child with VAF 0.9016393442622951\n",
      "found only low qual 32038948_T_TC in child with 0.9016393442622951 and 61, probably an artefact\n",
      "['1/1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038949_T_G in 77092529, checking inheritance; variant info: 0|1, 0.17857142857142858, 28\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038949_T_G not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32038949_T_G in 77092531 DV data - ['1/1', 1.0, 55]\n",
      "32038949_T_G found only in DV data of child\n",
      "found a homozygote 32038949_T_G in child with VAF 1.0\n",
      "found only low qual 32038949_T_G in child with 1.0 and 55, probably an artefact\n",
      "['1/1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038955_A_T in 77092529, checking inheritance; variant info: 0|1, 0.16666666666666666, 30\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038955_A_T not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32038955_A_T in 77092531 DV data - ['1/1', 1.0, 59]\n",
      "32038955_A_T found only in DV data of child\n",
      "found a homozygote 32038955_A_T in child with VAF 1.0\n",
      "found only low qual 32038955_A_T in child with 1.0 and 59, probably an artefact\n",
      "['1/1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038962_A_ATCC in 77092529, checking inheritance; variant info: 0|1, 0.19230769230769232, 26\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038962_A_ATCC not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32038962_A_ATCC in 77092531 DV data - ['1/1', 0.9333333333333333, 60]\n",
      "32038962_A_ATCC found only in DV data of child\n",
      "found a homozygote 32038962_A_ATCC in child with VAF 0.9333333333333333\n",
      "found only low qual 32038962_A_ATCC in child with 0.9333333333333333 and 60, probably an artefact\n",
      "['1/1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038971_C_A in 77092529, checking inheritance; variant info: 0|1, 0.19230769230769232, 26\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038971_C_A not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32038971_C_A in 77092531 DV data - ['1/1', 0.875, 88]\n",
      "32038971_C_A found only in DV data of child\n",
      "found a homozygote 32038971_C_A in child with VAF 0.875\n",
      "found only low qual 32038971_C_A in child with 0.875 and 88, probably an artefact\n",
      "['1/1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038983_TG_T in 77092529, checking inheritance; variant info: 0|1, 0.125, 24\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038983_TG_T not found in gt_hc of 77092531 - the child of 77092529\n",
      "32038983_TG_T not found in gt_dv of 77092531 - the child of 77092529\n",
      "no 32038983_TG_T found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038997_TGG_T in 77092529, checking inheritance; variant info: 0|1, 0.11538461538461539, 26\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32038997_TGG_T not found in gt_hc of 77092531 - the child of 77092529\n",
      "32038997_TGG_T not found in gt_dv of 77092531 - the child of 77092529\n",
      "no 32038997_TGG_T found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039001_G_C in 77092529, checking inheritance; variant info: 0|1, 0.125, 24\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32039001_G_C not found in gt_hc of 77092531 - the child of 77092529\n",
      "32039001_G_C not found in gt_dv of 77092531 - the child of 77092529\n",
      "no 32039001_G_C found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039002_G_A in 77092529, checking inheritance; variant info: 0|1, 0.125, 24\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32039002_G_A not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32039002_G_A in 77092531 DV data - ['1/1', 0.9487179487179487, 39]\n",
      "32039002_G_A found only in DV data of child\n",
      "found a homozygote 32039002_G_A in child with VAF 0.9487179487179487\n",
      "found only low qual 32039002_G_A in child with 0.9487179487179487 and 39, probably an artefact\n",
      "['1/1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039007_A_AAG in 77092529, checking inheritance; variant info: 0|1, 0.24, 25\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32039007_A_AAG not found in gt_hc of 77092531 - the child of 77092529\n",
      "32039007_A_AAG not found in gt_dv of 77092531 - the child of 77092529\n",
      "no 32039007_A_AAG found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039015_G_T in 77092529, checking inheritance; variant info: 0|1, 0.24, 25\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32039015_G_T not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32039015_G_T in 77092531 DV data - ['1/1', 0.9270833333333334, 96]\n",
      "32039015_G_T found only in DV data of child\n",
      "found a homozygote 32039015_G_T in child with VAF 0.9270833333333334\n",
      "found only low qual 32039015_G_T in child with 0.9270833333333334 and 96, probably an artefact\n",
      "['1/1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039020_G_A in 77092529, checking inheritance; variant info: 0|1, 0.2608695652173913, 23\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32039020_G_A not found in gt_hc of 77092531 - the child of 77092529\n",
      "32039020_G_A not found in gt_dv of 77092531 - the child of 77092529\n",
      "no 32039020_G_A found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039027_C_A in 77092529, checking inheritance; variant info: 0|1, 0.2727272727272727, 22\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32039027_C_A not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32039027_C_A in 77092531 DV data - ['1/1', 0.9495798319327731, 119]\n",
      "32039027_C_A found only in DV data of child\n",
      "found a homozygote 32039027_C_A in child with VAF 0.9495798319327731\n",
      "found only low qual 32039027_C_A in child with 0.9495798319327731 and 119, probably an artefact\n",
      "['1/1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039046_A_G in 77092529, checking inheritance; variant info: 0|1, 0.3076923076923077, 26\n",
      "{'spouse': 77092530, 'child': 77092531}\n",
      "32039046_A_G not found in gt_hc of 77092531 - the child of 77092529\n",
      "found 32039046_A_G in 77092531 DV data - ['1/1', 0.9921875, 128]\n",
      "32039046_A_G found only in DV data of child\n",
      "found a homozygote 32039046_A_G in child with VAF 0.9921875\n",
      "found only low qual 32039046_A_G in child with 0.9921875 and 128, probably an artefact\n",
      "['1/1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038983_TG_T in 77092530, checking inheritance; variant info: 0|1, 0.1875, 32\n",
      "{'spouse': 77092529, 'child': 77092531}\n",
      "32038983_TG_T not found in gt_hc of 77092531 - the child of 77092530\n",
      "32038983_TG_T not found in gt_dv of 77092531 - the child of 77092530\n",
      "no 32038983_TG_T found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092531 comparing 32038139_A_G with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.9793103448275862, DP-145\n",
      "for 77092531 comparing 32038141_T_G with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.9741935483870968, DP-155\n",
      "for 77092531 comparing 32038844_A_C with HC - 1|0 and DV - 1/1\n",
      "both wrong, checking inheritance; variant info: 1|0, 0.38461538461538464, 13 AND 1/1, 0.8266666666666667, 225\n",
      "{'father': 77092530, 'mother': 77092529}\n",
      "found 32038844_A_C in 77092530 HC data - ['1|1', 0.9821428571428571, 112]\n",
      "checking relative variant with ['1|1', 0.9821428571428571, 112] AND ['1/1', 0.8554216867469879, 249]\n",
      "found 32038844_A_C in father with genotype 1|1\n",
      "found 32038844_A_C in 77092529 HC data - ['0|1', 0.3813559322033898, 118]\n",
      "checking relative variant with ['0|1', 0.3813559322033898, 118] AND ['0/1', 0.33884297520661155, 242]\n",
      "found 32038844_A_C in mother with genotype 0|1\n",
      "['1|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038847_A_ATTT in 77092531, checking inheritance; variant info: 1|1, 0.8, 10\n",
      "{'father': 77092530, 'mother': 77092529}\n",
      "32038847_A_ATTT not found in gt_hc of 77092530 - the father of 77092531\n",
      "32038847_A_ATTT not found in gt_dv of 77092530 - the father of 77092531\n",
      "no 32038847_A_ATTT found in father\n",
      "32038847_A_ATTT not found in gt_hc of 77092529 - the mother of 77092531\n",
      "32038847_A_ATTT not found in gt_dv of 77092529 - the mother of 77092531\n",
      "no 32038847_A_ATTT found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039081_C_A in 77092531, checking inheritance; variant info: 1|1, 1.0, 9\n",
      "{'father': 77092530, 'mother': 77092529}\n",
      "found 32039081_C_A in 77092530 HC data - ['1|0', 0.5331325301204819, 332]\n",
      "checking relative variant with ['1|0', 0.5331325301204819, 332] AND ['1/0', 0.5821917808219178, 146]\n",
      "found 32039081_C_A in father with genotype 1|0\n",
      "found 32039081_C_A in 77092529 HC data - ['1|0', 0.8709677419354839, 93]\n",
      "checking relative variant with ['1|0', 0.8709677419354839, 93] AND ['1/0', 0.5384615384615384, 144]\n",
      "found 32039081_C_A in mother with genotype 1|0\n",
      "['1|0', '1|0']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038855_T_TTTG in 77092531, checking inheritance; variant info: 1/1, 0.4639175257731959, 194\n",
      "{'father': 77092530, 'mother': 77092529}\n",
      "found 32038855_T_TTTG in 77092530 HC data - ['0|1', 0.40540540540540543, 111]\n",
      "checking relative variant with ['0|1', 0.40540540540540543, 111] AND ['1/0', 0.7575757575757576, 241]\n",
      "found 32038855_T_TTTG in father with genotype 0|1\n",
      "found 32038855_T_TTTG in 77092529 HC data - ['0|1', 0.3805309734513274, 113]\n",
      "32038855_T_TTTG found only in HC data of mother\n",
      "found a heterozygote 32038855_T_TTTG in mother with VAF 0.3805309734513274\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "for 77092537 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8281437125748503, DP-1682\n",
      "low qual 32038855_T_TTG in 77092537, checking inheritance; variant info: 1|0, 0.1718562874251497, 1682\n",
      "{'mother': 77092536, 'father': 77092535}\n",
      "32038855_T_TTG not found in gt_hc of 77092536 - the mother of 77092537\n",
      "32038855_T_TTG not found in gt_dv of 77092536 - the mother of 77092537\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092535 - the father of 77092537\n",
      "32038855_T_TTG not found in gt_dv of 77092535 - the father of 77092537\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092543 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8343023255813954, DP-700\n",
      "low qual 32038855_T_TTG in 77092543, checking inheritance; variant info: 1|0, 0.16569767441860464, 700\n",
      "{'mother': 77092542, 'father': 77092541}\n",
      "32038855_T_TTG not found in gt_hc of 77092542 - the mother of 77092543\n",
      "32038855_T_TTG not found in gt_dv of 77092542 - the mother of 77092543\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092541 - the father of 77092543\n",
      "32038855_T_TTG not found in gt_dv of 77092541 - the father of 77092543\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038971_C_A in 77092563, checking inheritance; variant info: 1|0, 0.41379310344827586, 29\n",
      "{'sibling': 77092564, 'mother': 77092565}\n",
      "32038971_C_A not found in gt_hc of 77092565 - the mother of 77092563\n",
      "32038971_C_A not found in gt_dv of 77092565 - the mother of 77092563\n",
      "no 32038971_C_A found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038983_TG_T in 77092563, checking inheritance; variant info: 1|0, 0.45454545454545453, 22\n",
      "{'sibling': 77092564, 'mother': 77092565}\n",
      "32038983_TG_T not found in gt_hc of 77092565 - the mother of 77092563\n",
      "32038983_TG_T not found in gt_dv of 77092565 - the mother of 77092563\n",
      "no 32038983_TG_T found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038997_TGG_T in 77092563, checking inheritance; variant info: 1|0, 0.6071428571428571, 28\n",
      "{'sibling': 77092564, 'mother': 77092565}\n",
      "32038997_TGG_T not found in gt_hc of 77092565 - the mother of 77092563\n",
      "32038997_TGG_T not found in gt_dv of 77092565 - the mother of 77092563\n",
      "no 32038997_TGG_T found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039001_G_C in 77092563, checking inheritance; variant info: 1|0, 0.5862068965517241, 29\n",
      "{'sibling': 77092564, 'mother': 77092565}\n",
      "32039001_G_C not found in gt_hc of 77092565 - the mother of 77092563\n",
      "32039001_G_C not found in gt_dv of 77092565 - the mother of 77092563\n",
      "no 32039001_G_C found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039002_G_A in 77092563, checking inheritance; variant info: 1|0, 0.5862068965517241, 29\n",
      "{'sibling': 77092564, 'mother': 77092565}\n",
      "32039002_G_A not found in gt_hc of 77092565 - the mother of 77092563\n",
      "32039002_G_A not found in gt_dv of 77092565 - the mother of 77092563\n",
      "no 32039002_G_A found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038948_T_TC in 77092564, checking inheritance; variant info: 0|1, 0.14285714285714285, 140\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32038948_T_TC not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32038948_T_TC not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32038948_T_TC found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038949_T_G in 77092564, checking inheritance; variant info: 0|1, 0.14285714285714285, 140\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32038949_T_G not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32038949_T_G not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32038949_T_G found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038955_A_T in 77092564, checking inheritance; variant info: 0|1, 0.14814814814814814, 135\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32038955_A_T not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32038955_A_T not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32038955_A_T found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038962_A_ATCC in 77092564, checking inheritance; variant info: 0|1, 0.08620689655172414, 58\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32038962_A_ATCC not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32038962_A_ATCC not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32038962_A_ATCC found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038983_TG_T in 77092564, checking inheritance; variant info: 0|1, 0.1111111111111111, 36\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32038983_TG_T not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32038983_TG_T not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32038983_TG_T found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038997_TGG_T in 77092564, checking inheritance; variant info: 0|1, 0.1282051282051282, 39\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32038997_TGG_T not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32038997_TGG_T not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32038997_TGG_T found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039001_G_C in 77092564, checking inheritance; variant info: 0|1, 0.15151515151515152, 33\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32039001_G_C not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32039001_G_C not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32039001_G_C found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039002_G_A in 77092564, checking inheritance; variant info: 0|1, 0.15151515151515152, 33\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32039002_G_A not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32039002_G_A not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32039002_G_A found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039020_G_A in 77092564, checking inheritance; variant info: 0|1, 0.3448275862068966, 29\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32039020_G_A not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32039020_G_A not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32039020_G_A found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039027_C_A in 77092564, checking inheritance; variant info: 0|1, 0.3448275862068966, 29\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32039027_C_A not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32039027_C_A not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32039027_C_A found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039046_A_G in 77092564, checking inheritance; variant info: 0|1, 0.5185185185185185, 27\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32039046_A_G not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32039046_A_G not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32039046_A_G found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039050_G_T in 77092564, checking inheritance; variant info: 0|1, 0.6153846153846154, 26\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32039050_G_T not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32039050_G_T not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32039050_G_T found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039055_C_G in 77092564, checking inheritance; variant info: 0|1, 0.68, 25\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32039055_C_G not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32039055_C_G not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32039055_C_G found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039056_A_G in 77092564, checking inheritance; variant info: 0|1, 0.7083333333333334, 24\n",
      "{'sibling': 77092563, 'mother': 77092565}\n",
      "32039056_A_G not found in gt_hc of 77092565 - the mother of 77092564\n",
      "32039056_A_G not found in gt_dv of 77092565 - the mother of 77092564\n",
      "no 32039056_A_G found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038855_T_TTG in 77092571, checking inheritance; variant info: 1|0, 0.14919144496609285, 1917\n",
      "{'mother': 77092570}\n",
      "32038855_T_TTG not found in gt_hc of 77092570 - the mother of 77092571\n",
      "32038855_T_TTG not found in gt_dv of 77092570 - the mother of 77092571\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092571 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.8686131386861314, DP-137\n",
      "low qual 32038855_T_TTG in 77092572, checking inheritance; variant info: 1|0, 0.15139442231075698, 2008\n",
      "{'mother': 77092573}\n",
      "32038855_T_TTG not found in gt_hc of 77092573 - the mother of 77092572\n",
      "32038855_T_TTG not found in gt_dv of 77092573 - the mother of 77092572\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092572 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.848605577689243, DP-2008\n",
      "low qual 32038855_T_TTG in 77092576, checking inheritance; variant info: 1|0, 0.14457202505219208, 1926\n",
      "{'mother': 77092574, 'sibling': 77092575}\n",
      "32038855_T_TTG not found in gt_hc of 77092574 - the mother of 77092576\n",
      "32038855_T_TTG not found in gt_dv of 77092574 - the mother of 77092576\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092576 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "both wrong, checking inheritance; variant info: 1|0, 0.855427974947808, 1926 AND 1/1, 0.8248175182481752, 137\n",
      "{'mother': 77092574, 'sibling': 77092575}\n",
      "found 32038855_T_TTTG in 77092574 HC data - ['0|1', 0.4636604774535809, 1885]\n",
      "checking relative variant with ['0|1', 0.4636604774535809, 1885] AND ['0/1', 0.3310810810810811, 148]\n",
      "found 32038855_T_TTTG in mother with genotype 0|1\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "for 77092576 comparing 32041186_C_T with HC - 1|1 and DV - 0/1\n",
      "hc right with 1|1, VAF - 0.9994941831057157, DP-1977\n",
      "low qual 32038855_T_TTG in 77092577, checking inheritance; variant info: 1|0, 0.12160694896851248, 1853\n",
      "{'father': 77092579, 'sibling': 77092580, 'mother': 77092578}\n",
      "32038855_T_TTG not found in gt_hc of 77092579 - the father of 77092577\n",
      "32038855_T_TTG not found in gt_dv of 77092579 - the father of 77092577\n",
      "no 32038855_T_TTG found in father\n",
      "32038855_T_TTG not found in gt_hc of 77092578 - the mother of 77092577\n",
      "32038855_T_TTG not found in gt_dv of 77092578 - the mother of 77092577\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092577 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.8561151079136691, DP-139\n",
      "for 77092583 comparing 32038985_G_C with HC - 0|1 and DV - 1/1\n",
      "hc right with 0|1, VAF - 0.54411161263849, DP-2437\n",
      "low qual 32038855_T_TTG in 77092584, checking inheritance; variant info: 1|0, 0.1755666538609297, 2637\n",
      "{'mother': 77092583}\n",
      "32038855_T_TTG not found in gt_hc of 77092583 - the mother of 77092584\n",
      "32038855_T_TTG not found in gt_dv of 77092583 - the mother of 77092584\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092584 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8244333461390703, DP-2637\n",
      "low qual 32038855_T_TTG in 77092593, checking inheritance; variant info: 1|0, 0.1286549707602339, 172\n",
      "{'mother': 77092591, 'father': 77092592, 'sibling': 77092594}\n",
      "32038855_T_TTG not found in gt_hc of 77092591 - the mother of 77092593\n",
      "32038855_T_TTG not found in gt_dv of 77092591 - the mother of 77092593\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092592 - the father of 77092593\n",
      "32038855_T_TTG not found in gt_dv of 77092592 - the father of 77092593\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092593 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "both wrong, checking inheritance; variant info: 1|0, 0.8713450292397661, 172 AND 1/1, 0.634453781512605, 238\n",
      "{'mother': 77092591, 'father': 77092592, 'sibling': 77092594}\n",
      "found 32038855_T_TTTG in 77092591 HC data - ['0|1', 0.3825136612021858, 183]\n",
      "checking relative variant with ['0|1', 0.3825136612021858, 183] AND ['0/1', 0.26557377049180325, 305]\n",
      "found 32038855_T_TTTG in mother with genotype 0|1\n",
      "found 32038855_T_TTTG in 77092592 HC data - ['1|0', 0.45850914205344584, 2844]\n",
      "checking relative variant with ['1|0', 0.45850914205344584, 2844] AND ['0/1', 0.33881578947368424, 304]\n",
      "found 32038855_T_TTTG in father with genotype 1|0\n",
      "['0|1', '1|0']\n",
      "saving this variant even with bad QUAL\n",
      "for 77092594 comparing 32038949_T_G with HC - 1|1 and DV - 1/0\n",
      "hc right with 1|1, VAF - 1.0, DP-105\n",
      "low qual 32038949_T_C in 77092594, checking inheritance; variant info: 1/0, 0.145985401459854, 141\n",
      "{'mother': 77092591, 'father': 77092592, 'sibling': 77092595}\n",
      "32038949_T_C not found in gt_hc of 77092591 - the mother of 77092594\n",
      "32038949_T_C not found in gt_dv of 77092591 - the mother of 77092594\n",
      "no 32038949_T_C found in mother\n",
      "32038949_T_C not found in gt_hc of 77092592 - the father of 77092594\n",
      "32038949_T_C not found in gt_dv of 77092592 - the father of 77092594\n",
      "no 32038949_T_C found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038855_T_TTG in 77092595, checking inheritance; variant info: 1|0, 0.19863013698630136, 149\n",
      "{'mother': 77092591, 'father': 77092592, 'sibling': 77092594}\n",
      "32038855_T_TTG not found in gt_hc of 77092591 - the mother of 77092595\n",
      "32038855_T_TTG not found in gt_dv of 77092591 - the mother of 77092595\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092592 - the father of 77092595\n",
      "32038855_T_TTG not found in gt_dv of 77092592 - the father of 77092595\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092595 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8013698630136986, DP-149\n",
      "for 77092595 comparing 32038949_T_G with HC - 1|1 and DV - 1/0\n",
      "hc right with 1|1, VAF - 1.0, DP-133\n",
      "low qual 32038949_T_C in 77092595, checking inheritance; variant info: 1/0, 0.16216216216216217, 149\n",
      "{'mother': 77092591, 'father': 77092592, 'sibling': 77092594}\n",
      "32038949_T_C not found in gt_hc of 77092591 - the mother of 77092595\n",
      "32038949_T_C not found in gt_dv of 77092591 - the mother of 77092595\n",
      "no 32038949_T_C found in mother\n",
      "32038949_T_C not found in gt_hc of 77092592 - the father of 77092595\n",
      "32038949_T_C not found in gt_dv of 77092592 - the father of 77092595\n",
      "no 32038949_T_C found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038855_T_TTG in 77092609, checking inheritance; variant info: 1|0, 0.15171650055370986, 1862\n",
      "{'mother': 77092608, 'father': 77092607}\n",
      "32038855_T_TTG not found in gt_hc of 77092608 - the mother of 77092609\n",
      "32038855_T_TTG not found in gt_dv of 77092608 - the mother of 77092609\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092607 - the father of 77092609\n",
      "32038855_T_TTG not found in gt_dv of 77092607 - the father of 77092609\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092609 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8482834994462901, DP-1862\n",
      "for 77092617 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8250123578843302, DP-2047\n",
      "low qual 32038855_T_TTG in 77092617, checking inheritance; variant info: 1|0, 0.1749876421156698, 2047\n",
      "{'mother': 77092618}\n",
      "32038855_T_TTG not found in gt_hc of 77092618 - the mother of 77092617\n",
      "32038855_T_TTG not found in gt_dv of 77092618 - the mother of 77092617\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039007_A_AAG in 77092627, checking inheritance; variant info: 0|1, 0.44, 25\n",
      "{'child': 77092646}\n",
      "found 32039007_A_AAG in 77092646 HC data - ['1|1', 1.0, 40]\n",
      "checking relative variant with ['1|1', 1.0, 40] AND ['1/1', 0.978021978021978, 91]\n",
      "found 32039007_A_AAG in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039015_G_T in 77092627, checking inheritance; variant info: 0|1, 0.48148148148148145, 27\n",
      "{'child': 77092646}\n",
      "found 32039015_G_T in 77092646 HC data - ['1|1', 1.0, 35]\n",
      "checking relative variant with ['1|1', 1.0, 35] AND ['1/1', 0.967948717948718, 156]\n",
      "found 32039015_G_T in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039020_G_A in 77092627, checking inheritance; variant info: 0|1, 0.48148148148148145, 27\n",
      "{'child': 77092646}\n",
      "found 32039020_G_A in 77092646 HC data - ['1|1', 1.0, 34]\n",
      "checking relative variant with ['1|1', 1.0, 34] AND ['1/1', 0.9871794871794872, 156]\n",
      "found 32039020_G_A in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039006_G_A in 77092633, checking inheritance; variant info: 0|1, 0.14705882352941177, 68\n",
      "{'child': 77092644}\n",
      "32039006_G_A not found in gt_hc of 77092644 - the child of 77092633\n",
      "32039006_G_A not found in gt_dv of 77092644 - the child of 77092633\n",
      "no 32039006_G_A found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039027_C_A in 77092635, checking inheritance; variant info: 0|1, 0.8571428571428571, 35\n",
      "{'spouse': 77092636, 'child': 77092641}\n",
      "found 32039027_C_A in 77092641 HC data - ['0|1', 0.6140350877192983, 57]\n",
      "32039027_C_A found only in HC data of child\n",
      "found a heterozygote 32039027_C_A in child with VAF 0.6140350877192983\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039046_A_G in 77092635, checking inheritance; variant info: 0|1, 0.8717948717948718, 39\n",
      "{'spouse': 77092636, 'child': 77092641}\n",
      "found 32039046_A_G in 77092641 HC data - ['0|1', 0.603448275862069, 58]\n",
      "32039046_A_G found only in HC data of child\n",
      "found a heterozygote 32039046_A_G in child with VAF 0.603448275862069\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "for 77092640 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.734375, DP-68\n",
      "for 77092640 comparing 32038949_T_G with HC - 1|1 and DV - 1/0\n",
      "dv right with 1/0, VAF - 0.8145695364238411, DP-155\n",
      "low qual 32039001_G_C in 77092640, checking inheritance; variant info: 1|1, 1.0, 27\n",
      "{'mother': 77092629, 'father': 77092630}\n",
      "found 32039001_G_C in 77092629 HC data - ['0|1', 0.40384615384615385, 104]\n",
      "32039001_G_C found only in HC data of mother\n",
      "found a heterozygote 32039001_G_C in mother with VAF 0.40384615384615385\n",
      "32039001_G_C not found in gt_hc of 77092630 - the father of 77092640\n",
      "32039001_G_C not found in gt_dv of 77092630 - the father of 77092640\n",
      "no 32039001_G_C found in father\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038949_T_C in 77092640, checking inheritance; variant info: 1/0, 0.18543046357615894, 155\n",
      "{'mother': 77092629, 'father': 77092630}\n",
      "32038949_T_C not found in gt_hc of 77092629 - the mother of 77092640\n",
      "32038949_T_C not found in gt_dv of 77092629 - the mother of 77092640\n",
      "no 32038949_T_C found in mother\n",
      "32038949_T_C not found in gt_hc of 77092630 - the father of 77092640\n",
      "32038949_T_C not found in gt_dv of 77092630 - the father of 77092640\n",
      "no 32038949_T_C found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092646 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.7171717171717171, DP-99\n",
      "low qual 32038855_T_TTG in 77092662, checking inheritance; variant info: 1|0, 0.14657210401891252, 2115\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092662 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.8904109589041096, DP-146\n",
      "low qual 32038895_A_G in 77092666, checking inheritance; variant info: 0|1, 0.2916666666666667, 24\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038903_A_G in 77092666, checking inheritance; variant info: 0|1, 0.4117647058823529, 17\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038911_A_G in 77092666, checking inheritance; variant info: 0|1, 0.25, 16\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038948_T_TC in 77092666, checking inheritance; variant info: 0|1, 0.14285714285714285, 14\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038949_T_G in 77092666, checking inheritance; variant info: 0|1, 0.14285714285714285, 14\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038955_A_T in 77092666, checking inheritance; variant info: 0|1, 0.125, 16\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038962_A_ATCC in 77092666, checking inheritance; variant info: 0|1, 0.26666666666666666, 15\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038971_C_A in 77092666, checking inheritance; variant info: 0|1, 0.21052631578947367, 19\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038983_TG_T in 77092666, checking inheritance; variant info: 0|1, 0.38095238095238093, 21\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038997_TGG_T in 77092666, checking inheritance; variant info: 0|1, 0.4117647058823529, 17\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039001_G_C in 77092666, checking inheritance; variant info: 0|1, 0.4117647058823529, 17\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039002_G_A in 77092666, checking inheritance; variant info: 0|1, 0.4117647058823529, 17\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039007_A_AAG in 77092666, checking inheritance; variant info: 0|1, 0.5333333333333333, 15\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039015_G_T in 77092666, checking inheritance; variant info: 0|1, 0.5882352941176471, 17\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039020_G_A in 77092666, checking inheritance; variant info: 0|1, 0.65, 20\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039027_C_A in 77092666, checking inheritance; variant info: 0|1, 0.7368421052631579, 19\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039046_A_G in 77092666, checking inheritance; variant info: 0|1, 0.8148148148148148, 27\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038855_T_TTG in 77092668, checking inheritance; variant info: 1|0, 0.1458625525946704, 2144\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092668 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.8714285714285714, DP-140\n",
      "for 77092669 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8465346534653465, DP-2020\n",
      "low qual 32038855_T_TTG in 77092669, checking inheritance; variant info: 1|0, 0.15346534653465346, 2020\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038962_A_ATCC in 77092671, checking inheritance; variant info: 0|1, 0.11678832116788321, 137\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038971_C_A in 77092671, checking inheritance; variant info: 0|1, 0.11594202898550725, 138\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092672 comparing 32041186_C_T with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.6990241397021058, DP-1947\n",
      "low qual 32038855_T_TTG in 77092673, checking inheritance; variant info: 1|0, 0.13562091503267973, 613\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092673 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.8840579710144928, DP-138\n",
      "low qual 32038539_C_CTTG in 77092674, checking inheritance; variant info: 1|0, 0.1694915254237288, 59\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038997_TGG_T in 77092674, checking inheritance; variant info: 0|1, 0.15873015873015872, 63\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039001_G_C in 77092674, checking inheritance; variant info: 0|1, 0.15517241379310345, 58\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039002_G_A in 77092674, checking inheritance; variant info: 0|1, 0.15789473684210525, 57\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32040112_G_GCA in 77092674, checking inheritance; variant info: 0|1, 0.12643678160919541, 174\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092704 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.9166666666666666, DP-228\n",
      "low qual 32038855_T_TTG in 77092704, checking inheritance; variant info: 1|0, 0.13663616282739036, 3176\n",
      "{'father': 77092703, 'sibling': 77092705}\n",
      "32038855_T_TTG not found in gt_hc of 77092703 - the father of 77092704\n",
      "32038855_T_TTG not found in gt_dv of 77092703 - the father of 77092704\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092705 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8390659514042285, DP-3169\n",
      "low qual 32038855_T_TTG in 77092705, checking inheritance; variant info: 1|0, 0.16093404859577154, 3169\n",
      "{'father': 77092703, 'sibling': 77092704}\n",
      "32038855_T_TTG not found in gt_hc of 77092703 - the father of 77092705\n",
      "32038855_T_TTG not found in gt_dv of 77092703 - the father of 77092705\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038855_T_TTG in 77092720, checking inheritance; variant info: 1|0, 0.14342145244606502, 3310\n",
      "{'mother': 77092719}\n",
      "32038855_T_TTG not found in gt_hc of 77092719 - the mother of 77092720\n",
      "32038855_T_TTG not found in gt_dv of 77092719 - the mother of 77092720\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092720 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.8991228070175439, DP-228\n",
      "for 77092731 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8311418685121107, DP-2904\n",
      "low qual 32038855_T_TTG in 77092731, checking inheritance; variant info: 1|0, 0.16885813148788928, 2904\n",
      "{'mother': 77092732}\n",
      "32038855_T_TTG not found in gt_hc of 77092732 - the mother of 77092731\n",
      "32038855_T_TTG not found in gt_dv of 77092732 - the mother of 77092731\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038855_T_TTG in 77092742, checking inheritance; variant info: 1|0, 0.1587397758255074, 3302\n",
      "{'mother': 77092741}\n",
      "32038855_T_TTG not found in gt_hc of 77092741 - the mother of 77092742\n",
      "32038855_T_TTG not found in gt_dv of 77092741 - the mother of 77092742\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092742 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8412602241744925, DP-3302\n",
      "low qual 32038855_T_TTG in 77092745, checking inheritance; variant info: 1|0, 0.15129573170731708, 2641\n",
      "{'mother': 77092744, 'father': 77092743}\n",
      "32038855_T_TTG not found in gt_hc of 77092744 - the mother of 77092745\n",
      "32038855_T_TTG not found in gt_dv of 77092744 - the mother of 77092745\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092743 - the father of 77092745\n",
      "32038855_T_TTG not found in gt_dv of 77092743 - the father of 77092745\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092745 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.848704268292683, DP-2641\n",
      "for 77092758 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8470754052149401, DP-1419\n",
      "low qual 32038855_T_TTG in 77092758, checking inheritance; variant info: 1|0, 0.1529245947850599, 1419\n",
      "{'mother': 77092757, 'sibling': 77092754, 'father': 77092756}\n",
      "32038855_T_TTG not found in gt_hc of 77092757 - the mother of 77092758\n",
      "32038855_T_TTG not found in gt_dv of 77092757 - the mother of 77092758\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092756 - the father of 77092758\n",
      "32038855_T_TTG not found in gt_dv of 77092756 - the father of 77092758\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092758 comparing 32041186_C_T with HC - 1|1 and DV - 0/1\n",
      "hc right with 1|1, VAF - 0.9900332225913622, DP-1505\n",
      "for 77092780 comparing 32038949_T_G with HC - 1|1 and DV - 1/0\n",
      "hc right with 1|1, VAF - 1.0, DP-1509\n",
      "low qual 32038949_T_C in 77092780, checking inheritance; variant info: 1/0, 0.13529411764705881, 170\n",
      "{'sibling': 77092782, 'father': 77092781, 'mother': 77092783}\n",
      "32038949_T_C not found in gt_hc of 77092781 - the father of 77092780\n",
      "32038949_T_C not found in gt_dv of 77092781 - the father of 77092780\n",
      "no 32038949_T_C found in father\n",
      "32038949_T_C not found in gt_hc of 77092783 - the mother of 77092780\n",
      "32038949_T_C not found in gt_dv of 77092783 - the mother of 77092780\n",
      "no 32038949_T_C found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092782 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8448660714285714, DP-1805\n",
      "low qual 32038855_T_TTG in 77092782, checking inheritance; variant info: 1|0, 0.15513392857142858, 1805\n",
      "{'sibling': 77092780, 'father': 77092781, 'mother': 77092783}\n",
      "32038855_T_TTG not found in gt_hc of 77092781 - the father of 77092782\n",
      "32038855_T_TTG not found in gt_dv of 77092781 - the father of 77092782\n",
      "no 32038855_T_TTG found in father\n",
      "32038855_T_TTG not found in gt_hc of 77092783 - the mother of 77092782\n",
      "32038855_T_TTG not found in gt_dv of 77092783 - the mother of 77092782\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092791 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8440438871473355, DP-1279\n",
      "low qual 32038855_T_TTG in 77092791, checking inheritance; variant info: 1|0, 0.15595611285266459, 1279\n",
      "{'mother': 77092790, 'sibling': 77092792, 'father': 77092789}\n",
      "32038855_T_TTG not found in gt_hc of 77092790 - the mother of 77092791\n",
      "32038855_T_TTG not found in gt_dv of 77092790 - the mother of 77092791\n",
      "no 32038855_T_TTG found in mother\n",
      "32038855_T_TTG not found in gt_hc of 77092789 - the father of 77092791\n",
      "32038855_T_TTG not found in gt_dv of 77092789 - the father of 77092791\n",
      "no 32038855_T_TTG found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039007_A_AAG in 77092896, checking inheritance; variant info: 0|1, 0.1935483870967742, 31\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039046_A_G in 77092896, checking inheritance; variant info: 0|1, 0.42857142857142855, 28\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038903_A_G in 77092897, checking inheritance; variant info: 0|1, 0.3, 20\n",
      "{'child': 77092898}\n",
      "32038903_A_G not found in gt_hc of 77092898 - the child of 77092897\n",
      "found 32038903_A_G in 77092898 DV data - ['1/1', 1.0, 136]\n",
      "32038903_A_G found only in DV data of child\n",
      "found a homozygote 32038903_A_G in child with VAF 1.0\n",
      "found only low qual 32038903_A_G in child with 1.0 and 136, probably an artefact\n",
      "['1/1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038911_A_G in 77092897, checking inheritance; variant info: 0|1, 0.2222222222222222, 18\n",
      "{'child': 77092898}\n",
      "found 32038911_A_G in 77092898 HC data - ['1|1', 1.0, 429]\n",
      "checking relative variant with ['1|1', 1.0, 429] AND ['1/1', 0.9923664122137404, 131]\n",
      "found 32038911_A_G in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038927_A_G in 77092897, checking inheritance; variant info: 0|1, 0.25, 12\n",
      "{'child': 77092898}\n",
      "found 32038927_A_G in 77092898 HC data - ['1|1', 1.0, 316]\n",
      "checking relative variant with ['1|1', 1.0, 316] AND ['1/1', 1.0, 111]\n",
      "found 32038927_A_G in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038938_T_G in 77092897, checking inheritance; variant info: 0|1, 0.3333333333333333, 9\n",
      "{'child': 77092898}\n",
      "found 32038938_T_G in 77092898 HC data - ['1|1', 1.0, 238]\n",
      "checking relative variant with ['1|1', 1.0, 238] AND ['1/1', 1.0, 97]\n",
      "found 32038938_T_G in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038948_T_TC in 77092897, checking inheritance; variant info: 0|1, 0.21428571428571427, 14\n",
      "{'child': 77092898}\n",
      "found 32038948_T_TC in 77092898 HC data - ['1|1', 1.0, 137]\n",
      "checking relative variant with ['1|1', 1.0, 137] AND ['1/1', 0.9230769230769231, 52]\n",
      "found 32038948_T_TC in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038949_T_G in 77092897, checking inheritance; variant info: 0|1, 0.21428571428571427, 14\n",
      "{'child': 77092898}\n",
      "found 32038949_T_G in 77092898 HC data - ['1|1', 1.0, 179]\n",
      "checking relative variant with ['1|1', 1.0, 179] AND ['1/1', 1.0, 48]\n",
      "found 32038949_T_G in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038955_A_T in 77092897, checking inheritance; variant info: 0|1, 0.26666666666666666, 15\n",
      "{'child': 77092898}\n",
      "found 32038955_A_T in 77092898 HC data - ['1|1', 1.0, 134]\n",
      "checking relative variant with ['1|1', 1.0, 134] AND ['1/1', 1.0, 51]\n",
      "found 32038955_A_T in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038962_A_ATCC in 77092897, checking inheritance; variant info: 0|1, 0.4375, 16\n",
      "{'child': 77092898}\n",
      "found 32038962_A_ATCC in 77092898 HC data - ['1|1', 1.0, 126]\n",
      "checking relative variant with ['1|1', 1.0, 126] AND ['1/1', 0.9183673469387755, 49]\n",
      "found 32038962_A_ATCC in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038971_C_A in 77092897, checking inheritance; variant info: 0|1, 0.3, 20\n",
      "{'child': 77092898}\n",
      "found 32038971_C_A in 77092898 HC data - ['1|1', 1.0, 77]\n",
      "checking relative variant with ['1|1', 1.0, 77] AND ['1/1', 0.8987341772151899, 79]\n",
      "found 32038971_C_A in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038983_TG_T in 77092897, checking inheritance; variant info: 0|1, 0.4444444444444444, 18\n",
      "{'child': 77092898}\n",
      "found 32038983_TG_T in 77092898 HC data - ['1|1', 1.0, 39]\n",
      "32038983_TG_T found only in HC data of child\n",
      "found a homozygote 32038983_TG_T in child with VAF 1.0\n",
      "found only low qual 32038983_TG_T in child with 1.0 and 39, probably an artefact\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038997_TGG_T in 77092897, checking inheritance; variant info: 0|1, 0.5, 16\n",
      "{'child': 77092898}\n",
      "found 32038997_TGG_T in 77092898 HC data - ['1|1', 1.0, 25]\n",
      "checking relative variant with ['1|1', 1.0, 25] AND ['1/1', 1.0, 28]\n",
      "found 32038997_TGG_T in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039001_G_C in 77092897, checking inheritance; variant info: 0|1, 0.5714285714285714, 14\n",
      "{'child': 77092898}\n",
      "found 32039001_G_C in 77092898 HC data - ['1|1', 1.0, 20]\n",
      "32039001_G_C found only in HC data of child\n",
      "found a lowqual homozygote 32039001_G_C in child with right VAF 1.0, but low DP - 20\n",
      "found only low qual 32039001_G_C in child with 1.0 and 20, probably an artefact\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039002_G_A in 77092897, checking inheritance; variant info: 0|1, 0.5714285714285714, 14\n",
      "{'child': 77092898}\n",
      "found 32039002_G_A in 77092898 HC data - ['1|1', 1.0, 20]\n",
      "checking relative variant with ['1|1', 1.0, 20] AND ['1/1', 1.0, 28]\n",
      "found 32039002_G_A in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039007_A_AAG in 77092897, checking inheritance; variant info: 0|1, 0.6, 15\n",
      "{'child': 77092898}\n",
      "found 32039007_A_AAG in 77092898 HC data - ['1|1', 1.0, 15]\n",
      "32039007_A_AAG found only in HC data of child\n",
      "found a lowqual homozygote 32039007_A_AAG in child with right VAF 1.0, but low DP - 15\n",
      "found only low qual 32039007_A_AAG in child with 1.0 and 15, probably an artefact\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039015_G_T in 77092897, checking inheritance; variant info: 0|1, 0.6470588235294118, 17\n",
      "{'child': 77092898}\n",
      "found 32039015_G_T in 77092898 HC data - ['1|1', 1.0, 12]\n",
      "checking relative variant with ['1|1', 1.0, 12] AND ['1/1', 0.9146341463414634, 82]\n",
      "found 32039015_G_T in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039020_G_A in 77092897, checking inheritance; variant info: 0|1, 0.6842105263157895, 19\n",
      "{'child': 77092898}\n",
      "found 32039020_G_A in 77092898 HC data - ['1|1', 1.0, 11]\n",
      "checking relative variant with ['1|1', 1.0, 11] AND ['1/1', 0.9882352941176471, 85]\n",
      "found 32039020_G_A in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039027_C_A in 77092897, checking inheritance; variant info: 0|1, 0.6666666666666666, 21\n",
      "{'child': 77092898}\n",
      "found 32039027_C_A in 77092898 HC data - ['1|1', 1.0, 9]\n",
      "checking relative variant with ['1|1', 1.0, 9] AND ['1/1', 0.9411764705882353, 102]\n",
      "found 32039027_C_A in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038843_A_G in 77092898, checking inheritance; variant info: 0|1, 0.45454545454545453, 22\n",
      "{'mother': 77092897}\n",
      "32038843_A_G not found in gt_hc of 77092897 - the mother of 77092898\n",
      "32038843_A_G not found in gt_dv of 77092897 - the mother of 77092898\n",
      "no 32038843_A_G found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092898 comparing 32038844_A_C with HC - 1|0 and DV - 1/1\n",
      "both wrong, checking inheritance; variant info: 1|0, 0.45454545454545453, 22 AND 1/1, 0.8181818181818182, 242\n",
      "{'mother': 77092897}\n",
      "found 32038844_A_C in 77092897 HC data - ['0|1', 0.6575342465753424, 73]\n",
      "checking relative variant with ['0|1', 0.6575342465753424, 73] AND ['0/1', 0.3803921568627451, 255]\n",
      "found 32038844_A_C in mother with genotype 0|1\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "for 77092898 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "both wrong, checking inheritance; variant info: 1|0, 0.45454545454545453, 22 AND 1/1, 0.4830917874396135, 207\n",
      "{'mother': 77092897}\n",
      "found 32038855_T_TTTG in 77092897 HC data - ['0|1', 0.576271186440678, 59]\n",
      "checking relative variant with ['0|1', 0.576271186440678, 59] AND ['0/1', 0.23076923076923078, 234]\n",
      "found 32038855_T_TTTG in mother with genotype 0|1\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038856_TAA_T in 77092898, checking inheritance; variant info: 0|1, 0.23076923076923078, 13\n",
      "{'mother': 77092897}\n",
      "32038856_TAA_T not found in gt_hc of 77092897 - the mother of 77092898\n",
      "32038856_TAA_T not found in gt_dv of 77092897 - the mother of 77092898\n",
      "no 32038856_TAA_T found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038857_A_* in 77092898, checking inheritance; variant info: 1|0, 0.23076923076923078, 13\n",
      "{'mother': 77092897}\n",
      "32038857_A_* not found in gt_hc of 77092897 - the mother of 77092898\n",
      "32038857_A_* not found in gt_dv of 77092897 - the mother of 77092898\n",
      "no 32038857_A_* found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77092898 comparing 32038857_A_T with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.9787234042553191, DP-141\n",
      "low qual 32038859_G_GTT in 77092898, checking inheritance; variant info: 0|1, 0.75, 4\n",
      "{'mother': 77092897}\n",
      "32038859_G_GTT not found in gt_hc of 77092897 - the mother of 77092898\n",
      "32038859_G_GTT not found in gt_dv of 77092897 - the mother of 77092898\n",
      "no 32038859_G_GTT found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039001_G_C in 77092898, checking inheritance; variant info: 1|1, 1.0, 20\n",
      "{'mother': 77092897}\n",
      "found 32039001_G_C in 77092897 HC data - ['0|1', 0.5714285714285714, 14]\n",
      "32039001_G_C found only in HC data of mother\n",
      "found a lowqual heterozygote 32039001_G_C in mother with right VAF 0.5714285714285714, but low DP - 14\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039007_A_AAG in 77092898, checking inheritance; variant info: 1|1, 1.0, 15\n",
      "{'mother': 77092897}\n",
      "found 32039007_A_AAG in 77092897 HC data - ['0|1', 0.6, 15]\n",
      "32039007_A_AAG found only in HC data of mother\n",
      "found a lowqual heterozygote 32039007_A_AAG in mother with right VAF 0.6, but low DP - 15\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038962_A_ATCC in 77092900, checking inheritance; variant info: 1|0, 0.07246376811594203, 138\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038971_C_A in 77092900, checking inheritance; variant info: 1|0, 0.10416666666666667, 96\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039050_G_T in 77092900, checking inheritance; variant info: 1|0, 0.16666666666666666, 12\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039056_A_G in 77092900, checking inheritance; variant info: 1|0, 0.16666666666666666, 12\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32040791_C_CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCGGCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCGGGCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCGGGGGGGGGCGCCGGGGGGGGCCCCCCAGGGG in 77092900, checking inheritance; variant info: 0|1, 0.2222222222222222, 27\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038878_C_T in 77095100, checking inheritance; variant info: 0|1, 0.12365591397849462, 186\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038971_C_A in 77095100, checking inheritance; variant info: 0|1, 0.19047619047619047, 42\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039001_G_C in 77095100, checking inheritance; variant info: 0|1, 0.19444444444444445, 36\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038895_A_G in 77095101, checking inheritance; variant info: 0|1, 0.5925925925925926, 27\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038903_A_G in 77095101, checking inheritance; variant info: 0|1, 0.3684210526315789, 19\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038911_A_G in 77095101, checking inheritance; variant info: 0|1, 0.4, 20\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038927_A_G in 77095101, checking inheritance; variant info: 0|1, 0.2631578947368421, 19\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038938_T_G in 77095101, checking inheritance; variant info: 0|1, 0.2777777777777778, 18\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038948_T_TC in 77095101, checking inheritance; variant info: 0|1, 0.25, 20\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038949_T_G in 77095101, checking inheritance; variant info: 0|1, 0.25, 20\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038955_A_T in 77095101, checking inheritance; variant info: 0|1, 0.25, 20\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038962_A_ATCC in 77095101, checking inheritance; variant info: 0|1, 0.2777777777777778, 18\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038971_C_A in 77095101, checking inheritance; variant info: 0|1, 0.25, 16\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038983_TG_T in 77095101, checking inheritance; variant info: 0|1, 0.2857142857142857, 14\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039007_A_AAG in 77095101, checking inheritance; variant info: 0|1, 0.3, 10\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039015_G_T in 77095101, checking inheritance; variant info: 0|1, 0.3, 10\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039020_G_A in 77095101, checking inheritance; variant info: 0|1, 0.3, 10\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039027_C_A in 77095101, checking inheritance; variant info: 0|1, 0.23076923076923078, 13\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039046_A_G in 77095101, checking inheritance; variant info: 0|1, 0.13636363636363635, 22\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039055_C_G in 77095101, checking inheritance; variant info: 0|1, 0.11475409836065574, 61\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039056_A_G in 77095101, checking inheritance; variant info: 0|1, 0.11475409836065574, 61\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77095101 comparing 32039081_C_A with HC - 1|1 and DV - 1/0\n",
      "hc right with 1|1, VAF - 0.990909090909091, DP-110\n",
      "low qual 32039007_A_AAG in 77095103, checking inheritance; variant info: 0|1, 0.41379310344827586, 29\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039027_C_A in 77095103, checking inheritance; variant info: 0|1, 0.37037037037037035, 27\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77095103 comparing 32039119_G_C with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.5565217391304348, DP-115\n",
      "low qual 32038855_T_TTTG in 77095103, checking inheritance; variant info: 0/1, 0.19213973799126638, 229\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038948_T_TC in 77095106, checking inheritance; variant info: 1|0, 0.14130434782608695, 184\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038949_T_G in 77095106, checking inheritance; variant info: 1|0, 0.14130434782608695, 184\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038955_A_T in 77095106, checking inheritance; variant info: 1|0, 0.1452513966480447, 179\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038962_A_ATCC in 77095106, checking inheritance; variant info: 1|0, 0.0958904109589041, 73\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038971_C_A in 77095106, checking inheritance; variant info: 1|0, 0.08823529411764706, 68\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039027_C_A in 77095106, checking inheritance; variant info: 1|0, 0.17391304347826086, 23\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039046_A_G in 77095106, checking inheritance; variant info: 1|0, 0.13043478260869565, 23\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039050_G_T in 77095106, checking inheritance; variant info: 1|0, 0.16, 25\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039055_C_G in 77095106, checking inheritance; variant info: 1|0, 0.15384615384615385, 26\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039056_A_G in 77095106, checking inheritance; variant info: 1|0, 0.15384615384615385, 26\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038949_T_C in 77095107, checking inheritance; variant info: 0|1, 0.10526315789473684, 190\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038954_C_G in 77095107, checking inheritance; variant info: 0|1, 0.10582010582010581, 189\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038955_A_C in 77095107, checking inheritance; variant info: 0|1, 0.10638297872340426, 188\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038962_A_ATCC in 77095107, checking inheritance; variant info: 0|1, 0.13846153846153847, 65\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038971_C_A in 77095107, checking inheritance; variant info: 0|1, 0.15254237288135594, 59\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039056_A_G in 77095107, checking inheritance; variant info: 0|1, 0.14285714285714285, 14\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038903_A_G in 77095108, checking inheritance; variant info: 1|0, 0.2631578947368421, 19\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038927_A_G in 77095108, checking inheritance; variant info: 1|0, 0.23529411764705882, 17\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038938_T_G in 77095108, checking inheritance; variant info: 1|0, 0.17647058823529413, 17\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038938_T_G in 77095109, checking inheritance; variant info: 0|1, 0.5, 28\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038948_T_TC in 77095109, checking inheritance; variant info: 0|1, 0.30434782608695654, 23\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038949_T_G in 77095109, checking inheritance; variant info: 0|1, 0.5, 24\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038955_A_T in 77095109, checking inheritance; variant info: 0|1, 0.3684210526315789, 19\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038962_A_ATCC in 77095109, checking inheritance; variant info: 0|1, 0.3157894736842105, 19\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038971_C_A in 77095109, checking inheritance; variant info: 0|1, 0.2857142857142857, 14\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038983_TG_T in 77095109, checking inheritance; variant info: 0|1, 0.26666666666666666, 15\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038997_TGG_T in 77095109, checking inheritance; variant info: 0|1, 0.2, 20\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039001_G_C in 77095109, checking inheritance; variant info: 0|1, 0.18181818181818182, 22\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039002_G_A in 77095109, checking inheritance; variant info: 0|1, 0.18181818181818182, 22\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039007_A_AAG in 77095109, checking inheritance; variant info: 0|1, 0.17391304347826086, 23\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039015_G_T in 77095109, checking inheritance; variant info: 0|1, 0.13793103448275862, 29\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039055_C_G in 77095109, checking inheritance; variant info: 0|1, 0.15789473684210525, 152\n",
      "{}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038927_A_G in 77095331, checking inheritance; variant info: 0|1, 0.20689655172413793, 29\n",
      "{'child': 77095332, 'spouse': 77095330}\n",
      "32038927_A_G not found in gt_hc of 77095332 - the child of 77095331\n",
      "32038927_A_G not found in gt_dv of 77095332 - the child of 77095331\n",
      "no 32038927_A_G found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038997_TGG_T in 77095334, checking inheritance; variant info: 0|1, 0.4444444444444444, 27\n",
      "{'child': 77095335, 'spouse': 77095340}\n",
      "found 32038997_TGG_T in 77095335 HC data - ['1|1', 1.0, 8]\n",
      "32038997_TGG_T found only in HC data of child\n",
      "found a lowqual homozygote 32038997_TGG_T in child with right VAF 1.0, but low DP - 8\n",
      "found only low qual 32038997_TGG_T in child with 1.0 and 8, probably an artefact\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039001_G_C in 77095334, checking inheritance; variant info: 0|1, 0.48, 25\n",
      "{'child': 77095335, 'spouse': 77095340}\n",
      "found 32039001_G_C in 77095335 HC data - ['1|1', 1.0, 4]\n",
      "32039001_G_C found only in HC data of child\n",
      "found a lowqual homozygote 32039001_G_C in child with right VAF 1.0, but low DP - 4\n",
      "found only low qual 32039001_G_C in child with 1.0 and 4, probably an artefact\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039002_G_A in 77095334, checking inheritance; variant info: 0|1, 0.5217391304347826, 23\n",
      "{'child': 77095335, 'spouse': 77095340}\n",
      "found 32039002_G_A in 77095335 HC data - ['1|1', 1.0, 5]\n",
      "checking relative variant with ['1|1', 1.0, 5] AND ['1/1', 0.9629629629629629, 27]\n",
      "found 32039002_G_A in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039007_A_AAG in 77095334, checking inheritance; variant info: 0|1, 0.6666666666666666, 24\n",
      "{'child': 77095335, 'spouse': 77095340}\n",
      "found 32039007_A_AAG in 77095335 HC data - ['1|1', 1.0, 4]\n",
      "32039007_A_AAG found only in HC data of child\n",
      "found a lowqual homozygote 32039007_A_AAG in child with right VAF 1.0, but low DP - 4\n",
      "found only low qual 32039007_A_AAG in child with 1.0 and 4, probably an artefact\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039015_G_T in 77095334, checking inheritance; variant info: 0|1, 0.5714285714285714, 28\n",
      "{'child': 77095335, 'spouse': 77095340}\n",
      "found 32039015_G_T in 77095335 HC data - ['1|1', 1.0, 5]\n",
      "checking relative variant with ['1|1', 1.0, 5] AND ['1/1', 0.9444444444444444, 72]\n",
      "found 32039015_G_T in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038835_T_G in 77095335, checking inheritance; variant info: 0|1, 0.4, 20\n",
      "{'sibling': 77095336, 'mother': 77095334, 'father': 77095340}\n",
      "32038835_T_G not found in gt_hc of 77095334 - the mother of 77095335\n",
      "32038835_T_G not found in gt_dv of 77095334 - the mother of 77095335\n",
      "no 32038835_T_G found in mother\n",
      "32038835_T_G not found in gt_hc of 77095340 - the father of 77095335\n",
      "32038835_T_G not found in gt_dv of 77095340 - the father of 77095335\n",
      "no 32038835_T_G found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77095335 comparing 32038844_A_C with HC - 1|0 and DV - 1/1\n",
      "both wrong, checking inheritance; variant info: 1|0, 0.5882352941176471, 17 AND 1/1, 0.8461538461538461, 234\n",
      "{'sibling': 77095336, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038844_A_C in 77095334 HC data - ['0|1', 0.4305555555555556, 72]\n",
      "checking relative variant with ['0|1', 0.4305555555555556, 72] AND ['0/1', 0.37448559670781895, 243]\n",
      "found 32038844_A_C in mother with genotype 0|1\n",
      "found 32038844_A_C in 77095340 HC data - ['0|1', 0.5337837837837838, 148]\n",
      "checking relative variant with ['0|1', 0.5337837837837838, 148] AND ['0/1', 0.38645418326693226, 251]\n",
      "found 32038844_A_C in father with genotype 0|1\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "for 77095335 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "both wrong, checking inheritance; variant info: 1|0, 0.4444444444444444, 18 AND 1/1, 0.53, 200\n",
      "{'sibling': 77095336, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038855_T_TTTG in 77095334 HC data - ['0|1', 0.4225352112676056, 71]\n",
      "checking relative variant with ['0|1', 0.4225352112676056, 71] AND ['0/1', 0.23684210526315788, 228]\n",
      "found 32038855_T_TTTG in mother with genotype 0|1\n",
      "found 32038855_T_TTTG in 77095340 HC data - ['0|1', 0.49230769230769234, 130]\n",
      "checking relative variant with ['0|1', 0.49230769230769234, 130] AND ['0/1', 0.20689655172413793, 232]\n",
      "found 32038855_T_TTTG in father with genotype 0|1\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "for 77095335 comparing 32038857_A_T with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.9605263157894737, DP-152\n",
      "low qual 32038983_TG_T in 77095335, checking inheritance; variant info: 1|1, 1.0, 16\n",
      "{'sibling': 77095336, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038983_TG_T in 77095334 HC data - ['0|1', 0.4883720930232558, 43]\n",
      "32038983_TG_T found only in HC data of mother\n",
      "found a heterozygote 32038983_TG_T in mother with VAF 0.4883720930232558\n",
      "found 32038983_TG_T in 77095340 HC data - ['0|1', 0.6296296296296297, 27]\n",
      "32038983_TG_T found only in HC data of father\n",
      "found a lowqual heterozygote 32038983_TG_T in father with right VAF 0.6296296296296297, but low DP - 27\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038997_TGG_T in 77095335, checking inheritance; variant info: 1|1, 1.0, 8\n",
      "{'sibling': 77095336, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038997_TGG_T in 77095334 HC data - ['0|1', 0.4444444444444444, 27]\n",
      "32038997_TGG_T found only in HC data of mother\n",
      "found a lowqual heterozygote 32038997_TGG_T in mother with right VAF 0.4444444444444444, but low DP - 27\n",
      "found 32038997_TGG_T in 77095340 HC data - ['0|1', 0.47058823529411764, 17]\n",
      "32038997_TGG_T found only in HC data of father\n",
      "found a lowqual heterozygote 32038997_TGG_T in father with right VAF 0.47058823529411764, but low DP - 17\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039001_G_C in 77095335, checking inheritance; variant info: 1|1, 1.0, 4\n",
      "{'sibling': 77095336, 'mother': 77095334, 'father': 77095340}\n",
      "found 32039001_G_C in 77095334 HC data - ['0|1', 0.48, 25]\n",
      "32039001_G_C found only in HC data of mother\n",
      "found a lowqual heterozygote 32039001_G_C in mother with right VAF 0.48, but low DP - 25\n",
      "found 32039001_G_C in 77095340 HC data - ['0|1', 0.5, 14]\n",
      "32039001_G_C found only in HC data of father\n",
      "found a lowqual heterozygote 32039001_G_C in father with right VAF 0.5, but low DP - 14\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039007_A_AAG in 77095335, checking inheritance; variant info: 1|1, 1.0, 4\n",
      "{'sibling': 77095336, 'mother': 77095334, 'father': 77095340}\n",
      "found 32039007_A_AAG in 77095334 HC data - ['0|1', 0.6666666666666666, 24]\n",
      "32039007_A_AAG found only in HC data of mother\n",
      "found a lowqual heterozygote 32039007_A_AAG in mother with right VAF 0.6666666666666666, but low DP - 24\n",
      "found 32039007_A_AAG in 77095340 HC data - ['0|1', 0.5, 12]\n",
      "32039007_A_AAG found only in HC data of father\n",
      "found a lowqual heterozygote 32039007_A_AAG in father with right VAF 0.5, but low DP - 12\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038818_G_T in 77095336, checking inheritance; variant info: 0|1, 0.13636363636363635, 22\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "32038818_G_T not found in gt_hc of 77095334 - the mother of 77095336\n",
      "32038818_G_T not found in gt_dv of 77095334 - the mother of 77095336\n",
      "no 32038818_G_T found in mother\n",
      "32038818_G_T not found in gt_hc of 77095340 - the father of 77095336\n",
      "32038818_G_T not found in gt_dv of 77095340 - the father of 77095336\n",
      "no 32038818_G_T found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77095336 comparing 32038844_A_C with HC - 1|0 and DV - 1/1\n",
      "both wrong, checking inheritance; variant info: 1|0, 0.5294117647058824, 17 AND 1/1, 0.8220338983050848, 236\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038844_A_C in 77095334 HC data - ['0|1', 0.4305555555555556, 72]\n",
      "checking relative variant with ['0|1', 0.4305555555555556, 72] AND ['0/1', 0.37448559670781895, 243]\n",
      "found 32038844_A_C in mother with genotype 0|1\n",
      "found 32038844_A_C in 77095340 HC data - ['0|1', 0.5337837837837838, 148]\n",
      "checking relative variant with ['0|1', 0.5337837837837838, 148] AND ['0/1', 0.38645418326693226, 251]\n",
      "found 32038844_A_C in father with genotype 0|1\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038856_TAAGAGA_T in 77095336, checking inheritance; variant info: 0|1, 0.21428571428571427, 14\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "32038856_TAAGAGA_T not found in gt_hc of 77095334 - the mother of 77095336\n",
      "32038856_TAAGAGA_T not found in gt_dv of 77095334 - the mother of 77095336\n",
      "no 32038856_TAAGAGA_T found in mother\n",
      "32038856_TAAGAGA_T not found in gt_hc of 77095340 - the father of 77095336\n",
      "32038856_TAAGAGA_T not found in gt_dv of 77095340 - the father of 77095336\n",
      "no 32038856_TAAGAGA_T found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038863_TGG_T in 77095336, checking inheritance; variant info: 1|1, 1.0, 3\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "32038863_TGG_T not found in gt_hc of 77095334 - the mother of 77095336\n",
      "32038863_TGG_T not found in gt_dv of 77095334 - the mother of 77095336\n",
      "no 32038863_TGG_T found in mother\n",
      "32038863_TGG_T not found in gt_hc of 77095340 - the father of 77095336\n",
      "32038863_TGG_T not found in gt_dv of 77095340 - the father of 77095336\n",
      "no 32038863_TGG_T found in father\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038855_T_TTTG in 77095336, checking inheritance; variant info: 1/1, 0.4816753926701571, 191\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038855_T_TTTG in 77095334 HC data - ['0|1', 0.4225352112676056, 71]\n",
      "checking relative variant with ['0|1', 0.4225352112676056, 71] AND ['0/1', 0.23684210526315788, 228]\n",
      "found 32038855_T_TTTG in mother with genotype 0|1\n",
      "found 32038855_T_TTTG in 77095340 HC data - ['0|1', 0.49230769230769234, 130]\n",
      "checking relative variant with ['0|1', 0.49230769230769234, 130] AND ['0/1', 0.20689655172413793, 232]\n",
      "found 32038855_T_TTTG in father with genotype 0|1\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038983_TG_T in 77095336, checking inheritance; variant info: 1/1, 0.3717948717948718, 78\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038983_TG_T in 77095334 HC data - ['0|1', 0.4883720930232558, 43]\n",
      "32038983_TG_T found only in HC data of mother\n",
      "found a heterozygote 32038983_TG_T in mother with VAF 0.4883720930232558\n",
      "found 32038983_TG_T in 77095340 HC data - ['0|1', 0.6296296296296297, 27]\n",
      "32038983_TG_T found only in HC data of father\n",
      "found a lowqual heterozygote 32038983_TG_T in father with right VAF 0.6296296296296297, but low DP - 27\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038997_TGG_T in 77095336, checking inheritance; variant info: 1/1, 0.8, 30\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32038997_TGG_T in 77095334 HC data - ['0|1', 0.4444444444444444, 27]\n",
      "32038997_TGG_T found only in HC data of mother\n",
      "found a lowqual heterozygote 32038997_TGG_T in mother with right VAF 0.4444444444444444, but low DP - 27\n",
      "found 32038997_TGG_T in 77095340 HC data - ['0|1', 0.47058823529411764, 17]\n",
      "32038997_TGG_T found only in HC data of father\n",
      "found a lowqual heterozygote 32038997_TGG_T in father with right VAF 0.47058823529411764, but low DP - 17\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039002_G_A in 77095336, checking inheritance; variant info: 1/1, 0.8, 30\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32039002_G_A in 77095334 HC data - ['0|1', 0.5217391304347826, 23]\n",
      "32039002_G_A found only in HC data of mother\n",
      "found a lowqual heterozygote 32039002_G_A in mother with right VAF 0.5217391304347826, but low DP - 23\n",
      "found 32039002_G_A in 77095340 HC data - ['0|1', 0.5, 14]\n",
      "32039002_G_A found only in HC data of father\n",
      "found a lowqual heterozygote 32039002_G_A in father with right VAF 0.5, but low DP - 14\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039007_A_AAG in 77095336, checking inheritance; variant info: 1/1, 0.8620689655172413, 29\n",
      "{'sibling': 77095335, 'mother': 77095334, 'father': 77095340}\n",
      "found 32039007_A_AAG in 77095334 HC data - ['0|1', 0.6666666666666666, 24]\n",
      "32039007_A_AAG found only in HC data of mother\n",
      "found a lowqual heterozygote 32039007_A_AAG in mother with right VAF 0.6666666666666666, but low DP - 24\n",
      "found 32039007_A_AAG in 77095340 HC data - ['0|1', 0.5, 12]\n",
      "32039007_A_AAG found only in HC data of father\n",
      "found a lowqual heterozygote 32039007_A_AAG in father with right VAF 0.5, but low DP - 12\n",
      "['0|1', '0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038997_TGG_T in 77095338, checking inheritance; variant info: 1|0, 0.16279069767441862, 43\n",
      "{'sibling': 77095337, 'father': 77095339, 'mother': 77095341}\n",
      "found 32038997_TGG_T in 77095339 HC data - ['0|1', 0.4722222222222222, 36]\n",
      "32038997_TGG_T found only in HC data of father\n",
      "found a heterozygote 32038997_TGG_T in father with VAF 0.4722222222222222\n",
      "32038997_TGG_T not found in gt_hc of 77095341 - the mother of 77095338\n",
      "32038997_TGG_T not found in gt_dv of 77095341 - the mother of 77095338\n",
      "no 32038997_TGG_T found in mother\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039001_G_C in 77095338, checking inheritance; variant info: 1|0, 0.15625, 32\n",
      "{'sibling': 77095337, 'father': 77095339, 'mother': 77095341}\n",
      "found 32039001_G_C in 77095339 HC data - ['0|1', 0.37777777777777777, 45]\n",
      "32039001_G_C found only in HC data of father\n",
      "found a heterozygote 32039001_G_C in father with VAF 0.37777777777777777\n",
      "32039001_G_C not found in gt_hc of 77095341 - the mother of 77095338\n",
      "32039001_G_C not found in gt_dv of 77095341 - the mother of 77095338\n",
      "no 32039001_G_C found in mother\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039002_G_A in 77095338, checking inheritance; variant info: 1|0, 0.16129032258064516, 31\n",
      "{'sibling': 77095337, 'father': 77095339, 'mother': 77095341}\n",
      "found 32039002_G_A in 77095339 HC data - ['0|1', 0.3695652173913043, 46]\n",
      "32039002_G_A found only in HC data of father\n",
      "found a heterozygote 32039002_G_A in father with VAF 0.3695652173913043\n",
      "32039002_G_A not found in gt_hc of 77095341 - the mother of 77095338\n",
      "32039002_G_A not found in gt_dv of 77095341 - the mother of 77095338\n",
      "no 32039002_G_A found in mother\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039007_A_AAG in 77095338, checking inheritance; variant info: 1|0, 0.2, 25\n",
      "{'sibling': 77095337, 'father': 77095339, 'mother': 77095341}\n",
      "found 32039007_A_AAG in 77095339 HC data - ['0|1', 0.3877551020408163, 49]\n",
      "32039007_A_AAG found only in HC data of father\n",
      "found a heterozygote 32039007_A_AAG in father with VAF 0.3877551020408163\n",
      "32039007_A_AAG not found in gt_hc of 77095341 - the mother of 77095338\n",
      "32039007_A_AAG not found in gt_dv of 77095341 - the mother of 77095338\n",
      "no 32039007_A_AAG found in mother\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039015_G_T in 77095338, checking inheritance; variant info: 1|0, 0.18518518518518517, 27\n",
      "{'sibling': 77095337, 'father': 77095339, 'mother': 77095341}\n",
      "found 32039015_G_T in 77095339 HC data - ['0|1', 0.43023255813953487, 86]\n",
      "32039015_G_T found only in HC data of father\n",
      "found a heterozygote 32039015_G_T in father with VAF 0.43023255813953487\n",
      "32039015_G_T not found in gt_hc of 77095341 - the mother of 77095338\n",
      "32039015_G_T not found in gt_dv of 77095341 - the mother of 77095338\n",
      "no 32039015_G_T found in mother\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039020_G_A in 77095338, checking inheritance; variant info: 1|0, 0.18518518518518517, 27\n",
      "{'sibling': 77095337, 'father': 77095339, 'mother': 77095341}\n",
      "found 32039020_G_A in 77095339 HC data - ['0|1', 0.46153846153846156, 91]\n",
      "32039020_G_A found only in HC data of father\n",
      "found a heterozygote 32039020_G_A in father with VAF 0.46153846153846156\n",
      "32039020_G_A not found in gt_hc of 77095341 - the mother of 77095338\n",
      "32039020_G_A not found in gt_dv of 77095341 - the mother of 77095338\n",
      "no 32039020_G_A found in mother\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039027_C_A in 77095338, checking inheritance; variant info: 1|0, 0.14705882352941177, 34\n",
      "{'sibling': 77095337, 'father': 77095339, 'mother': 77095341}\n",
      "found 32039027_C_A in 77095339 HC data - ['0|1', 0.376, 125]\n",
      "32039027_C_A found only in HC data of father\n",
      "found a heterozygote 32039027_C_A in father with VAF 0.376\n",
      "32039027_C_A not found in gt_hc of 77095341 - the mother of 77095338\n",
      "32039027_C_A not found in gt_dv of 77095341 - the mother of 77095338\n",
      "no 32039027_C_A found in mother\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039056_A_G in 77095338, checking inheritance; variant info: 1|0, 0.17647058823529413, 136\n",
      "{'sibling': 77095337, 'father': 77095339, 'mother': 77095341}\n",
      "found 32039056_A_G in 77095339 HC data - ['0|1', 0.397212543554007, 287]\n",
      "32039056_A_G found only in HC data of father\n",
      "found a heterozygote 32039056_A_G in father with VAF 0.397212543554007\n",
      "32039056_A_G not found in gt_hc of 77095341 - the mother of 77095338\n",
      "32039056_A_G not found in gt_dv of 77095341 - the mother of 77095338\n",
      "no 32039056_A_G found in mother\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038903_A_G in 77095339, checking inheritance; variant info: 0|1, 0.39285714285714285, 28\n",
      "{'child': 77095337, 'spouse': 77095341}\n",
      "32038903_A_G not found in gt_hc of 77095337 - the child of 77095339\n",
      "32038903_A_G not found in gt_dv of 77095337 - the child of 77095339\n",
      "no 32038903_A_G found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038938_T_G in 77095339, checking inheritance; variant info: 0|1, 0.5217391304347826, 23\n",
      "{'child': 77095337, 'spouse': 77095341}\n",
      "32038938_T_G not found in gt_hc of 77095337 - the child of 77095339\n",
      "32038938_T_G not found in gt_dv of 77095337 - the child of 77095339\n",
      "no 32038938_T_G found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038948_T_TC in 77095339, checking inheritance; variant info: 0|1, 0.4074074074074074, 27\n",
      "{'child': 77095337, 'spouse': 77095341}\n",
      "32038948_T_TC not found in gt_hc of 77095337 - the child of 77095339\n",
      "32038948_T_TC not found in gt_dv of 77095337 - the child of 77095339\n",
      "no 32038948_T_TC found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038962_A_ATCC in 77095339, checking inheritance; variant info: 0|1, 0.36, 25\n",
      "{'child': 77095337, 'spouse': 77095341}\n",
      "32038962_A_ATCC not found in gt_hc of 77095337 - the child of 77095339\n",
      "32038962_A_ATCC not found in gt_dv of 77095337 - the child of 77095339\n",
      "no 32038962_A_ATCC found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038971_C_A in 77095339, checking inheritance; variant info: 0|1, 0.30434782608695654, 23\n",
      "{'child': 77095337, 'spouse': 77095341}\n",
      "32038971_C_A not found in gt_hc of 77095337 - the child of 77095339\n",
      "32038971_C_A not found in gt_dv of 77095337 - the child of 77095339\n",
      "no 32038971_C_A found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038983_TG_T in 77095339, checking inheritance; variant info: 0|1, 0.3684210526315789, 19\n",
      "{'child': 77095337, 'spouse': 77095341}\n",
      "32038983_TG_T not found in gt_hc of 77095337 - the child of 77095339\n",
      "32038983_TG_T not found in gt_dv of 77095337 - the child of 77095339\n",
      "no 32038983_TG_T found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038983_TG_T in 77095340, checking inheritance; variant info: 0|1, 0.6296296296296297, 27\n",
      "{'child': 77095335, 'spouse': 77095334}\n",
      "found 32038983_TG_T in 77095335 HC data - ['1|1', 1.0, 16]\n",
      "32038983_TG_T found only in HC data of child\n",
      "found a lowqual homozygote 32038983_TG_T in child with right VAF 1.0, but low DP - 16\n",
      "found only low qual 32038983_TG_T in child with 1.0 and 16, probably an artefact\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038997_TGG_T in 77095340, checking inheritance; variant info: 0|1, 0.47058823529411764, 17\n",
      "{'child': 77095335, 'spouse': 77095334}\n",
      "found 32038997_TGG_T in 77095335 HC data - ['1|1', 1.0, 8]\n",
      "32038997_TGG_T found only in HC data of child\n",
      "found a lowqual homozygote 32038997_TGG_T in child with right VAF 1.0, but low DP - 8\n",
      "found only low qual 32038997_TGG_T in child with 1.0 and 8, probably an artefact\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039001_G_C in 77095340, checking inheritance; variant info: 0|1, 0.5, 14\n",
      "{'child': 77095335, 'spouse': 77095334}\n",
      "found 32039001_G_C in 77095335 HC data - ['1|1', 1.0, 4]\n",
      "32039001_G_C found only in HC data of child\n",
      "found a lowqual homozygote 32039001_G_C in child with right VAF 1.0, but low DP - 4\n",
      "found only low qual 32039001_G_C in child with 1.0 and 4, probably an artefact\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039002_G_A in 77095340, checking inheritance; variant info: 0|1, 0.5, 14\n",
      "{'child': 77095335, 'spouse': 77095334}\n",
      "found 32039002_G_A in 77095335 HC data - ['1|1', 1.0, 5]\n",
      "checking relative variant with ['1|1', 1.0, 5] AND ['1/1', 0.9629629629629629, 27]\n",
      "found 32039002_G_A in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039007_A_AAG in 77095340, checking inheritance; variant info: 0|1, 0.5, 12\n",
      "{'child': 77095335, 'spouse': 77095334}\n",
      "found 32039007_A_AAG in 77095335 HC data - ['1|1', 1.0, 4]\n",
      "32039007_A_AAG found only in HC data of child\n",
      "found a lowqual homozygote 32039007_A_AAG in child with right VAF 1.0, but low DP - 4\n",
      "found only low qual 32039007_A_AAG in child with 1.0 and 4, probably an artefact\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039015_G_T in 77095340, checking inheritance; variant info: 0|1, 0.45454545454545453, 11\n",
      "{'child': 77095335, 'spouse': 77095334}\n",
      "found 32039015_G_T in 77095335 HC data - ['1|1', 1.0, 5]\n",
      "checking relative variant with ['1|1', 1.0, 5] AND ['1/1', 0.9444444444444444, 72]\n",
      "found 32039015_G_T in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039020_G_A in 77095340, checking inheritance; variant info: 0|1, 0.45454545454545453, 11\n",
      "{'child': 77095335, 'spouse': 77095334}\n",
      "found 32039020_G_A in 77095335 HC data - ['1|1', 1.0, 7]\n",
      "checking relative variant with ['1|1', 1.0, 7] AND ['1/1', 0.9863013698630136, 73]\n",
      "found 32039020_G_A in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039027_C_A in 77095340, checking inheritance; variant info: 0|1, 0.6666666666666666, 18\n",
      "{'child': 77095335, 'spouse': 77095334}\n",
      "found 32039027_C_A in 77095335 HC data - ['1|1', 1.0, 7]\n",
      "checking relative variant with ['1|1', 1.0, 7] AND ['1/1', 0.9130434782608695, 92]\n",
      "found 32039027_C_A in child with genotype 1|1\n",
      "['1|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038962_A_ATCC in 77095342, checking inheritance; variant info: 0|1, 0.17857142857142858, 28\n",
      "{'spouse': 77095343, 'child': 77095344}\n",
      "found 32038962_A_ATCC in 77095344 HC data - ['1|0', 0.35, 40]\n",
      "32038962_A_ATCC found only in HC data of child\n",
      "found a heterozygote 32038962_A_ATCC in child with VAF 0.35\n",
      "['1|0']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038983_TG_T in 77095342, checking inheritance; variant info: 0|1, 0.10714285714285714, 28\n",
      "{'spouse': 77095343, 'child': 77095344}\n",
      "32038983_TG_T not found in gt_hc of 77095344 - the child of 77095342\n",
      "32038983_TG_T not found in gt_dv of 77095344 - the child of 77095342\n",
      "no 32038983_TG_T found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038997_TGG_T in 77095342, checking inheritance; variant info: 0|1, 0.1282051282051282, 39\n",
      "{'spouse': 77095343, 'child': 77095344}\n",
      "32038997_TGG_T not found in gt_hc of 77095344 - the child of 77095342\n",
      "32038997_TGG_T not found in gt_dv of 77095344 - the child of 77095342\n",
      "no 32038997_TGG_T found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039001_G_C in 77095342, checking inheritance; variant info: 0|1, 0.15384615384615385, 39\n",
      "{'spouse': 77095343, 'child': 77095344}\n",
      "32039001_G_C not found in gt_hc of 77095344 - the child of 77095342\n",
      "32039001_G_C not found in gt_dv of 77095344 - the child of 77095342\n",
      "no 32039001_G_C found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039002_G_A in 77095342, checking inheritance; variant info: 0|1, 0.15384615384615385, 39\n",
      "{'spouse': 77095343, 'child': 77095344}\n",
      "32039002_G_A not found in gt_hc of 77095344 - the child of 77095342\n",
      "32039002_G_A not found in gt_dv of 77095344 - the child of 77095342\n",
      "no 32039002_G_A found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038971_C_A in 77095344, checking inheritance; variant info: 1|0, 0.36363636363636365, 22\n",
      "{'father': 77095343, 'mother': 77095342}\n",
      "32038971_C_A not found in gt_hc of 77095343 - the father of 77095344\n",
      "32038971_C_A not found in gt_dv of 77095343 - the father of 77095344\n",
      "no 32038971_C_A found in father\n",
      "32038971_C_A not found in gt_hc of 77095342 - the mother of 77095344\n",
      "32038971_C_A not found in gt_dv of 77095342 - the mother of 77095344\n",
      "no 32038971_C_A found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32039015_G_T in 77095344, checking inheritance; variant info: 1|0, 0.15384615384615385, 13\n",
      "{'father': 77095343, 'mother': 77095342}\n",
      "32039015_G_T not found in gt_hc of 77095343 - the father of 77095344\n",
      "32039015_G_T not found in gt_dv of 77095343 - the father of 77095344\n",
      "no 32039015_G_T found in father\n",
      "found 32039015_G_T in 77095342 HC data - ['0|1', 0.38095238095238093, 84]\n",
      "32039015_G_T found only in HC data of mother\n",
      "found a heterozygote 32039015_G_T in mother with VAF 0.38095238095238093\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039020_G_A in 77095344, checking inheritance; variant info: 1|0, 0.21428571428571427, 14\n",
      "{'father': 77095343, 'mother': 77095342}\n",
      "32039020_G_A not found in gt_hc of 77095343 - the father of 77095344\n",
      "32039020_G_A not found in gt_dv of 77095343 - the father of 77095344\n",
      "no 32039020_G_A found in father\n",
      "found 32039020_G_A in 77095342 HC data - ['0|1', 0.3977272727272727, 88]\n",
      "32039020_G_A found only in HC data of mother\n",
      "found a heterozygote 32039020_G_A in mother with VAF 0.3977272727272727\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039027_C_A in 77095344, checking inheritance; variant info: 1|0, 0.21428571428571427, 14\n",
      "{'father': 77095343, 'mother': 77095342}\n",
      "32039027_C_A not found in gt_hc of 77095343 - the father of 77095344\n",
      "32039027_C_A not found in gt_dv of 77095343 - the father of 77095344\n",
      "no 32039027_C_A found in father\n",
      "found 32039027_C_A in 77095342 HC data - ['0|1', 0.4017094017094017, 117]\n",
      "32039027_C_A found only in HC data of mother\n",
      "found a heterozygote 32039027_C_A in mother with VAF 0.4017094017094017\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32039046_A_G in 77095344, checking inheritance; variant info: 1|0, 0.4117647058823529, 17\n",
      "{'father': 77095343, 'mother': 77095342}\n",
      "32039046_A_G not found in gt_hc of 77095343 - the father of 77095344\n",
      "32039046_A_G not found in gt_dv of 77095343 - the father of 77095344\n",
      "no 32039046_A_G found in father\n",
      "found 32039046_A_G in 77095342 HC data - ['0|1', 0.47282608695652173, 184]\n",
      "32039046_A_G found only in HC data of mother\n",
      "found a heterozygote 32039046_A_G in mother with VAF 0.47282608695652173\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "for 77095346 comparing 32038844_A_C with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.8739130434782608, DP-230\n",
      "for 77095346 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "both wrong, checking inheritance; variant info: 1|0, 0.3, 10 AND 1/1, 0.5228426395939086, 197\n",
      "{'sibling': 77095345}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77095346 comparing 32038857_A_T with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.9657534246575342, DP-146\n",
      "for 77095346 comparing 32038949_T_G with HC - 1|1 and DV - 1/0\n",
      "hc right with 1|1, VAF - 1.0, DP-182\n",
      "low qual 32038949_T_C in 77095346, checking inheritance; variant info: 1/0, 0.1509433962264151, 54\n",
      "{'sibling': 77095345}\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038948_T_TC in 77095350, checking inheritance; variant info: 0|1, 0.15675675675675677, 185\n",
      "{'spouse': 77095347, 'child': 77095354}\n",
      "32038948_T_TC not found in gt_hc of 77095354 - the child of 77095350\n",
      "32038948_T_TC not found in gt_dv of 77095354 - the child of 77095350\n",
      "no 32038948_T_TC found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038949_T_G in 77095350, checking inheritance; variant info: 0|1, 0.15675675675675677, 185\n",
      "{'spouse': 77095347, 'child': 77095354}\n",
      "32038949_T_G not found in gt_hc of 77095354 - the child of 77095350\n",
      "32038949_T_G not found in gt_dv of 77095354 - the child of 77095350\n",
      "no 32038949_T_G found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "low qual 32038955_A_T in 77095350, checking inheritance; variant info: 0|1, 0.16111111111111112, 180\n",
      "{'spouse': 77095347, 'child': 77095354}\n",
      "32038955_A_T not found in gt_hc of 77095354 - the child of 77095350\n",
      "32038955_A_T not found in gt_dv of 77095354 - the child of 77095350\n",
      "no 32038955_A_T found in child\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77095350 comparing 32039081_C_A with HC - 1|1 and DV - 1/0\n",
      "dv right with 1/0, VAF - 0.4936708860759494, DP-163\n",
      "for 77095354 comparing 32039015_G_A with HC - 1|1 and DV - 0/1\n",
      "hc right with 1|1, VAF - 1.0, DP-46\n",
      "for 77095354 comparing 32039027_C_G with HC - 1|1 and DV - 1/0\n",
      "hc right with 1|1, VAF - 0.9807692307692307, DP-52\n",
      "for 77095354 comparing 32039081_C_A with HC - 1|1 and DV - 1/0\n",
      "hc right with 1|1, VAF - 1.0, DP-243\n",
      "low qual 32038855_T_TTTG in 77095354, checking inheritance; variant info: 0/1, 0.17982456140350878, 228\n",
      "{'mother': 77095347, 'sibling': 77095349, 'father': 77095350}\n",
      "32038855_T_TTTG not found in gt_hc of 77095347 - the mother of 77095354\n",
      "32038855_T_TTTG not found in gt_dv of 77095347 - the mother of 77095354\n",
      "no 32038855_T_TTTG found in mother\n",
      "found 32038855_T_TTTG in 77095350 HC data - ['0|1', 0.23478260869565218, 115]\n",
      "checking relative variant with ['0|1', 0.23478260869565218, 115] AND ['0/1', 0.17903930131004367, 229]\n",
      "found 32038855_T_TTTG in father with genotype 0|1\n",
      "['0|1']\n",
      "saving this variant even with bad QUAL\n",
      "low qual 32038855_T_TTG in 77095360, checking inheritance; variant info: 1|0, 0.14919735599622286, 2118\n",
      "{'sibling': 77095358, 'father': 77095361, 'mother': 77095363}\n",
      "32038855_T_TTG not found in gt_hc of 77095361 - the father of 77095360\n",
      "32038855_T_TTG not found in gt_dv of 77095361 - the father of 77095360\n",
      "no 32038855_T_TTG found in father\n",
      "32038855_T_TTG not found in gt_hc of 77095363 - the mother of 77095360\n",
      "32038855_T_TTG not found in gt_dv of 77095363 - the mother of 77095360\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77095360 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "dv right with 1/1, VAF - 0.8582089552238806, DP-134\n",
      "low qual 32038855_T_TTG in 77095362, checking inheritance; variant info: 1|0, 0.1612769485903814, 2419\n",
      "{'sibling': 77095358, 'father': 77095361, 'mother': 77095363}\n",
      "32038855_T_TTG not found in gt_hc of 77095361 - the father of 77095362\n",
      "32038855_T_TTG not found in gt_dv of 77095361 - the father of 77095362\n",
      "no 32038855_T_TTG found in father\n",
      "32038855_T_TTG not found in gt_hc of 77095363 - the mother of 77095362\n",
      "32038855_T_TTG not found in gt_dv of 77095363 - the mother of 77095362\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77095362 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8387230514096186, DP-2419\n",
      "for 77095362 comparing 32041186_C_T with HC - 1|1 and DV - 0/1\n",
      "hc right with 1|1, VAF - 0.9983733224888166, DP-2459\n",
      "low qual 32038855_T_TTG in 77095384, checking inheritance; variant info: 1|0, 0.15266168059832821, 2286\n",
      "{'father': 77095383, 'mother': 77095381, 'sibling': 77095382}\n",
      "32038855_T_TTG not found in gt_hc of 77095383 - the father of 77095384\n",
      "32038855_T_TTG not found in gt_dv of 77095383 - the father of 77095384\n",
      "no 32038855_T_TTG found in father\n",
      "32038855_T_TTG not found in gt_hc of 77095381 - the mother of 77095384\n",
      "32038855_T_TTG not found in gt_dv of 77095381 - the mother of 77095384\n",
      "no 32038855_T_TTG found in mother\n",
      "[]\n",
      "droping this variant, probably artefact\n",
      "for 77095384 comparing 32038855_T_TTTG with HC - 1|0 and DV - 1/1\n",
      "hc right with 1|0, VAF - 0.8473383194016718, DP-2286\n"
     ]
    }
   ],
   "source": [
    "res = defaultdict(dict)\n",
    "for idx, row in files_list.iterrows():\n",
    "    # dv\n",
    "    vcf = read_vcf(row['dv'], mode = 'dv')\n",
    "    vcf = vcf[vcf['FILTER'] == 'PASS']\n",
    "    vcf = vcf[(vcf['POS'] >= calling_region[0]) & (vcf['POS'] <= calling_region[1])]\n",
    "    sample_dv=gt_vectors(vcf, mode = 'dv')\n",
    "    # hc \n",
    "    vcf = read_vcf(row['hc'], mode='hc')\n",
    "    vcf = vcf[(vcf['POS'] >= calling_region[0]) & (vcf['POS'] <= calling_region[1])]\n",
    "    sample_hc=gt_vectors(vcf, mode= 'hc')\n",
    "\n",
    "    #pileup\n",
    "    vcf = read_vcf(row['pileup'], mode='bam')\n",
    "    vcf = vcf[(vcf['POS'] >= calling_region[0]) & (vcf['POS'] <= calling_region[1])]\n",
    "    sample_pp=gt_vectors(vcf, mode= 'dv')\n",
    "\n",
    "    for position in sample_hc.keys():\n",
    "        hc_value=sample_hc[position][0]\n",
    "        if position in sample_dv.keys():\n",
    "            dv_value=sample_dv[position][0]\n",
    "            res[idx][position] = check_variant(idx, position, hc_value, dv_value, sample_hc, sample_dv, trios, gt_hc, gt_dv)\n",
    "        else:\n",
    "            hc_value=sample_hc[position][0]\n",
    "            hc=hc_value.split('|')\n",
    "            hc=int(hc[0])+int(hc[1])\n",
    "            if hc == 2:\n",
    "                if sample_hc[position][1]>0.85 and sample_hc[position][2]>=30:\n",
    "                    # print(f'{position} not called by DV, saving it with {hc_value}, {sample_hc[position][1]}, {sample_hc[position][2]}')\n",
    "                    res[idx][position] = hc_value\n",
    "                else:\n",
    "                    print(f'low qual {position} in {idx}, checking inheritance; variant info: {hc_value}, {sample_hc[position][1]}, {sample_hc[position][2]}')\n",
    "                    inheritance = check_inheritance(idx, position, trios, gt_hc, gt_dv)\n",
    "                    if inheritance == True:\n",
    "                        print('saving this variant even with bad QUAL')\n",
    "                        res[idx][position] = hc_value\n",
    "                    else:\n",
    "                        print('droping this variant, probably artefact')\n",
    "            if hc == 1:\n",
    "                if (sample_hc[position][1] >= 0.2 and sample_hc[position][1] < 0.85) and sample_hc[position][2]>=30:\n",
    "                    # print(f'{position} not called by DV, saving it with {hc_value}, {sample_hc[position][1]}, {sample_hc[position][2]}')\n",
    "                    res[idx][position] = hc_value\n",
    "                else:\n",
    "                    print(f'low qual {position} in {idx}, checking inheritance; variant info: {hc_value}, {sample_hc[position][1]}, {sample_hc[position][2]}')\n",
    "                    inheritance = check_inheritance(idx, position, trios, gt_hc, gt_dv)\n",
    "                    if inheritance == True:\n",
    "                        print('saving this variant even with bad QUAL')\n",
    "                        res[idx][position] = hc_value\n",
    "                    else:\n",
    "                        print('droping this variant, probably artefact')\n",
    "    for position in sample_dv.keys():\n",
    "        if position not in res[idx].keys():\n",
    "            dv_value=sample_dv[position][0]\n",
    "            dv=dv_value.split('/')\n",
    "            dv=int(dv[0])+int(dv[1])\n",
    "            if dv == 2:\n",
    "                if sample_dv[position][1]> 0.85 and sample_dv[position][2]>=30:\n",
    "                    # print(f'{position} not called by HC, saving it with {dv_value}, {sample_dv[position][1]}, {sample_dv[position][2]}')\n",
    "                    res[idx][position] = dv_value\n",
    "                else:\n",
    "                    print(f'low qual {position} in {idx}, checking inheritance; variant info: {dv_value}, {sample_dv[position][1]}, {sample_dv[position][2]}')\n",
    "                    inheritance = check_inheritance(idx, position, trios, gt_hc, gt_dv)\n",
    "                    if inheritance == True:\n",
    "                        print('saving this variant even with bad QUAL')\n",
    "                        res[idx][position] = dv_value\n",
    "                    else:\n",
    "                        print('droping this variant, probably artefact')\n",
    "            if dv == 1:\n",
    "                if (sample_dv[position][1] >= 0.2 and sample_dv[position][1] < 0.85) and sample_dv[position][2]>=30:\n",
    "                    # print(f'{position} not called by HC, saving it with {dv_value}, {sample_dv[position][1]}, {sample_dv[position][2]}')\n",
    "                    res[idx][position] = dv_value\n",
    "                else:\n",
    "                    print(f'low qual {position} in {idx}, checking inheritance; variant info: {dv_value}, {sample_dv[position][1]}, {sample_dv[position][2]}')\n",
    "                    inheritance = check_inheritance(idx, position, trios, gt_hc, gt_dv)\n",
    "                    if inheritance == True:\n",
    "                        print('saving this variant even with bad QUAL')\n",
    "                        res[idx][position] = dv_value\n",
    "                    else:\n",
    "                        print('droping this variant, probably artefact')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "left_aligned=defaultdict(dict)\n",
    "for idx, row in files_list.iterrows():\n",
    "    # dv\n",
    "    vcf = read_vcf(row['dv'], mode = 'dv')\n",
    "    vcf = vcf[vcf['FILTER'] == 'PASS']\n",
    "    vcf = vcf[(vcf['POS'] >= calling_region[0]) & (vcf['POS'] <= calling_region[1])]\n",
    "    if 32038855 in list(vcf['POS']):\n",
    "        left_aligned[str(idx)+'_DV']={'POS': 32038855, \n",
    "                           'REF': vcf.loc[vcf['POS']==32038855, 'REF'].values[0],\n",
    "                           'ALT': vcf.loc[vcf['POS']==32038855, 'ALT'].values[0],\n",
    "                           'GT': vcf.loc[vcf['POS']==32038855, 'GT'].values[0],\n",
    "                           'AD': vcf.loc[vcf['POS']==32038855, 'AD'].values[0]}\n",
    "    # hc \n",
    "    vcf = read_vcf(row['hc'], mode='hc')\n",
    "    vcf = vcf[(vcf['POS'] >= calling_region[0]) & (vcf['POS'] <= calling_region[1])]\n",
    "    if 32038855 in list(vcf['POS']):\n",
    "        left_aligned[str(idx)+'_HC']={'POS': 32038855, \n",
    "                           'REF': vcf.loc[vcf['POS']==32038855, 'REF'].values[0],\n",
    "                           'ALT': vcf.loc[vcf['POS']==32038855, 'ALT'].values[0],\n",
    "                           'GT': vcf.loc[vcf['POS']==32038855, 'GT'].values[0],\n",
    "                           'AD': vcf.loc[vcf['POS']==32038855, 'AD'].values[0]}\n",
    "    # gt_hc[idx]=gt_vectors(vcf, mode= 'hc')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "conf=pd.DataFrame.from_dict(conflicts).transpose()\n",
    "conf.to_excel('/storage/haplotypes/excels/conflicts.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.DataFrame.from_dict(res)\n",
    "df.columns = df.columns.astype(str)\n",
    "df['pos'] = df.index.to_series().str.extract(r'^(\\d+)').astype(int)\n",
    "df['rest'] = df.index.to_series().str.extract(r'^\\d+_(.+)$')\n",
    "df_sorted = df.sort_values(by=['pos', 'rest']).drop(columns=['pos', 'rest'])\n",
    "df_sorted.to_excel('/storage/haplotypes/excels/not_phased_sorted.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>77050386</th>\n",
       "      <th>77091048</th>\n",
       "      <th>77092306</th>\n",
       "      <th>77092307</th>\n",
       "      <th>77092308</th>\n",
       "      <th>77092309</th>\n",
       "      <th>77092310</th>\n",
       "      <th>77092311</th>\n",
       "      <th>77092312</th>\n",
       "      <th>77092313</th>\n",
       "      <th>...</th>\n",
       "      <th>77095381</th>\n",
       "      <th>77095382</th>\n",
       "      <th>77095383</th>\n",
       "      <th>77095384</th>\n",
       "      <th>77095385</th>\n",
       "      <th>77095386</th>\n",
       "      <th>77095387</th>\n",
       "      <th>77095388</th>\n",
       "      <th>77095389</th>\n",
       "      <th>77095390</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32038302_C_T</th>\n",
       "      <td>0|1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32038437_CCTG_C</th>\n",
       "      <td>1|0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0|1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1|0</td>\n",
       "      <td>...</td>\n",
       "      <td>0|1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32038540_C_T</th>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1|0</td>\n",
       "      <td>...</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32038560_C_A</th>\n",
       "      <td>1|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1|0</td>\n",
       "      <td>...</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1|0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32038820_C_T</th>\n",
       "      <td>0|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1|0</td>\n",
       "      <td>...</td>\n",
       "      <td>1|0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 571 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                77050386 77091048 77092306 77092307 77092308 77092309  \\\n",
       "32038302_C_T         0|1      NaN      NaN      NaN      NaN      NaN   \n",
       "32038437_CCTG_C      1|0      NaN      NaN      NaN      NaN      NaN   \n",
       "32038540_C_T         1|1      1|1      1|0      0|1      1|1      NaN   \n",
       "32038560_C_A         1|1      0|1      1|0      0|1      1|1      NaN   \n",
       "32038820_C_T         0|1      1|1      NaN      0|1      0|1      NaN   \n",
       "\n",
       "                77092310 77092311 77092312 77092313  ... 77095381 77095382  \\\n",
       "32038302_C_T         NaN      NaN      NaN      NaN  ...      NaN      NaN   \n",
       "32038437_CCTG_C      NaN      0|1      NaN      1|0  ...      0|1      NaN   \n",
       "32038540_C_T         0|1      0|1      NaN      1|0  ...      0|1      0|1   \n",
       "32038560_C_A         0|1      0|1      NaN      1|0  ...      0|1      0|1   \n",
       "32038820_C_T         1|0      0|1      NaN      1|0  ...      1|0      NaN   \n",
       "\n",
       "                77095383 77095384 77095385 77095386 77095387 77095388  \\\n",
       "32038302_C_T         NaN      NaN      NaN      NaN      NaN      NaN   \n",
       "32038437_CCTG_C      NaN      NaN      NaN      1|1      1|1      1|0   \n",
       "32038540_C_T         1|0      NaN      1|1      1|1      1|1      1|0   \n",
       "32038560_C_A         1|0      NaN      1|1      1|1      1|1      1|0   \n",
       "32038820_C_T         NaN      NaN      NaN      1|1      1|1      0|1   \n",
       "\n",
       "                77095389 77095390  \n",
       "32038302_C_T         NaN      NaN  \n",
       "32038437_CCTG_C      NaN      NaN  \n",
       "32038540_C_T         0|1      1|0  \n",
       "32038560_C_A         NaN      1|0  \n",
       "32038820_C_T         1|1      1|0  \n",
       "\n",
       "[5 rows x 571 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_af=df.copy()\n",
    "df_af.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "to_phase= df.loc[:, df.apply(lambda col: col.isin(['1/0', '0/1']).any())]\n",
    "to_phase.to_excel('/storage/haplotypes/excels/to_phase.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_cell(value):\n",
    "    if pd.isna(value):\n",
    "        return 0\n",
    "    # Replace both '/' and '|' with '+' and evaluate\n",
    "    try:\n",
    "        return eval(str(value).replace('/', '+').replace('|', '+'))\n",
    "    except:\n",
    "        return 0 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2772568/3509597075.py:1: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  processed_df = df.applymap(process_cell)\n"
     ]
    }
   ],
   "source": [
    "processed_df = df.applymap(process_cell)\n",
    "processed_df['haplotype_count'] = processed_df.sum(axis=1)\n",
    "processed_df['AF_cohort'] = processed_df['haplotype_count']/1142\n",
    "processed_df.to_excel('/storage/haplotypes/excels/w_af.xlsx')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'father': 77092368, 'mother': 77092371}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trios[77092370]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['1|1', 1.0, 25] NONE ['1/0', 0.7857142857142857, 168] ['1/0', 0.19642857142857142, 168] ['1/0', 0.35555555555555557, 90] ['1/0', 0.5666666666666667, 90]\n"
     ]
    }
   ],
   "source": [
    "print(gt_hc[77092370]['32038949_T_G'], 'NONE', gt_dv[77092370]['32038949_T_G'], gt_dv[77092370]['32038949_T_C'], gt_bam[77092370]['32038949_T_G'], gt_bam[77092370]['32038949_T_C'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['1|0', 0.5511111111111111, 225] NONE NONE NONE NONE NONE\n"
     ]
    }
   ],
   "source": [
    "print(gt_hc[77092368]['32038949_T_G'], 'NONE', 'NONE', 'NONE', 'NONE', 'NONE')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['1|0', 0.5049035115469788, 3161] NONE NONE NONE ['0/1', 0.26745240253853125, 1103] NONE\n"
     ]
    }
   ],
   "source": [
    "print(gt_hc[77092371]['32038949_T_G'], 'NONE', 'NONE', 'NONE', gt_bam[77092371]['32038949_T_G'], 'NONE')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'mother': 77095347, 'sibling': 77095349, 'father': 77095350}"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trios[77095354] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HC 32039015_G_T NONE ['1|1', 1.0, 46] DV 32039015_G_T NONE ['0/1', 0.7916666666666666, 72] ['1/0', 0.3144208037825059, 846] ['1/0', 0.5638297872340425, 846]\n"
     ]
    }
   ],
   "source": [
    "print('HC 32039015_G_T NONE', gt_hc[77095354]['32039015_G_A'], 'DV 32039015_G_T NONE', gt_dv[77095354]['32039015_G_A'], gt_bam[77095354]['32039015_G_T'], gt_bam[77095354]['32039015_G_A'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(gt_hc[77095354]['32039015_G_T'], gt_hc[77095354]['32039015_G_A'], gt_hc[77095354]['32039015_G_T'], gt_dv[77095354]['32039015_G_A'], gt_bam[77095354]['32039015_G_T'], gt_bam[77095354]['32039015_G_A'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "def manual_checking(patient, pos1, pos2, relatives):\n",
    "    print(gt_hc[patient].get(pos1, f'{pos1} not found in HC data of {patient}'))\n",
    "    print(gt_dv[patient].get(pos1, f'{pos1} not found in DV data of {patient}'))\n",
    "    print(gt_bam[patient].get(pos1, f'{pos1} not found in bam data of {patient}'))\n",
    "    print(gt_hc[patient].get(pos2, f'{pos2} not found in HC data of {patient}'))\n",
    "    print(gt_dv[patient].get(pos2, f'{pos1} not found in DV data of {patient}'))\n",
    "    print(gt_bam[patient].get(pos2, f'{pos1} not found in bam data of {patient}'))\n",
    "    for rel in relatives[patient]:\n",
    "        rel_num=int(relatives[patient][rel])\n",
    "        print(f'searching in {rel_num}  - the {rel} of {patient}')\n",
    "        print(gt_hc[rel_num].get(pos1, f'{pos1} not found in HC data of {rel_num}'))\n",
    "        print(gt_dv[rel_num].get(pos1, f'{pos1} not found in DV data of {rel_num}'))\n",
    "        print(gt_bam[rel_num].get(pos1, f'{pos1} not found in bam data of {rel_num}'))\n",
    "        print(gt_hc[rel_num].get(pos2, f'{pos2} not found in HC data of {rel_num}'))\n",
    "        print(gt_dv[rel_num].get(pos2, f'{pos1} not found in DV data of {rel_num}'))\n",
    "        print(gt_bam[rel_num].get(pos2, f'{pos1} not found in bam data of {rel_num}'))\n",
    "\n",
    "\n",
    "    return None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32039002_G_A not found in HC data of 77092531\n",
      "['1/1', 0.9487179487179487, 39]\n",
      "32039002_G_A not found in bam data of 77092531\n",
      "32039015_G_A not found in HC data of 77092531\n",
      "32039002_G_A not found in DV data of 77092531\n",
      "32039002_G_A not found in bam data of 77092531\n",
      "searching in 77092530  - the father of 77092531\n",
      "['0|1', 0.23255813953488372, 43]\n",
      "32039002_G_A not found in DV data of 77092530\n",
      "32039002_G_A not found in bam data of 77092530\n",
      "32039015_G_A not found in HC data of 77092530\n",
      "32039002_G_A not found in DV data of 77092530\n",
      "32039002_G_A not found in bam data of 77092530\n",
      "searching in 77092529  - the mother of 77092531\n",
      "['0|1', 0.125, 24]\n",
      "32039002_G_A not found in DV data of 77092529\n",
      "32039002_G_A not found in bam data of 77092529\n",
      "32039015_G_A not found in HC data of 77092529\n",
      "32039002_G_A not found in DV data of 77092529\n",
      "32039002_G_A not found in bam data of 77092529\n"
     ]
    }
   ],
   "source": [
    "manual_checking(77092531, '32039002_G_A', '32039015_G_A', trios)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "haplotypes",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
